diff -Naur ../work/mc-4.6.1/syntax/c.syntax mc-4.6.1/syntax/c.syntax
--- ../work/mc-4.6.1/syntax/c.syntax	2005-05-31 10:03:45.000000000 +0400
+++ mc-4.6.1/syntax/c.syntax	2006-01-17 14:00:52.000000000 +0300
@@ -32,34 +32,7 @@
     keyword whole volatile yellow
     keyword whole while yellow
     keyword whole asm yellow
-    keyword whole catch yellow
-    keyword whole class yellow
-    keyword whole friend yellow
-    keyword whole delete yellow
     keyword whole inline yellow
-    keyword whole new yellow
-    keyword whole operator yellow
-    keyword whole private yellow
-    keyword whole protected yellow
-    keyword whole public yellow
-    keyword whole this yellow
-    keyword whole throw yellow
-    keyword whole template yellow
-    keyword whole try yellow
-    keyword whole virtual yellow
-    keyword whole bool yellow
-    keyword whole const_cast yellow
-    keyword whole dynamic_cast yellow
-    keyword whole explicit yellow
-    keyword whole false yellow
-    keyword whole mutable yellow
-    keyword whole namespace yellow
-    keyword whole reinterpret_cast yellow
-    keyword whole static_cast yellow
-    keyword whole true yellow
-    keyword whole typeid yellow
-    keyword whole typename yellow
-    keyword whole using yellow
     keyword whole wchar_t yellow
     keyword whole ... yellow
     keyword whole linestart \{\s\t\}\[\s\t\]#*\n brightmagenta
diff -Naur ../work/mc-4.6.1/syntax/Syntax mc-4.6.1/syntax/Syntax
--- ../work/mc-4.6.1/syntax/Syntax	2006-01-17 13:06:16.000000000 +0300
+++ mc-4.6.1/syntax/Syntax	2006-01-17 13:59:40.000000000 +0300
@@ -72,9 +72,12 @@
 file ..\*\.(texi|texinfo|TEXI|TEXINFO)$ Texinfo\sDocument
 include texinfo.syntax
 
-file ..\*\\.([chC]|CC|cxx|cc|cpp|CPP|CXX|hxx|h\.in|hh|hpp)$ C/C\+\+\sProgram
+file ..\*\\.c$ C\sProgram
 include c.syntax
 
+file ..\*\\.([hC]|CC|cxx|cc|cpp|CPP|CXX|hxx|h\.in)$ C/C\+\+\sProgram
+include cxx.syntax
+
 file ..\*\\.[fF]$ Fortran\sProgram
 include fortran.syntax
 
