--- lib/mc.menu	2005-05-02 18:08:11.000000000 +0200
+++ lib/mc.menu	2005-05-02 18:08:37.000000000 +0200
@@ -15,7 +15,7 @@ shell_patterns=0
 
 0       Edit a bug report and send it to root
	I=`mktemp ${MC_TMPDIR:-/tmp}/mail.XXXXXX` || exit 1
-	${EDITOR-vi} $I
+	${EDITOR-editor} $I
	test -r $I && mail root < $I
	rm -f $I
 
@@ -330,3 +330,7 @@
 o       Open next a free console
         open -s -- sh
 
+=+ f \.dsc$ & t r
+x       Extract the contents of a Debian source package
+        dpkg-source -x %f
+
--- src/util.h 2006-06-13 04:39:20.000000000 +0300
+++ src/util.h 2006-06-13 04:39:53.000000000 +0300
@@ -217,7 +217,7 @@ void save_file_position (const char *fil
 #define PATH_ENV_SEP ':'
 #define TMPDIR_DEFAULT "/tmp"
 #define SCRIPT_SUFFIX ""
-#define get_default_editor() "vi"
+#define get_default_editor() "editor"
 #define OS_SORT_CASE_SENSITIVE_DEFAULT 1
 #define STRCOMP strcmp
 #define STRNCOMP strncmp
