Index: source/shared_lib/sources/platform/posix/socket.cpp
===================================================================
--- source/shared_lib/sources/platform/posix/socket.cpp	(revision 176)
+++ source/shared_lib/sources/platform/posix/socket.cpp	(working copy)
@@ -8,6 +8,8 @@
 
 #include "socket.h"
 
+#include <cstring>
+#include <cstdlib>
 #include <stdexcept>
 #include <sstream>
 #if defined(HAVE_SYS_IOCTL_H)
Index: source/shared_lib/sources/util/conversion.cpp
===================================================================
--- source/shared_lib/sources/util/conversion.cpp	(revision 176)
+++ source/shared_lib/sources/util/conversion.cpp	(working copy)
@@ -13,6 +13,7 @@
 
 #include <stdexcept>
 #include <cstdio>
+#include <cstdlib>
 
 #include "leak_dumper.h"
 
Index: source/shared_lib/sources/util/properties.cpp
===================================================================
--- source/shared_lib/sources/util/properties.cpp	(revision 176)
+++ source/shared_lib/sources/util/properties.cpp	(working copy)
@@ -13,6 +13,7 @@
 
 #include <fstream>
 #include <stdexcept>
+#include <cstring>
 
 #include "conversion.h"
 #include "leak_dumper.h"
Index: source/shared_lib/sources/util/util.cpp
===================================================================
--- source/shared_lib/sources/util/util.cpp	(revision 176)
+++ source/shared_lib/sources/util/util.cpp	(working copy)
@@ -14,6 +14,7 @@
 #include <ctime>
 #include <cassert>
 #include <stdexcept>
+#include <cstring>
 
 #include "leak_dumper.h"
 
