Index: ivman-0.6.14/src/manager.c
===================================================================
--- ivman-0.6.14.orig/src/manager.c	2007-03-07 04:01:36.000000000 -0500
+++ ivman-0.6.14/src/manager.c	2007-03-07 04:03:07.000000000 -0500
@@ -1042,19 +1042,19 @@
                                               "volume.mount_point",
                                               NULL);
 
-        while (( !mount_point || (strlen(mount_point) < 2) )
+        while (( !mount_point || (strlen(mount_point) < 1) )
                && (i < 5))
         {
             sleep(1);
             mount_point = (char *)
                 libhal_device_get_property_string(hal_ctx, udi,
-                                                  " volume.mount_point ",
+                                                  "volume.mount_point",
                                                   NULL);
             i++;
         }
 
 
-        if ( !mount_point || strlen(mount_point) < 2)
+        if ( !mount_point || strlen(mount_point) < 1)
         {
             DEBUG(_("Couldn't get mount point of device %s after "
                     "5 seconds\n"), device);
