--- udev-056/udev_add.c~	2005-03-19 02:39:30.000000000 -0500
+++ udev-056/udev_add.c	2005-04-15 16:23:01.000000000 -0400
@@ -221,7 +221,7 @@
 		dbg("symlink(%s, %s)", linktarget, filename);
 		if (!udev->test_run) {
 			unlink(filename);
-			selinux_setfscreatecon(filename, udev->kernel_name, S_IFLNK);
+			selinux_setfscreatecon(filename, NULL, S_IFLNK);
 			retval = symlink(linktarget, filename);
 			selinux_resetfscreatecon();
 			if (retval != 0)
