--- udev-048/udev_add.c.selinux	2004-12-08 14:43:17.000000000 -0500
+++ udev-048/udev_add.c	2004-12-21 16:50:43.849335229 -0500
@@ -92,6 +92,7 @@
 create:
 	selinux_setfscreatecon(file, mode);
 	retval = mknod(file, mode, makedev(major, minor));
+	selinux_restore();
 	if (retval != 0) {
 		dbg("mknod(%s, %#o, %u, %u) failed with error '%s'",
 		    file, mode, major, minor, strerror(errno));
@@ -242,6 +243,7 @@
 			if (symlink(linktarget, filename) != 0)
 				dbg("symlink(%s, %s) failed with error '%s'",
 				    linktarget, filename, strerror(errno));
+			selinux_restore();
 		}
 	}
 
@@ -335,7 +337,6 @@
 	}
 
 exit:
-	selinux_restore();
 
 	return retval;
 }
