diff --git a/bindings/Java/Makefile.am b/bindings/Java/Makefile.am
index a2ead24..1500dcd 100644
--- a/bindings/Java/Makefile.am
+++ b/bindings/Java/Makefile.am
@@ -67,6 +67,8 @@ java_classes= \
 	datebooktst.class	\
 	adrtest.class
 
+$(java_classes): java_compile
+
 java_compile: $(java_files)
 	$(JAVAC) -d . $(java_files)
 
@@ -81,16 +83,16 @@ java_pisock: java_compile java_lib java_jar
 install: all
 	ARCH=""; \
 	for a in i386 amd64 ppc; do \
-		if [ -e ${JAVABASE}/jre/lib/${a} ]; then \
-			 ARCH="${a}"; \
+		if [ -e ${JAVABASE}/jre/lib/$${a} ]; then \
+			 ARCH="$${a}"; \
 			 break; \
 		fi; \
 	done; \
-	if [ -n "${ARCH}" ]; then \
+	if [ -n "$${ARCH}" ]; then \
 		echo "Unknown architecture. Copy lipijsock.so to ${JAVABASE}/jre/lib/ARCH/"; \
 	else \
-		mkdir -p "$(DESTDIR)${JAVABASE}/jre/lib/${ARCH}" && \
-		cp libjpisock.so "$(DESTDIR)${JAVABASE}/jre/lib/${ARCH}"; \
+		mkdir -p "$(DESTDIR)${JAVABASE}/jre/lib/$${ARCH}" && \
+		cp libjpisock.so "$(DESTDIR)${JAVABASE}/jre/lib/$${ARCH}"; \
 		echo "done."; \
 	fi
  
diff --git a/configure.ac b/configure.ac
index 68a05f2..4da0626 100644
--- a/configure.ac
+++ b/configure.ac
@@ -220,6 +220,7 @@ AC_MSG_RESULT($enable_conduits)
 dnl ******************************
 dnl PNG Support
 dnl ******************************
+msg_png=no
 if test "x$enable_conduits" != "xno"; then
 	have_png=no
 
@@ -254,6 +255,7 @@ if test "x$enable_conduits" != "xno"; then
 			if test "$have_png" = yes; then
 				AC_DEFINE(HAVE_PNG, 1, [Define if we have PNG support])
 				have_png=yes
+				msg_png=yes
 			fi
 		fi
 	fi
@@ -351,12 +353,13 @@ PKG_CHECK_MODULES(BLUEZ, bluez, have_bluez=yes, AC_MSG_RESULT(no))
 AC_MSG_CHECKING([for BlueZ bluetooth support])
 AC_ARG_WITH(bluez,
                 AC_HELP_STRING([--with-bluez],
-                        [Enable usage of BlueZ]),,
+                        [Enable usage of BlueZ]),[use_bluez="$withval"],
                 use_bluez=auto)
 if test "x$use_bluez" != "xno"; then
         if test "x$have_bluez" = xno && test "x$use_bluez" = xyes; then
                 AC_MSG_ERROR([BlueZ explicitly requested but no support found])
         fi
+	use_bluez="$have_bluez"
         if test "x$have_bluez" = xyes; then
                 AC_DEFINE(HAVE_BLUEZ, 1, [Define if we have BlueZ bluetooth support])
                 AC_MSG_RESULT([enabled])
@@ -366,7 +369,7 @@ if test "x$use_bluez" != "xno"; then
         fi
 fi
 
-AM_CONDITIONAL(WITH_BLUEZ, test x$have_bluez = "xyes")
+AM_CONDITIONAL(WITH_BLUEZ, test x$use_bluez != "xno")
 
 
 dnl *********************************
diff --git a/pilot-link.m4.in b/pilot-link.m4.in
index 129dfc3..a271c74 100644
--- a/pilot-link.m4.in
+++ b/pilot-link.m4.in
@@ -1,5 +1,3 @@
-AC_DEFUN([IT_PROG_INTLTOOL], [AC_PREREQ([2.50])dnl
-
 AC_DEFUN([AC_PILOT_LINK_HOOK],
 [
 PILOT_LINK_VERS=0
