diff -urN liblinebreak-old/liblinebreak/Makefile liblinebreak/liblinebreak/Makefile
--- liblinebreak-old/liblinebreak/Makefile	2008-03-22 23:25:44.000000000 +0100
+++ liblinebreak/liblinebreak/Makefile	2008-04-05 03:12:41.000000000 +0200
@@ -39,16 +39,16 @@
 RELEASE := ReleaseDir
 
 $(DEBUG)/%.o: %.c
-	$(CC) $(CFLAGS) $(CPPFLAGS) $(DBGFLAGS) $(TARGET_ARCH) -c -o $@ $<
+	$(CC) $(CFLAGS) -fPIC $(CPPFLAGS) $(DBGFLAGS) $(TARGET_ARCH) -c -o $@ $<
 
 $(RELEASE)/%.o: %.c
-	$(CC) $(CFLAGS) $(CPPFLAGS) $(RELFLAGS) $(TARGET_ARCH) -c -o $@ $<
+	$(CC) $(CFLAGS) -fPIC $(CPPFLAGS) $(RELFLAGS) $(TARGET_ARCH) -c -o $@ $<
 
 $(DEBUG)/%.o: %.cpp
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(DBGFLAGS) $(TARGET_ARCH) -c -o $@ $<
+	$(CXX) $(CXXFLAGS) -fPIC $(CPPFLAGS) $(DBGFLAGS) $(TARGET_ARCH) -c -o $@ $<
 
 $(RELEASE)/%.o: %.cpp
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(RELFLAGS) $(TARGET_ARCH) -c -o $@ $<
+	$(CXX) $(CXXFLAGS) -fPIC $(CPPFLAGS) $(RELFLAGS) $(TARGET_ARCH) -c -o $@ $<
 
 ifeq ($(OLDGCC),N)
 
