--- configure.in.orig	2006-01-27 01:25:47.000000000 +0100
+++ configure.in	2006-01-31 23:30:57.000000000 +0100
@@ -1108,6 +1108,7 @@
 distro-configs/NovellWin32.conf
 distro-configs/NovellWin32ISO.conf
 distro-configs/Gentoo.conf
+distro-configs/GentooUnstable.conf
 distro-configs/Frugalware.conf
 distro-configs/openmamba.conf
 distro-configs/translate-org-za-linux.conf
--- distro-configs/Common.conf.in.orig	2006-06-15 20:03:35.000000000 +0200
+++ distro-configs/Common.conf.in	2006-06-15 20:05:11.000000000 +0200
@@ -1,5 +1,3 @@
---disable-binfilter
---disable-crashdump
 --with-lang=\"@OOO_LANGS@\"
 --with-use-shell=bash
 --without-gpc
--- distro-configs/CommonLinux.conf.in.orig	2006-11-04 23:51:23.000000000 +0100
+++ distro-configs/CommonLinux.conf.in	2006-11-04 23:51:43.000000000 +0100
@@ -1,16 +1,12 @@
 --disable-epm
---with-openldap
 --with-build-version=\"Build @OOO_BUILDVERSION@\"
 --without-fonts
 --with-system-freetype
 --with-system-gcc
 --with-system-jpeg
 --with-system-libxml
---with-system-mozilla
 --with-system-python
 --with-system-stdlibs
 --with-system-zlib
 --with-unix-wrapper=ooffice@OOO_BINSUFFIX@
---enable-evolution2
---enable-atkbridge
 --with-alloc=system
 --with-vba-package-format=builtin
 @CAIRO_CONFIGURE_FLAG@
--- patches/src680/apply.orig	2007-07-14 17:36:19.000000000 +0200
+++ patches/src680/apply	2007-07-14 17:37:32.000000000 +0200
@@ -2276,6 +2276,20 @@
 #only enable enhanced field work in Novell OOo by default
 officecfg-field-patch.diff, flr, n#248354 
 
+[ GentooExperimental ]
+SectionOwner => hmth
+
+# test jemalloc, FreeBSD 7 allocator
+jemalloc.diff
+# test system rhino
+system-rhino.diff
+
+# OOo 3.0 Chart2 features
+cws-chart19.diff
+cws-chart20.diff
+cws-chart20_sc.diff
+cws-chart22.diff
+cws-chart23.diff
 
 [ Fixes ]
 sc-xclimpchangetrack-discard-bogus-formula-size.diff, n#355304, tml
