--- Makefile.am.orig	2008-09-26 16:57:32.000000000 +0200
+++ Makefile.am	2008-09-26 16:58:25.000000000 +0200
@@ -2,7 +2,7 @@
 #  MAYBE_BONOBO = bonobo
 # endif
 
-SUBDIRS = po desktop distro-configs bin etc patches man doc src stamp templates www # $(MAYBE_BONOBO)
+SUBDIRS = po desktop distro-configs bin etc patches man src stamp templates # $(MAYBE_BONOBO)
 
 include $(top_srcdir)/Makefile.shared
 
--- configure.in.orig	2008-09-26 16:57:39.000000000 +0200
+++ configure.in	2008-09-26 17:00:03.000000000 +0200
@@ -1316,7 +1316,6 @@
 
 
 AC_CONFIG_FILES([download], [chmod +x download])
-AC_CONFIG_FILES([scratch/place], [chmod +x scratch/place])
 AC_CONFIG_FILES([intltool-extract intltool-merge intltool-update], [chmod +x intltool*])
 AC_OUTPUT([ 
 Makefile
@@ -1410,7 +1409,6 @@
 patches/64bit/Makefile
 po/Makefile
 po/sdf-templates/Makefile
-doc/Makefile
 src/Makefile
 src/icons/Makefile
 src/icons/tango/Makefile
@@ -1419,9 +1417,6 @@
 stamp/Makefile
 templates/Makefile
 man/Makefile
-www/Makefile
-www/images/Makefile
-www/tinder-scripts/Makefile
 ])
 
 if test "$ENABLE_CCACHE" = "yes"; then
