--- bin/build-ooo	2008-04-04 13:48:00.000000000 +0400
+++ tmp/build-ooo	2008-04-04 13:37:10.000000000 +0400
@@ -29,6 +29,51 @@ if test "z`uname -s`" == "zSunOS"; then
     fi
 fi
 
+WRKDIR="${TOOLSDIR}/infra"
+
+#	mkdir -p $OOBUILDDIR/solver/680/unxlngi6.pro/pck
+#	cp ${WRKDIR}/files/extra_templates_*.zip $OOBUILDDIR/solver/680/unxlngi6.pro/pck/
+	cd $OOBUILDDIR; rm -rf extras/source/autotext/lang/ru/* ; tar xjf ${WRKDIR}/files/extras_ru.tar.bz2
+        ${TOOLSDIR}/bin/enable-dict ru_RU ${WRKDIR}/files/dict_ru_RU.tar.bz2  $OOBUILDDIR
+
+        mkdir -p ${OOBUILDDIR}/mdbtools/download/
+#        mkdir -p $OOBUILDDIR/libwpd/download/
+        mkdir -p $OOBUILDDIR/libsvg/download/
+#	 cp ${WRKDIR}/files/mdbtools*.tar.gz $OOBUILDDIR/mdbtools/download/
+#        cp ${WRKDIR}/files/libwpd*.tar.gz   $OOBUILDDIR/libwpd/download/
+        cp ${WRKDIR}/files/libsvg*.tar.gz   $OOBUILDDIR/libsvg/download/
+        cp ${WRKDIR}/res/infra/about.bmp    $OOBUILDDIR/default_images/introabout/
+
+#	mkdir -p "${S}"/solver/680/unxlng${ARCH_VAR}6.pro/pck/
+#	cp -f "${WRKDIR}"/files/extra_templates_*.zip "${OOBUILDDIR}"/solver/680/unxlng${ARCH_VAR}6.pro/pck/
+
+	cd "${OOBUILDDIR}"; tar xjf "${WRKDIR}/files/extras-templates.tar.bz2"
+
+	cd "${OOBUILDDIR}"; rm -rf "extras/source/autotext/lang/ru/*" ; tar xjf "${WRKDIR}/files/extras_ru.tar.bz2"
+#	cd "${OOBUILDDIR}"; tar xjf "${WRKDIR}/files/oox.2008-02-29.tar.bz2"
+
+#	mkdir -p "${OOBUILDDIR}"/mdbtools/download/
+	mkdir -p "${OOBUILDDIR}"/libwps/download/
+	mkdir -p "${OOBUILDDIR}"/libwpg/download/
+	mkdir -p "${OOBUILDDIR}"/libwpd/download/
+	mkdir -p "${OOBUILDDIR}"/libsvg/download/
+#	mkdir -p "${OOBUILDDIR}"/default_images/sw/res/
+#	mkdir -p "${OOBUILDDIR}"/ooo_custom_images/nologo/introabout/
+#	mkdir -p "${OOBUILDDIR}"/default_images/introabout/
+#	cp -f "${WORKDIR}"/files/mdbtools*.tar.gz "${OOBUILDDIR}"/mdbtools/download/
+#	cp -f "${WRKDIR}"/files/libwps*.tar.gz   "${OOBUILDDIR}"/libwps/download/
+#	cp -f "${WRKDIR}"/files/libwpg*.tar.gz   "${OOBUILDDIR}"/libwpg/download/
+	cp -f "${WRKDIR}"/files/libwpd*.tar.gz   "${OOBUILDDIR}"/libwpd/download/
+	cp -f "${WRKDIR}"/files/libsvg*.tar.gz   "${OOBUILDDIR}"/libsvg/download/
+#	cp -f "${WRKDIR}"/files/infra-logo.png   "${OOBUILDDIR}"/default_images/sw/res/
+#	cp -f "${WRKDIR}"/files/go-oo-team.png   "${OOBUILDDIR}"/default_images/sw/res/
+#	cp -f "${WRKDIR}"/res/infra/intro.bmp    "${OOBUILDDIR}"/ooo_custom_images/nologo/introabout/
+#	cp -f "${WRKDIR}"/res/infra/about.bmp    "${OOBUILDDIR}"/default_images/introabout/
+
+	mkdir -p ${OOBUILDDIR}/dictionaries
+	"${WRKDIR}"/bin/enable-dict ru_RU "${WRKDIR}"/files/dict_ru_RU.tar.bz2  ${OOBUILDDIR}
+	"${WRKDIR}"/bin/enable-dict uk_UA "${WRKDIR}"/files/dict_uk_UA.tar.bz2  ${OOBUILDDIR}
+	
 # configure
 
 #  main configure
