--- bin/unpack.orig	2007-11-30 17:13:06.000000000 +0100
+++ bin/unpack	2007-11-30 17:49:58.000000000 +0100
@@ -98,29 +98,54 @@
 fi
 
 echo "Unpacking OO.o build tree - [ go and have some $DRINK ] ..."
-echo "Unpacking $OOO_TARBALL..."
-($OOO_DECOMPRESS_CMD $SRCDIR/$OOO_TARBALL | $GNUTAR xpf - ) || exit 1
+if test ! -e $OOBUILDDIR/.unpacked_core; then
+   echo "Unpacking $OOO_TARBALL..."
+   ($OOO_DECOMPRESS_CMD $SRCDIR/$OOO_TARBALL | $GNUTAR xpf - ) || exit 1
+   touch $OOBUILDDIR/.unpacked_core
+else
+   echo "$OOO_TARBALL already unpacked..."
+fi
 
 # -system source
 if test -f $SRCDIR/$OOO_SYSTEM_TARBALL; then
-    # System is optional, but parts of it are useful
-    echo "Unpacking $OOO_SYSTEM_TARBALL..."
-    ($OOO_DECOMPRESS_CMD $SRCDIR/$OOO_SYSTEM_TARBALL | $GNUTAR xpf - )
+ if test ! -e $OOBUILDDIR/.unpacked_system; then
+   # System is optional, but parts of it are useful
+   echo "Unpacking $OOO_SYSTEM_TARBALL..."
+   ($OOO_DECOMPRESS_CMD $SRCDIR/$OOO_SYSTEM_TARBALL | $GNUTAR xpf - ) || exit 1
+   touch $OOBUILDDIR/.unpacked_system
+ else
+   echo "$OOO_SYSTEM_TARBALL already unpacked..."
+ fi
 fi
 # -binfilter source
 if test -f $SRCDIR/$OOO_BINFILTER_TARBALL; then
+ if test ! -e $OOBUILDDIR/.unpacked_binfilter; then
    echo "Unpacking $OOO_BINFILTER_TARBALL..."
-   ($OOO_DECOMPRESS_CMD $SRCDIR/$OOO_BINFILTER_TARBALL | $GNUTAR xpf - )
+   ($OOO_DECOMPRESS_CMD $SRCDIR/$OOO_BINFILTER_TARBALL | $GNUTAR xpf - ) || exit 1
+   touch $OOBUILDDIR/.unpacked_binfilter
+ else
+   echo "$OOO_BINFILTER_TARBALL already unpacked..."
+ fi
 fi
 # -lang source
 if test -f $SRCDIR/$OOO_LANG_TARBALL; then
+ if test ! -e $OOBUILDDIR/.unpacked_lang; then
    echo "Unpacking $OOO_LANG_TARBALL..."
-    ($OOO_DECOMPRESS_CMD $SRCDIR/$OOO_LANG_TARBALL | $GNUTAR xpf - )
+   ($OOO_DECOMPRESS_CMD $SRCDIR/$OOO_LANG_TARBALL | $GNUTAR xpf - ) || exit 1
+   touch $OOBUILDDIR/.unpacked_lang
+ else
+   echo "$OOO_LANG_TARBALL already unpacked..."
+ fi
 fi
 # -sdk_oo source
 if test -f $SRCDIR/$OOO_SDK_OO_TARBALL; then
+ if test ! -e $OOBUILDDIR/.unpacked_sdk; then
    echo "Unpacking $OOO_SDK_OO_TARBALL"
-    ($OOO_DECOMPRESS_CMD $SRCDIR/$OOO_SDK_OO_TARBALL | $GNUTAR xpf - )
+   ($OOO_DECOMPRESS_CMD $SRCDIR/$OOO_SDK_OO_TARBALL | $GNUTAR xpf - ) || exit 1
+   touch $OOBUILDDIR/.unpacked_sdk
+ else
+   echo "$OOO_SDK_OO_TARBALL already unpacked..."
+ fi
 fi
 
 if test "$ENABLE_CCACHE" = "yes" ; then
@@ -454,41 +479,66 @@
 $GNUCP -a $SRCDIR/$LPSOLVE_PACKAGE $OOBUILDDIR/lpsolve/download/ || exit 1
 
 if test "$SYSTEM_LIBWPD" != "YES"; then
+ if test ! -e $OOBUILDDIR/.unpacked_wpd; then
   LIBWPD_PACKAGE=libwpd-0.8.14.tar.gz
   check_tarball $SRCDIR/$LIBWPD_PACKAGE
   echo "Copying libwpd package into the tree"
   mkdir -p $OOBUILDDIR/libwpd/download || exit 1
   $GNUCP -a $SRCDIR/$LIBWPD_PACKAGE $OOBUILDDIR/libwpd/download/ || exit 1
+  touch $OOBUILDDIR/.unpacked_wpd
+ else
+  echo "libwpd already copied"
+ fi
 fi
 
 if test "$SYSTEM_LIBWPS" != "YES"; then
+ if test ! -e $OOBUILDDIR/.unpacked_wps; then
   LIBWPS_PACKAGE=libwps-0.1.2.tar.gz
   check_tarball $SRCDIR/$LIBWPS_PACKAGE
   echo "Copying libwps package into the tree"
   mkdir -p $OOBUILDDIR/libwps/download || exit 1
   $GNUCP -a $SRCDIR/$LIBWPS_PACKAGE $OOBUILDDIR/libwps/download/ || exit 1
+  touch $OOBUILDDIR/.unpacked_wps
+ else
+  echo "libwps already copied"
+ fi
 fi
 
 if test "$SYSTEM_LIBWPG" != "YES"; then
+ if test ! -e $OOBUILDDIR/.unpacked_wpg; then
   LIBWPG_PACKAGE=libwpg-0.1.2.tar.gz
   check_tarball $SRCDIR/$LIBWPG_PACKAGE
   echo "Copying libwpg package into the tree"
   mkdir -p $OOBUILDDIR/libwpg/download || exit 1
   $GNUCP -a $SRCDIR/$LIBWPG_PACKAGE $OOBUILDDIR/libwpg/download/ || exit 1
+  touch $OOBUILDDIR/.unpacked_wpg
+ else
+  echo "libwpg already copied"
+ fi
 fi
 
 # oox source
+if test ! -e $OOBUILDDIR/.unpacked_oox; then
 check_tarball $SRCDIR/$OOX_MODULE_TARBALL
 echo "Unpacking $OOX_MODULE_TARBALL"
 cd $OOBUILDDIR
 ($OOO_DECOMPRESS_CMD $SRCDIR/$OOX_MODULE_TARBALL | $GNUTAR xpf - ) || exit 1
+touch $OOBUILDDIR/.unpacked_oox
+else
+  echo "oox already copied"
+fi
 
 # writerfilter source
 if test -f $SRCDIR/$WRITERFILTER_MODULE_TARBALL; then
+ if test ! -e $OOBUILDDIR/.unpacked_writerfilter; then
     echo "Unpacking $WRITERFILTER_MODULE_TARBALL"
     cd $OOBUILDDIR
     rm -rf writerfilter
     ($OOO_DECOMPRESS_CMD $SRCDIR/$WRITERFILTER_MODULE_TARBALL | $GNUTAR xpf - ) || exit 1
+    touch $OOBUILDDIR/.unpacked_writerfilter
+ else
+  echo "writerfilter already copied"
+ fi
 else
     echo "The writerfilter module is missing - please run the download script."
     exit 1
