--- kmail-4.1.0/kmail/maildirsettings.ui	2008/07/31 22:06:20	840345
+++ kmail-4.1.0/kmail/maildirsettings.ui	2008/07/31 22:07:09	840346
@@ -12,7 +12,7 @@
   <property name="windowTitle" >
    <string>Form</string>
   </property>
-  <layout class="QGridLayout" name="gridLayout" >
+  <layout class="QGridLayout" >
    <item row="0" column="0" colspan="2" >
     <widget class="QLabel" name="titleLabel" >
      <property name="font" >
@@ -53,7 +53,11 @@
     </widget>
    </item>
    <item row="3" column="1" >
-    <widget class="QComboBox" name="locationEdit" />
+    <widget class="QComboBox" name="locationEdit" >
+     <property name="editable" >
+      <bool>true</bool>
+     </property>
+    </widget>
    </item>
    <item row="3" column="2" >
     <widget class="QPushButton" name="choose" >
@@ -87,8 +91,8 @@
     </widget>
    </item>
    <item row="6" column="1" >
-    <widget class="KIntNumInput" name="intervalSpin" >
-     <property name="suffix" >
+    <widget class="KIntNumInput" native="1" name="intervalSpin" >
+     <property name="suffix" stdset="0" >
       <string> min</string>
      </property>
     </widget>
@@ -120,11 +124,11 @@
     <widget class="KLineEdit" name="precommand" />
    </item>
    <item row="9" column="0" >
-    <spacer name="verticalSpacer" >
+    <spacer>
      <property name="orientation" >
       <enum>Qt::Vertical</enum>
      </property>
-     <property name="sizeHint" stdset="0" >
+     <property name="sizeHint" >
       <size>
        <width>20</width>
        <height>0</height>
@@ -149,6 +153,7 @@
    <class>KSeparator</class>
    <extends>QFrame</extends>
    <header>kseparator.h</header>
+   <container>1</container>
   </customwidget>
  </customwidgets>
  <resources/>
