diff --git a/CHANGELOG b/CHANGELOG
index b9a1e87..144c0c8 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -3,6 +3,7 @@
 - lookup_init cleanup and fix missed memory leak.
 - use nis map order to check if update is needed.
 - fix couple of memory leaks in lookup_yp.c.
+- fix pasre error in replicated server module.
 
 20/6/2006 autofs-5.0.0_beta5
 ---------------------------
diff --git a/modules/replicated.c b/modules/replicated.c
index e7f378b..2d7e94a 100644
--- a/modules/replicated.c
+++ b/modules/replicated.c
@@ -824,7 +824,7 @@ int parse_location(struct host **hosts, 
 				/* No spaces in host names at least */
 				if (*next == ':') {
 					while (*next &&
-					      (*next != ' ' || *next != '\t'))
+					      (*next != ' ' && *next != '\t'))
 						next--;
 					*next++ = '\0';
 				}
