diff --git a/CHANGELOG b/CHANGELOG
index 144c0c8..7c193e8 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -4,6 +4,7 @@
 - use nis map order to check if update is needed.
 - fix couple of memory leaks in lookup_yp.c.
 - fix pasre error in replicated server module.
+- correct spelling error in default config.
 
 20/6/2006 autofs-5.0.0_beta5
 ---------------------------
diff --git a/samples/autofs.conf.default.in b/samples/autofs.conf.default.in
index 9c4b1ca..eca1392 100644
--- a/samples/autofs.conf.default.in
+++ b/samples/autofs.conf.default.in
@@ -21,7 +21,7 @@ # Define the default LDAP schema to use 
 #
 # System default
 #
-#DEFAULT_MAP_OBJECT_CALSS="nisMap"
+#DEFAULT_MAP_OBJECT_CLASS="nisMap"
 #DEFAULT_ENTRY_OBJECT_CLASS="nisObject"
 #DEFAULT_MAP_ATTRIBUTE="nisMapName"
 #DEFAULT_ENTRY_ATTRIBUTE="cn"
@@ -29,13 +29,13 @@ #DEFAULT_VALUE_ATTRIBUTE="nisMapEntry"
 #
 # Other common LDAP nameing
 #
-#DEFAULT_MAP_OBJECT_CALSS="automountMap"
+#DEFAULT_MAP_OBJECT_CLASS="automountMap"
 #DEFAULT_ENTRY_OBJECT_CLASS="automount"
 #DEFAULT_MAP_ATTRIBUTE="ou"
 #DEFAULT_ENTRY_ATTRIBUTE="cn"
 #DEFAULT_VALUE_ATTRIBUTE="automountInformation"
 #
-#DEFAULT_MAP_OBJECT_CALSS="automountMap"
+#DEFAULT_MAP_OBJECT_CLASS="automountMap"
 #DEFAULT_ENTRY_OBJECT_CLASS="automount"
 #DEFAULT_MAP_ATTRIBUTE="automountMapName"
 #DEFAULT_ENTRY_ATTRIBUTE="automountKey"
