diff --git a/CHANGELOG b/CHANGELOG
index 13d0679..77fde7e 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -8,6 +8,7 @@
 - fix default auth config not being installed.
 - change LDAP query method as my test db was incorrect.
 - change ldap defaults code to handle missing auth config.
+- fix mistake in parsing old style LDAP specs.
 
 20/6/2006 autofs-5.0.0_beta5
 ---------------------------
diff --git a/lib/master_parse.y b/lib/master_parse.y
index 54543af..4def672 100644
--- a/lib/master_parse.y
+++ b/lib/master_parse.y
@@ -514,6 +514,7 @@ int master_parse_entry(const char *buffe
 	source = master_add_map_source(entry, type, format, age, 
 					local_argc, (const char **) local_argv);
 	if (!source) {
+		error(LOGOPT_ANY, "failed to add source");
 		if (new)
 			master_free_mapent(new);
 		return 0;
diff --git a/lib/master_tok.l b/lib/master_tok.l
index 1f8a7ea..2abdfc3 100644
--- a/lib/master_tok.l
+++ b/lib/master_tok.l
@@ -80,7 +80,9 @@ OPTIONSTR	([\-]?([[:alpha:]_]([[:alnum:]
 MACROSTR	(-D{OPTWS}([[:alpha:]_]([[:alnum:]_\-])*)=([[:alnum:]_\-])+)
 NUMBER		[0-9]+
 
-DNSERVERSTR	(\/\/[[:alpha:]][[:alnum:]\-.]*(:[0-9])?\/)
+DNSERVSTR1	([[:alpha:]][[:alnum:]\-.]*(:[0-9]+)?:)
+DNSERVSTR2	(\/\/[[:alpha:]][[:alnum:]\-.]*(:[0-9]+)?\/)
+DNSERVERSTR	{DNSERVSTR1}|{DNSERVSTR2}
 DNATTRSTR	(ou|cn|nisMapName|automountMapName|dc)
 DNNAMESTR	([[:alnum:]_.\-]+)
 
@@ -198,6 +200,11 @@ MTYPE           ((file|program|yp|nis|ni
 		yyless(0);
 	}
 
+	{DNSERVERSTR} {
+		BEGIN(DNSTR);
+		yyless(0);
+	}
+
 	{DNATTRSTR}= {
 		BEGIN(DNSTR);
 		yyless(0);
diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index 45d755d..9794c5f 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -510,7 +510,7 @@ #endif
 
 /*
  *  Take an input string as specified in the master map, and break it
- *  down into a basedn, servername, and port.
+ *  down into a server name and basedn.
  */
 static int parse_server_string(const char *url, struct lookup_context *ctxt)
 {
@@ -528,11 +528,7 @@ static int parse_server_string(const cha
 		const char *s = ptr + 2;
 		const char *q = NULL;
 
-		/*
-		 * Isolate the server(s).  The ':' breaks the SUN
-		 * parser for submounts, so we can't actually
-		 * use it.
-		 */
+		/* Isolate the server(s). */
 		if ((q = strchr(s, '/'))) {
 			l = q - s;
 			tmp = malloc(l + 1);
@@ -547,6 +543,10 @@ static int parse_server_string(const cha
 			memcpy(ctxt->server, s, l);
 			ptr = q + 1;
 		} else {
+			crit(LOGOPT_ANY,
+			     MODPREFIX "invalid LDAP map syntax %s", ptr);
+			return 0;
+/* TODO: why did I put this here, the parser shouldn't let this by
 			l = strlen(ptr);
 			tmp = malloc(l + 1);
 			if (!tmp) {
@@ -558,9 +558,24 @@ static int parse_server_string(const cha
 			ctxt->server = tmp;
 			memset(ctxt->server, 0, l + 1);
 			memcpy(ctxt->server, s, l);
+*/
 		}
 	} else if (strchr(ptr, ':') != NULL) {
-		l = strchr(ptr, ':') - ptr;
+		char *q = NULL;
+
+		/* Isolate the server(s). Include the port spec */
+		q = strchr(ptr, ':');
+		if (isdigit(*q))
+			while (isdigit(*q))
+				q++;
+
+		if (*q != ':') {
+			crit(LOGOPT_ANY,
+			     MODPREFIX "invalid LDAP map syntax %s", ptr);
+			return 0;
+		}
+
+		l = q - ptr;
 		/* Isolate the server's name. */
 		tmp = malloc(l + 1);
 		if (!tmp) {
@@ -575,6 +590,7 @@ static int parse_server_string(const cha
 		ptr += l + 1;
 	}
 
+	/* TODO: why did I do this - how can the map name "and" base dn be missing? */
 	if (!ptr)
 		goto done;
 
