diff --git a/daemon/automount.c b/daemon/automount.c
index aeece1b..38940c7 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1637,15 +1637,9 @@ #endif
 
 	become_daemon(foreground);
 
-	if (argc == 0) {
-		const char *name;
-
-		name = defaults_get_master_map();
-		if (!name)
-			master = master_new(NULL, timeout, ghost);
-		else
-			master = master_new(name, timeout, ghost);
-	} else
+	if (argc == 0)
+		master = master_new(NULL, timeout, ghost);
+	else
 		master = master_new(argv[0], timeout, ghost);
 
 	if (!master) {
diff --git a/daemon/lookup.c b/daemon/lookup.c
index 4db7cef..dda27f8 100644
--- a/daemon/lookup.c
+++ b/daemon/lookup.c
@@ -189,6 +189,8 @@ int lookup_nss_read_master(struct master
 
 	result = nsswitch_parse(&nsslist);
 	if (result) {
+		if (!list_empty(&nsslist))
+			free_sources(&nsslist);
 		error(LOGOPT_ANY, "can't to read name service switch config.");
 		return 0;
 	}
diff --git a/daemon/spawn.c b/daemon/spawn.c
index 8195820..b3fd9de 100644
--- a/daemon/spawn.c
+++ b/daemon/spawn.c
@@ -229,12 +229,14 @@ void reset_signals(void)
 	sa.sa_flags = 0;
 
 	for (i = 1; i < NSIG; i++)
-		sigaction(i, &sa, NULL);
+		if (i != SIGKILL && i != SIGSTOP)
+			sigaction(i, &sa, NULL);
 
 	sa.sa_handler = SIG_DFL;
 
 	for (i = 1; i < NSIG; i++)
-		sigaction(i, &sa, NULL);
+		if (i != SIGKILL && i != SIGSTOP)
+			sigaction(i, &sa, NULL);
 
 	/* Ignore the user signals that may be sent so that we
 	 *  don't terminate execed program by mistake */
