diff --git a/CHANGELOG b/CHANGELOG
index a1143ad..56873a7 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -12,6 +12,7 @@
 - update LDAP so that new query methos also works for old
   syntax.
 - allow global macro defines to override system macros.
+- cleanup defaults_read_config.
 
 20/6/2006 autofs-5.0.0_beta5
 ---------------------------
diff --git a/lib/defaults.c b/lib/defaults.c
index 0170f9b..594afb3 100644
--- a/lib/defaults.c
+++ b/lib/defaults.c
@@ -173,9 +173,19 @@ unsigned int defaults_read_config(void)
 			ret = setenv(res, value, 0);
 			if (ret)
 				fprintf(stderr,
-				        "can't set config value for %s", res);
+				        "can't set config value for %s, "
+					"error %d", res, ret);
 		}
 	}
+
+	if (!feof(f)) {
+		fprintf(stderr, "fgets returned error %d while reading %s\n",
+			ferror(f), DEFAULTS_CONFIG_FILE);
+		fclose(f);
+		return 0;
+	}
+
+	fclose(f);
 	return 1;
 }
 
