diff --git a/CHANGELOG b/CHANGELOG
index 56873a7..40fb783 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -13,6 +13,7 @@
   syntax.
 - allow global macro defines to override system macros.
 - cleanup defaults_read_config.
+- add rfc2307-bis example map configurations.
 
 20/6/2006 autofs-5.0.0_beta5
 ---------------------------
diff --git a/samples/ldap-automount-rfc2307-bis-auto.direct b/samples/ldap-automount-rfc2307-bis-auto.direct
new file mode 100644
index 0000000..e2d72c9
--- /dev/null
+++ b/samples/ldap-automount-rfc2307-bis-auto.direct
@@ -0,0 +1,48 @@
+#
+# ident $Id: ldap-automount-auto.direct,v 1.2 2003/12/10 14:54:33 raven Exp $
+#
+dn: automountMapName=auto_direct,dc=themaw,dc=net
+objectClass: top
+objectClass: automountMap
+automountMapName: auto_direct
+
+dn: automountKey=/nfs/budgie/man,automountMapName=auto_direct,dc=themaw,dc=net
+objectClass: automount
+automountKey: /nfs/budgie/man
+automountInformation: 	budgie:/usr/local/man
+
+dn: automountKey=/nfs/budgie/bin,automountMapName=auto_direct,dc=themaw,dc=net
+objectClass: automount
+automountKey: /nfs/budgie/bin
+automountInformation: 	budgie:/local/data/bin
+
+dn: automountKey=/nfs/budgie/etc,automountMapName=auto_direct,dc=themaw,dc=net
+objectClass: automount
+automountKey: /nfs/budgie/etc
+automountInformation: 	budgie:/usr/local/etc
+
+dn: automountKey=/nfs/sbin,automountMapName=auto_direct,dc=themaw,dc=net
+objectClass: automount
+automountKey: /nfs/sbin
+automountInformation:	budgie:/usr/local/sbin
+
+dn: automountKey=/tst/budgie/man,automountMapName=auto_direct,dc=themaw,dc=net
+objectClass: automount
+automountKey: /tst/budgie/man
+automountInformation:	budgie:/usr/local/man
+
+dn: automountKey=/nfs/budgie/sbin,automountMapName=auto_direct,dc=themaw,dc=net
+objectClass: automount
+automountKey: /nfs/budgie/sbin
+automountInformation:	budgie:/usr/local/sbin
+
+dn: automountKey=/nfs/budgie/test,automountMapName=auto_direct,dc=themaw,dc=net
+objectClass: automount
+automountKey: /nfs/budgie/test
+automountInformation:	budgie:/usr/local/test
+
+dn: automountKey=/tst/budgie/etc,automountMapName=auto_direct,dc=themaw,dc=net
+objectClass: automount
+automountKey: /tst/budgie/etc
+automountInformation:	budgie:/usr/local/etc
+
diff --git a/samples/ldap-automount-rfc2307-bis-auto.indirect b/samples/ldap-automount-rfc2307-bis-auto.indirect
new file mode 100644
index 0000000..9d43915
--- /dev/null
+++ b/samples/ldap-automount-rfc2307-bis-auto.indirect
@@ -0,0 +1,30 @@
+#
+dn: automountMapName=auto_indirect,dc=themaw,dc=net
+objectClass: top
+objectClass: automountMap
+automountMapName: auto_indirect
+
+dn: automountKey=bin,automountMapName=auto_indirect,dc=themaw,dc=net
+objectClass: top
+objectClass: automount
+automountKey: bin
+automountInformation: 	budgie:/usr/local/bin
+
+dn: automountKey=etc,automountMapName=auto_indirect,dc=themaw,dc=net
+objectClass: top
+objectClass: automount
+automountKey: etc
+automountInformation: 	budgie:/usr/local/etc
+
+dn: automountKey=lib,automountMapName=auto_indirect,dc=themaw,dc=net
+objectClass: top
+objectClass: automount
+automountKey: lib
+automountInformation: 	budgie:/usr/local/lib
+
+dn: automountKey=/,automountMapName=auto_indirect,dc=themaw,dc=net
+objectClass: top
+objectClass: automount
+automountKey: /
+automountInformation: 	budgie:/usr/local/&
+
diff --git a/samples/ldap-automount-rfc2307-bis-auto.master b/samples/ldap-automount-rfc2307-bis-auto.master
new file mode 100644
index 0000000..b7b6237
--- /dev/null
+++ b/samples/ldap-automount-rfc2307-bis-auto.master
@@ -0,0 +1,19 @@
+dn: automountMapName=auto_master,dc=themaw,dc=net
+objectClass: top
+objectClass: automountMap
+automountMapName: auto_master
+
+dn: description=/ldap, automountMapName=auto_master,dc=themaw,dc=net
+objectClass: top
+objectClass: automount
+automountKey: /ldap
+automountInformation: auto_indirect
+description: /ldap
+
+dn: description=/-,automountMapName=auto_master,dc=themaw,dc=net
+objectClass: top
+objectClass: automount
+automountKey: /-
+automountInformation: auto_direct
+description: /-
+
diff --git a/samples/ldap-automount-rfc2307-bis-old-style-auto.master b/samples/ldap-automount-rfc2307-bis-old-style-auto.master
new file mode 100644
index 0000000..0880ff0
--- /dev/null
+++ b/samples/ldap-automount-rfc2307-bis-old-style-auto.master
@@ -0,0 +1,19 @@
+dn: automountMapName=auto_master,dc=themaw,dc=net
+objectClass: top
+objectClass: automountMap
+automountMapName: auto_master
+
+dn: description=/ldap, automountMapName=auto_master,dc=themaw,dc=net
+objectClass: top
+objectClass: automount
+automountKey: /ldap
+automountInformation: ldap:budgie:automountMapName=auto_indirect,dc=themaw,dc=net
+description: /ldap
+
+dn: description=/-,automountMapName=auto_master,dc=themaw,dc=net
+objectClass: top
+objectClass: automount
+automountKey: /-
+automountInformation: ldap:budgie:automountMap=auto_direct,dc=themaw,dc=net
+description: /-
+
