diff -urN vhba-module-1.1.0.orig/Makefile vhba-module-1.1.0/Makefile
--- vhba-module-1.1.0.orig/Makefile	2008-06-28 12:08:33.000000000 +0200
+++ vhba-module-1.1.0/Makefile	2008-07-16 15:33:07.000000000 +0200
@@ -12,7 +12,7 @@
 KDIR := /lib/modules/$(KERNELRELEASE)/build
 KMAKE := $(MAKE) -C $(KDIR) M=$(PWD)
 
-all: kernel.api.h modules
+all: modules
 
 kernel.api.h: kat/*.c
 	kat/kat ${KDIR} $@ $^
diff -urN vhba-module-1.1.0.orig/vhba.c vhba-module-1.1.0/vhba.c
--- vhba-module-1.1.0.orig/vhba.c	2008-06-28 12:08:33.000000000 +0200
+++ vhba-module-1.1.0/vhba.c	2008-07-16 15:37:45.000000000 +0200
@@ -18,6 +18,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
  */
 
+#include <linux/version.h>
 #include <linux/init.h>
 #include <linux/module.h>
 #include <linux/highmem.h>
@@ -32,7 +33,14 @@
 #include <scsi/scsi_cmnd.h>
 #include <scsi/scsi_device.h>
 
-#include <kernel.api.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24)
+#include <linux/scatterlist.h>
+#define KAT_SCATTERLIST_HAS_PAGE
+#endif
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,23)
+#define KAT_HAVE_SCSI_MACROS
+#endif
 
 MODULE_AUTHOR("Chia-I Wu");
 MODULE_VERSION(VHBA_VERSION);
