--- qtiplot/src/core/ApplicationWindow.h.old	2015-01-12 15:43:56.159063506 +0100
+++ qtiplot/src/core/ApplicationWindow.h	2015-01-12 15:44:28.337584796 +0100
@@ -72,7 +72,7 @@
 class QShortcut;
 class QMenu;
 class QToolBar;
-class QAssistantClient;
+//class QAssistantClient;
 class QLocale;
 class QMdiArea;
 class QUndoView;
@@ -1451,7 +1451,7 @@
 	Graph *lastCopiedLayer;
 	QSplitter *explorerSplitter;
 
-	QAssistantClient *assistant;
+//	QAssistantClient *assistant;
 	ScriptWindow *scriptWindow;
 	QTranslator *appTranslator, *qtTranslator;
 	QDockWidget *explorerWindow, *undoStackWindow;
--- qtiplot/src/core/ApplicationWindow.cpp.old	2015-01-12 15:44:00.773138239 +0100
+++ qtiplot/src/core/ApplicationWindow.cpp	2015-01-12 15:45:03.775159156 +0100
@@ -174,7 +174,7 @@
 #include <QVarLengthArray>
 #include <QList>
 #include <QUrl>
-#include <QAssistantClient>
+//#include <QAssistantClient>
 #include <QFontComboBox>
 #include <QSpinBox>
 #include <QMdiArea>
@@ -349,7 +349,7 @@
 	insertTranslatedStrings();
 	disableToolbars();
 
-	assistant = new QAssistantClient( QString(), this );
+//	assistant = new QAssistantClient( QString(), this );
 
 	connect(tablesDepend, SIGNAL(activated(int)), this, SLOT(showTable(int)));
 
@@ -11327,17 +11327,17 @@
 	if (!QFile(profilePath).exists())
 	{
 		QMessageBox::critical(0, tr("QtiPlot - Help Profile Not Found!"),
-				tr("The assistant could not start because the file <b>%1</b> was not found in the help file directory!").arg("qtiplot.adp")+"<br>"+
+//				tr("The assistant could not start because the file <b>%1</b> was not found in the help file directory!").arg("qtiplot.adp")+"<br>"+
 				tr("This file is provided with the QtiPlot manual which can be downloaded from the following internet address:")+
 				"<p><a href = http://soft.proindependent.com/manuals.html>http://soft.proindependent.com/manuals.html</a></p>");
 		exit(0);
 	}
 
 	QStringList cmdLst = QStringList() << "-profile" << profilePath;
-	QAssistantClient *assist = new QAssistantClient( QString(), 0);
-	assist->setArguments( cmdLst );
-	assist->showPage(helpPath);
-	connect(assist, SIGNAL(assistantClosed()), qApp, SLOT(quit()) );
+//	QAssistantClient *assist = new QAssistantClient( QString(), 0);
+//	assist->setArguments( cmdLst );
+//	assist->showPage(helpPath);
+//	connect(assist, SIGNAL(assistantClosed()), qApp, SLOT(quit()) );
 }
 
 void ApplicationWindow::showHelp()
@@ -11361,15 +11361,15 @@
 	if (!QFile(profilePath).exists())
 	{
 		QMessageBox::critical(this,tr("QtiPlot - Help Profile Not Found!"),
-				tr("The assistant could not start because the file <b>%1</b> was not found in the help file directory!").arg("qtiplot.adp")+"<br>"+
+//				tr("The assistant could not start because the file <b>%1</b> was not found in the help file directory!").arg("qtiplot.adp")+"<br>"+
 				tr("This file is provided with the QtiPlot manual which can be downloaded from the following internet address:")+
 				"<p><a href = http://soft.proindependent.com/manuals.html>http://soft.proindependent.com/manuals.html</a></p>");
 		return;
 	}
 
 	QStringList cmdLst = QStringList() << "-profile" << profilePath;
-	assistant->setArguments( cmdLst );
-	assistant->showPage(helpFilePath);
+//	assistant->setArguments( cmdLst );
+//	assistant->showPage(helpFilePath);
 }
 
 void ApplicationWindow::showPlotWizard()
