diff -U3 -r ./jeuclid-core/src/main/java/net/sourceforge/jeuclid/converter/ConverterRegistry.java ./jeuclid-core_my/src/main/java/net/sourceforge/jeuclid/converter/ConverterRegistry.java
--- ./jeuclid-core/src/main/java/net/sourceforge/jeuclid/converter/ConverterRegistry.java	2008-06-03 12:02:00.000000000 +0200
+++ ./jeuclid-core_my/src/main/java/net/sourceforge/jeuclid/converter/ConverterRegistry.java	2008-06-03 14:27:38.000000000 +0200
@@ -47,7 +47,6 @@
     private ConverterRegistry() {
         ImageIODetector.detectConversionPlugins(this);
         BatikDetector.detectConversionPlugins(this);
-        FreeHepDetector.detectConversionPlugins(this);
     }
 
     /**
Only in ./jeuclid-core/src/main/java/net/sourceforge/jeuclid/converter: FreeHepConverter.java
Only in ./jeuclid-core/src/main/java/net/sourceforge/jeuclid/converter: FreeHepDetector.java
Only in ./jeuclid-core/src/main/java/net/sourceforge/jeuclid/converter: FreeHepInternalDetector.java
