diff -Naur root/io/io/inc/TStreamerInfo.h root/io/io/inc/TStreamerInfo.h
--- root/io/io/inc/TStreamerInfo.h	2009-04-23 14:13:56.000000000 +0800
+++ root/io/io/inc/TStreamerInfo.h	2009-06-13 07:02:00.513711721 +0800
@@ -27,11 +27,11 @@
 
 #include "TVirtualCollectionProxy.h"
 
-#if (defined(_MSC_VER) && (_MSC_VER < 1300)) || defined(R__ALPHA) || \
-    (defined(R__MACOSX) && defined(R__GNU) && __GNUC__==3 && __GNUC_MINOR<=3) || \
-    (defined(R__MACOSX) && defined(__xlC__))
+// #if (defined(_MSC_VER) && (_MSC_VER < 1300)) || defined(R__ALPHA) || \
+//    (defined(R__MACOSX) && defined(R__GNU) && __GNUC__==3 && __GNUC_MINOR<=3) || \
+//    (defined(R__MACOSX) && defined(__xlC__))
 #define R__BROKEN_FUNCTION_TEMPLATES
-#endif
+// #endif
 
 class TFile;
 class TClass;
diff -Naur root/net/xrootd/src/xrootd/src/XrdBwm/XrdBwmHandle.cc root/net/xrootd/src/xrootd/src/XrdBwm/XrdBwmHandle.cc
--- root/net/xrootd/src/xrootd/src/XrdBwm/XrdBwmHandle.cc	2009-04-23 14:13:47.000000000 +0800
+++ root/net/xrootd/src/xrootd/src/XrdBwm/XrdBwmHandle.cc	2009-05-24 01:52:41.774722196 +0800
@@ -13,6 +13,7 @@
 const char *XrdBwmHandleCVSID = "$Id: XrdBwmHandle.cc 24468 2008-06-22 16:47:03Z ganis $";
 
 #include <string.h>
+#include <cstdio>
 
 #include "XrdBwm/XrdBwmHandle.hh"
 #include "XrdBwm/XrdBwmLogger.hh"
diff -Naur root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsAdmin.cc root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsAdmin.cc
--- root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsAdmin.cc	2009-04-23 14:13:47.000000000 +0800
+++ root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsAdmin.cc	2009-05-24 00:40:44.279472575 +0800
@@ -17,6 +17,7 @@
 #include <limits.h>
 #include <unistd.h>
 #include <sys/types.h>
+#include <cstdio>
 
 #include "XProtocol/YProtocol.hh"
 
diff -Naur root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsCache.cc root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsCache.cc
--- root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsCache.cc	2009-04-23 14:13:47.000000000 +0800
+++ root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsCache.cc	2009-05-24 00:48:10.877472271 +0800
@@ -15,6 +15,7 @@
 const char *XrdCmsCacheCVSID = "$Id: XrdCmsCache.cc 24468 2008-06-22 16:47:03Z ganis $";
   
 #include <sys/types.h>
+#include <cstdio>
 
 #include "XrdCms/XrdCmsCache.hh"
 #include "XrdCms/XrdCmsRRQ.hh"
diff -Naur root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsParser.cc root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsParser.cc
--- root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsParser.cc	2009-04-23 14:13:47.000000000 +0800
+++ root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsParser.cc	2009-05-24 01:12:12.132755989 +0800
@@ -20,6 +20,7 @@
 #include <string.h>
 #include <netinet/in.h>
 #include <sys/types.h>
+#include <cstdio>
 
 #include "XrdCms/XrdCmsParser.hh"
 #include "XrdCms/XrdCmsRRData.hh"
diff -Naur root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsReq.cc root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsReq.cc
--- root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsReq.cc	2009-04-23 14:13:47.000000000 +0800
+++ root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsReq.cc	2009-05-24 01:04:26.612472961 +0800
@@ -184,7 +184,8 @@
 void XrdCmsReq::Reply_Redirect(const char *sname, 
                                const char *lcgi, const char *ocgi)
 {
-   char hbuff[256], *colon;
+   const char *colon;
+   char hbuff[256];
    const char *hP = hbuff;
    int hlen, Port;
 
diff -Naur root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsResp.cc root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsResp.cc
--- root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsResp.cc	2009-04-23 14:13:47.000000000 +0800
+++ root/net/xrootd/src/xrootd/src/XrdCms/XrdCmsResp.cc	2009-05-24 01:33:33.624472107 +0800
@@ -16,6 +16,7 @@
 
 #include <stdlib.h>
 #include <string.h>
+#include <cstdio>
   
 #include "XrdCms/XrdCmsClientMsg.hh"
 #include "XrdCms/XrdCmsParser.hh"
diff -Naur root/net/xrootd/src/xrootd/src/XrdNet/XrdNetDNS.cc root/net/xrootd/src/xrootd/src/XrdNet/XrdNetDNS.cc
--- root/net/xrootd/src/xrootd/src/XrdNet/XrdNetDNS.cc	2009-04-23 14:13:47.000000000 +0800
+++ root/net/xrootd/src/xrootd/src/XrdNet/XrdNetDNS.cc	2009-05-24 00:56:52.173472254 +0800
@@ -471,7 +471,8 @@
 int XrdNetDNS::Host2Dest(const char      *hostname,
                          struct sockaddr &DestAddr,
                          char           **errtxt)
-{ char *cp, hbuff[256];
+{ const char *cp; 
+  char hbuff[256];
   int port, i;
   struct sockaddr_in InetAddr;
 
diff -Naur root/net/xrootd/src/xrootd/src/XrdXrootd/XrdXrootdPrepare.cc root/net/xrootd/src/xrootd/src/XrdXrootd/XrdXrootdPrepare.cc
--- root/net/xrootd/src/xrootd/src/XrdXrootd/XrdXrootdPrepare.cc	2009-04-23 14:13:46.000000000 +0800
+++ root/net/xrootd/src/xrootd/src/XrdXrootd/XrdXrootdPrepare.cc	2009-05-24 02:11:11.212471735 +0800
@@ -74,7 +74,8 @@
   
 int XrdXrootdPrepare::List(XrdXrootdPrepArgs &pargs, char *resp, int resplen)
 {
-   char *up, path[2048];
+   char *up; 
+   char path[2048];
    struct dirent *dp;
    struct stat buf;
    int rc;
@@ -259,7 +260,8 @@
 {
    DIR *prepD;
    time_t stale = time(0) - scrubkeep;
-   char *up, path[2048], *fn = path+LogDirLen;
+   const char *up; 
+   char path[2048], *fn = path+LogDirLen;
    struct dirent *dp;
    struct stat buf;
 
