diff -Naur root.orig/montecarlo/g4root/Module.mk root/montecarlo/g4root/Module.mk
--- root.orig/montecarlo/g4root/Module.mk	2009-08-09 23:24:43.568044521 +0400
+++ root/montecarlo/g4root/Module.mk	2009-08-09 23:26:11.918036344 +0400
@@ -58,7 +58,7 @@
 
 $(G4ROOTDS1):   $(G4ROOTH1) $(G4ROOTL1) $(ROOTCINTTMPDEP)
 		@echo "Generating dictionary $@..."
-		$(ROOTCINTTMP) -f $@ -c $(CXXFLAGS) -I$(G4ROOTINCDIR) \
+		$(ROOTCINTTMP) -f $@ -c -Iinclude -I$(G4ROOTINCDIR) \
 		   $(G4ROOTH2) $(G4ROOTL1)
 
 $(G4ROOTMAP):   $(RLIBMAP) $(MAKEFILEDEP) $(G4ROOTL)
