diff -Naur root.orig/configure root/configure
--- root.orig/configure	2009-08-09 23:22:15.561395174 +0400
+++ root/configure	2009-08-09 23:23:48.018050705 +0400
@@ -2121,8 +2121,8 @@
 if test ! "x$enable_pgsql" = "xno"; then
     # Check for PgSQL include and library
     check_header "libpq-fe.h" "$pgsqlincdir" $PGSQL $PGSQL/include \
-	/usr/include /usr/local/include /usr/local/pgsql/include \
-        /usr/include/pgsql /usr/include/postgresql \
+	/usr/include/postgresql /usr/include /usr/local/include \
+	/usr/local/pgsql/include /usr/include/pgsql \
 	/usr/include/postgresql/8.0 /usr/pgsql/include \
         /usr/local/include/pgsql /usr/local/include/postgresql \
 	/usr/include/postgresql/8.0 /usr/pgsql /usr/local/pgsql \
@@ -2204,7 +2204,7 @@
     else
         check_header "sqlext.h" "$odbcincdir" \
             $ODBC $ODBC/include $ODBCINCDIR /usr/local/include \
-            /usr/include $finkdir/include
+            /usr/include $finkdir/include /usr/include/iodbc
         odbcinc=$found_hdr
         odbcincdir=$found_dir
 
@@ -2304,7 +2304,7 @@
 	    check_library "libQtCore QtCore QtCore$qt4suffix QtCore4" \
 		"$enable_shared" \
 		"$qtlibdir" $QTDIR $QTDIR/lib /usr/local/lib \
-		/usr/local/qt/lib /usr/lib /usr/qt/lib
+		/usr/local/qt/lib /usr/lib /usr/qt/lib /usr/lib/qt4
 	    qtlib=$found_lib
 	    qtlibdir=$found_dir
 	    check_library "libQtGui QtGui QtGui$qt4suffix QtGui4" \
@@ -2588,14 +2588,14 @@
         G4SYSLIBDIR=$G4LIB/$G4SYSTEM
     fi
     check_library "libG4navigation" "$enable_shared" "$g4libdir" \
-	$G4SYSLIBDIR /opt/geant4/lib /opt/g4/lib /usr/local/lib /usr/lib
+	$G4SYSLIBDIR /opt/geant4/lib /opt/g4/lib /usr/local/lib /usr/lib /usr/lib/geant4
 
     g4rootlib=$found_lib
     g4libdir=$found_dir
 
     # Check for CLHEP include
     check_header "CLHEP/Vector/Rotation.h" "$clhepincdir" "$clhepincdir" \
-        $CLHEP_INCLUDE_DIR
+        $CLHEP_INCLUDE_DIR /usr/include
     clhepinc=$found_hdr
     clhepincdir=$found_dir
 
@@ -2736,7 +2736,7 @@
 if test ! "x$enable_pythia8" = "xno" ; then
     # Check for PYTHIA8 include and library
     check_header "Pythia.h" "$pythia8incdir" $PYTHIA8 $PYTHIA8/include \
-        /opt/pythia8/include /usr/local/include /usr/include
+        /opt/pythia8/include /usr/local/include /usr/include /usr/include/pythia
     pythia8incdir=$found_dir
 
     check_library "libpythia8" "$enable_shared" "$pythia8libdir" \
@@ -3077,7 +3077,7 @@
                    libpython python Python" \
         "$enable_shared" "$pythonlibdir" $PYTHONDIR/libs $PYTHONDIR/lib \
         $PYTHONDIR $pythonsyslibdir $pythonsyslibdir/config \
-        $pythonsyslibdir2
+        $pythonsyslibdir2 /usr/lib
     pythonlib=$found_lib
     pythonlibdir=$found_dir
 
