--- root/math/minuit2/src/Numerical2PGradientCalculator.cxx.orig	2009-08-10 03:05:25.196102125 +0400
+++ root/math/minuit2/src/Numerical2PGradientCalculator.cxx	2009-08-10 03:06:17.649416449 +0400
@@ -208,9 +208,11 @@
    //   std::cout<<"final grd: "<<grd<<std::endl;
    //   std::cout<<"########### return from Numerical2PDerivative"<<std::endl;
 
+#ifndef _OPENMP
    mpiproc.SyncVector(grd);
    mpiproc.SyncVector(g2);
    mpiproc.SyncVector(gstep);
+#endif
 
    return FunctionGradient(grd, g2, gstep);
 }
