diff -Naur root.orig/cint/cint/Module.mk root/cint/cint/Module.mk
--- root.orig/cint/cint/Module.mk	2009-06-30 11:22:30.000000000 +0400
+++ root/cint/cint/Module.mk	2009-08-09 16:32:52.767182272 +0400
@@ -249,7 +249,7 @@
 
 $(CINTLIB):     $(CINTO)
 		@$(MAKELIB) $(PLATFORM) $(LD) "$(LDFLAGS)" \
-		   "$(SOFLAGS)" libCint.$(SOEXT) $@ "$^" "$(CINTLIBEXTRA)"
+		   "$(SOFLAGS)" libCint.$(SOEXT) $@ "$^" "$(CILIBS)" "$(CINTLIBEXTRA)"
 
 $(CINT):        $(CINTEXEO) $(CINTLIB)
 		$(LD) $(LDFLAGS) -o $@ $(CINTEXEO) \
diff -Naur root.orig/configure root/configure
--- root.orig/configure	2009-06-30 11:23:50.000000000 +0400
+++ root/configure	2009-08-09 16:32:52.767182272 +0400
@@ -667,7 +667,7 @@
 	    cat <<EOF > conftest.mk
 include config/Makefile.${arch}
 conftest:conftest.c $symbolfile
-	\$(CC) \$(CFLAGS) \$(LDFLAGS) $symbolfile \$< -o \$@
+	\$(CC) \$(CFLAGS) \$(LDFLAGS) \$< $symbolfile -o \$@
 
 conftest.c:
 	echo "extern int $symbol (); " > \$@
