--- check-updates.sh.orig	2002-10-24 11:15:36.000000000 +0400
+++ check-updates.sh	2002-10-24 11:16:19.000000000 +0400
@@ -38,9 +38,9 @@
 PARAMETERS=$@
 CRON=0
 QUIET=0
-TEMPDIR="/var/tmp/f-prot/"
+TEMPDIR="/opt/f-prot/tmp/"
 DOWNLOADSIGN=0
-DEFDIR="/usr/local/f-prot/"
+DEFDIR="/opt/f-prot/"
 SERVER=""
 FALLBACKSERVER="http://updates.f-prot.com/files/"
 FPROTDIR=$DEFDIR
--- f-prot.sh.org	2002-11-16 21:11:54.000000000 +0100
+++ f-prot.sh	2002-11-16 21:12:01.000000000 +0100
@@ -2,4 +2,4 @@
 #
 # This is a shell script to invoke the F-Prot OnDemand Scanner for Linux.
 #
-exec /usr/local/f-prot/f-prot ${@+"$@"}
+exec /opt/f-prot/f-prot ${@+"$@"}
