--- kcontrol-3.5.6/kcontrol/kicker/menutab_impl.cpp.orig	2007-02-18 22:30:44.000000000 -0330
+++ kcontrol-3.5.6/kcontrol/kicker/menutab_impl.cpp	2007-02-19 12:22:41.000000000 -0330
@@ -146,6 +146,12 @@
 {
     if (m_comboMenuStyle->currentItem()==1) {
        m_openOnHover->setEnabled(false);
+       kcfg_ScrollFlipView->setEnabled(false);
+       kcfg_KickoffSwitchTabsOnHover->setEnabled(false);
+       m_KickoffFontPointSizeOffset->setEnabled(false);
+       kcfg_KickoffFontPointSizeOffset->setEnabled(false);
+       m_KickoffTabBarFormat->setEnabled(false);
+       kcfg_KickoffTabBarFormat->setEnabled(false);
        m_subMenus->setEnabled(true);
        kcfg_UseSidePixmap->setEnabled(true);
        kcfg_MenuEntryFormat->setEnabled(true);
@@ -154,6 +160,12 @@
     }
     else {
        m_openOnHover->setEnabled(true);
+       kcfg_ScrollFlipView->setEnabled(true);
+       kcfg_KickoffSwitchTabsOnHover->setEnabled(true);
+       m_KickoffFontPointSizeOffset->setEnabled(true);
+       kcfg_KickoffFontPointSizeOffset->setEnabled(true);
+       m_KickoffTabBarFormat->setEnabled(true);
+       kcfg_KickoffTabBarFormat->setEnabled(true);
        m_subMenus->setEnabled(false);
        kcfg_UseSidePixmap->setEnabled(false);
        kcfg_MenuEntryFormat->setEnabled(false);
@@ -196,7 +208,7 @@
     c->writeEntry("OpenOnHover", m_openOnHover->isChecked());
     c->sync();
 
-    if (kmenusetting != oldkmenusetting)
+    // if (kmenusetting != oldkmenusetting)
         DCOPRef ("kicker", "default").call("restart()");
 }
 
--- kcontrol-3.5.6/kcontrol/kicker/menutab.ui.orig	2007-02-18 22:31:00.000000000 -0330
+++ kcontrol-3.5.6/kcontrol/kicker/menutab.ui	2007-02-19 10:54:13.000000000 -0330
@@ -37,16 +37,19 @@
                     <property name="buddy" stdset="0">
                         <cstring>comboMenuStyle</cstring>
                     </property>
+		    <property name="whatsThis" stdset="0">
+		    	<string>Here you can choose between the new Kickoff menu or the Legacy K-Menu</string>
+                    </property>
                 </widget>
                 <widget class="QComboBox">
                     <item>
                         <property name="text">
-                            <string>SUSE</string>
+                            <string>Kickoff</string>
                         </property>
                     </item>
                     <item>
                         <property name="text">
-                            <string>KDE</string>
+                            <string>Legacy</string>
                         </property>
                     </item>
                     <property name="name">
@@ -91,14 +94,6 @@
                         <property name="name">
                             <cstring>unnamed</cstring>
                         </property>
-                <widget class="QCheckBox" row="0" column="0">
-                    <property name="name">
-                        <cstring>m_openOnHover</cstring>
-                    </property>
-                    <property name="text">
-                        <string>Open menu on mouse hover</string>
-                    </property>
-                </widget>
                         <widget class="QButtonGroup">
                             <property name="name">
                                 <cstring>kcfg_MenuEntryFormat</cstring>
@@ -257,6 +252,186 @@
         </widget>
         <widget class="QGroupBox">
             <property name="name">
+                <cstring>m_kickoffGroup</cstring>
+            </property>
+            <property name="title">
+                <string>Kickoff Menu</string>
+            </property>
+            <grid>
+                <property name="name">
+                    <cstring>unnamed</cstring>
+                </property>
+                <widget class="QCheckBox" row="0" column="0">
+                    <property name="name">
+                        <cstring>m_openOnHover</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Open menu on mouse hover</string>
+                    </property>
+                    <property name="whatsThis" stdset="0">
+                        <string>Enables or Disables the Kickoff menu pop-up on mouse hover.</string>
+                    </property>
+                </widget>
+                <spacer row="0" column="1" rowspan="3" colspan="1">
+                    <property name="name">
+                        <cstring>SpacerKickoff1</cstring>
+                    </property>
+                    <property name="orientation">
+                        <enum>Horizontal</enum>
+                    </property>
+                    <property name="sizeType">
+                        <enum>Fixed</enum>
+                    </property>
+                    <property name="sizeHint">
+                        <size>
+                            <width>20</width>
+                            <height>0</height>
+                        </size>
+                    </property>
+                </spacer>
+                <widget class="QCheckBox" row="0" column="2">
+                    <property name="name">
+                        <cstring>kcfg_ScrollFlipView</cstring>
+                    </property>
+                    <property name="text">
+                        <string>S&amp;croll flip view</string>
+                    </property>
+                    <property name="checked">
+                        <bool>true</bool>
+                    </property>
+                    <property name="whatsThis" stdset="0">
+                        <string>Enables or Disables the scrolling effect in the Applications menu.</string>
+                    </property>
+                </widget>
+                <spacer row="0" column="3" rowspan="3" colspan="1">
+                    <property name="name">
+                        <cstring>SpacerKickoff2</cstring>
+                    </property>
+                    <property name="orientation">
+                        <enum>Horizontal</enum>
+                    </property>
+                    <property name="sizeType">
+                        <enum>Expanding</enum>
+                    </property>
+                    <property name="sizeHint">
+                        <size>
+                            <width>20</width>
+                            <height>0</height>
+                        </size>
+                    </property>
+                </spacer>
+                <widget class="QCheckBox" row="1" column="0">
+                    <property name="name">
+                        <cstring>kcfg_KickoffSwitchTabsOnHover</cstring>
+                    </property>
+                    <property name="text">
+                        <string>S&amp;witch tabs on hover</string>
+                   </property>
+                    <property name="checked">
+                        <bool>false</bool>
+                    </property>
+                    <property name="whatsThis" stdset="0">
+                        <string>Enables or Disables whether hovering the mouse will switch to new tab in the Kickoff Menu.</string>
+                    </property>
+                </widget>
+                <widget class="QLayoutWidget" row="2" column="0">
+                    <property name="name">
+                        <cstring>Layout3</cstring>
+                    </property>
+                    <hbox>
+                        <property name="name">
+                            <cstring>unnamed</cstring>
+                        </property>
+                        <widget class="QLabel">
+                            <property name="name">
+                                <cstring>m_KickoffFontPointSizeOffset</cstring>
+                            </property>
+                            <property name="text">
+                                <string>Relative font size:</string>
+                            </property>
+                            <property name="buddy" stdset="0">
+                                <cstring>kcfg_KickoffFontPointSizeOffset</cstring>
+                            </property>
+                            <property name="whatsThis" stdset="0">
+                                <string>Takes positive and negative values and is added to the calculated font sizes, relative to your system font size.</string>
+                            </property>
+                        </widget>
+                        <widget class="KIntNumInput">
+                            <property name="name">
+                                <cstring>kcfg_KickoffFontPointSizeOffset</cstring>
+                            </property>
+                            <property name="value">
+                                <number>2</number>
+                            </property>
+                            <property name="minValue">
+                                <number>-100</number>
+                            </property>
+                            <property name="maxValue">
+                                <number>100</number>
+                            </property>
+                            <property name="whatsThis" stdset="0">
+                                <string>Takes positive and negative values and is added to the calculated font sizes, relative to your system font size.</string>
+                            </property>
+                        </widget>
+                    </hbox>
+                </widget>
+                <widget class="QLayoutWidget" row="2" column="2">
+                    <property name="name">
+                        <cstring>Layout3</cstring>
+                    </property>
+                    <hbox>
+                        <property name="name">
+                            <cstring>unnamed</cstring>
+                        </property>
+                        <widget class="QLabel">
+                            <property name="name">
+                                <cstring>m_KickoffTabBarFormat</cstring>
+                            </property>
+                            <property name="text">
+                                <string>T&amp;ab bar format:</string>
+                            </property>
+                            <property name="buddy" stdset="0">
+                                <cstring>comboMenuStyle</cstring>
+                            </property>
+		            <property name="whatsThis" stdset="0">
+		    	        <string>Used to alter the Menu Tab appearance. Valid values are LabelAndIcon, LabelOnly and IconOnly.</string>
+                            </property>
+                        </widget>
+                        <widget class="QComboBox">
+                            <item>
+                                <property name="text">
+                                    <string>Label and Icon</string>
+                                </property>
+                                <property name="value">
+                                    <string>LabelAndIcon</string>
+                                </property>
+                            </item>
+                            <item>
+                                <property name="text">
+                                    <string>Label Only</string>
+                                </property>
+                                <property name="value">
+                                    <string>LabelOnly</string>
+                                </property>
+                            </item>
+                            <item>
+                                <property name="text">
+                                    <string>Icon Only</string>
+                                </property>
+                                <property name="value">
+                                    <string>IconOnly</string>
+                                </property>
+                            </item>
+                            <property name="name">
+                                <cstring>kcfg_KickoffTabBarFormat</cstring>
+                            </property>
+                        </widget>
+                    </hbox>
+                </widget>
+            </grid>
+        </widget>
+        <widget class="QGroupBox">
+            <property name="name">
                 <cstring>m_browserGroup</cstring>
             </property>
             <property name="title">
@@ -441,12 +616,18 @@
 <customwidgets>
 </customwidgets>
 <tabstops>
+    <tabstop>m_comboMenuStyle</tabstop>
     <tabstop>m_formatSimple</tabstop>
     <tabstop>m_formatNameDesc</tabstop>
+    <tabstop>m_formatDescOnly</tabstop>
     <tabstop>m_formDescName</tabstop>
     <tabstop>kcfg_UseSidePixmap</tabstop>
     <tabstop>m_editKMenuButton</tabstop>
     <tabstop>m_subMenus</tabstop>
+    <tabstop>m_openOnHover</tabstop>
+    <tabstop>kcfg_ScrollFlipView</tabstop>
+    <tabstop>kcfg_KickoffSwitchTabsOnHover</tabstop>
+    <tabstop>kcfg_KickoffFontPointSizeOffset</tabstop>
     <tabstop>kcfg_ShowHiddenFiles</tabstop>
     <tabstop>kcfg_MaxEntries2</tabstop>
     <tabstop>kcfg_RecentVsOften</tabstop>
