diff -Naurp etoile-foundation-0.4.1_pre4508/Frameworks/EtoileFoundation/Source/NSFileManager+TempFile.m etoile-foundation-0.4.1_pre4508.new/Frameworks/EtoileFoundation/Source/NSFileManager+TempFile.m
--- etoile-foundation-0.4.1_pre4508/Frameworks/EtoileFoundation/Source/NSFileManager+TempFile.m	2009-03-13 21:00:54.000000000 +0300
+++ etoile-foundation-0.4.1_pre4508.new/Frameworks/EtoileFoundation/Source/NSFileManager+TempFile.m	2009-03-13 21:03:13.000000000 +0300
@@ -22,7 +22,11 @@ static char * makeTempPattern(void)
 - (NSString*) tempDirectory
 {
 	char * pattern = makeTempPattern();
-	mkdtemp(pattern);
+	if (NULL == mkdtemp(pattern))
+	{
+		free(pattern);
+		return NULL;
+	}
 	NSString * dirName = [NSString stringWithUTF8String:pattern];
 	free(pattern);
 	return dirName;
