diff -Naur graphite2-1.0.2/CMakeLists.txt graphite2-1.0.2-new/CMakeLists.txt
--- graphite2-1.0.2/CMakeLists.txt	2011-08-18 23:02:50.000000000 +0400
+++ graphite2-1.0.2-new/CMakeLists.txt	2011-08-18 23:04:15.000000000 +0400
@@ -61,7 +61,7 @@
 
 add_subdirectory(src)
 add_subdirectory(gr2fonttest)
-add_subdirectory(tests)
+# add_subdirectory(tests)
 add_subdirectory(doc)
 set(version 2.0.0)
 set(libdir ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
