--- WindowMaker-0.80.2/WINGs/WINGs/WINGs.h.textfield	2003-10-20 23:57:14 +0400
+++ WindowMaker-0.80.2/WINGs/WINGs/WINGs.h	2003-10-22 09:21:11 +0400
@@ -1154,6 +1154,8 @@
 
 void WMSetTextFieldCursorPosition(WMTextField *tPtr, unsigned int position);
 
+unsigned WMGetTextFieldCursorPosition(WMTextField *tPtr);
+
 void WMSetTextFieldNextTextField(WMTextField *tPtr, WMTextField *next);
 
 void WMSetTextFieldPrevTextField(WMTextField *tPtr, WMTextField *prev);
--- WindowMaker-0.80.2/WINGs/wtextfield.c.textfield	2003-10-20 23:57:14 +0400
+++ WindowMaker-0.80.2/WINGs/wtextfield.c	2003-10-22 09:20:31 +0400
@@ -617,6 +617,14 @@
     }
 }
 
+unsigned
+WMGetTextFieldCursorPosition(WMTextField *tPtr)
+{
+    CHECK_CLASS(tPtr, WC_TextField);
+
+    return tPtr->cursorPosition;
+}
+
 
 void
 WMSetTextFieldNextTextField(WMTextField *tPtr, WMTextField *next)
