Index: configure.in
===================================================================
--- configure.in	(Revision 2743)
+++ configure.in	(Arbeitskopie)
@@ -191,7 +191,6 @@
 Makefile
 src/Makefile
 auth-dialog/Makefile
-auth-dialog-general/Makefile
 properties/Makefile
 po/Makefile.in
 ])
Index: Makefile.am
===================================================================
--- Makefile.am	(Revision 2743)
+++ Makefile.am	(Arbeitskopie)
@@ -1,7 +1,7 @@
 AUTOMAKE_OPTIONS = foreign
 
 if WITH_GNOME
-SUBDIRS = src auth-dialog auth-dialog-general properties po
+SUBDIRS = src auth-dialog properties po
 else
 SUBDIRS = src
 endif
