Index: Makefile.am
===================================================================
--- Makefile.am	(Revision 2743)
+++ Makefile.am	(Arbeitskopie)
@@ -15,7 +15,7 @@
 
 %.name: %.name.in Makefile
 	@sed -e "s|%bindir%|$(bindir)|g" \
-	     -e "s|%libdir%|$(libdir)|g" \
+	     -e "s|%prefix%|$(prefix)|g" \
 	     -e "s|%libexecdir%|$(libexecdir)|g" $< > $@
 
 if WITH_GNOME
Index: nm-ppp-starter.name.in
===================================================================
--- nm-ppp-starter.name.in	(Revision 2743)
+++ nm-ppp-starter.name.in	(Arbeitskopie)
@@ -1,8 +1,8 @@
 [VPN Connection]
 name=ppp
 service=org.freedesktop.NetworkManager.ppp_starter
-program=@bindir@/nm-ppp-starter
+program=%bindir%/nm-ppp-starter
 
 [GNOME]
-auth-dialog=@libexecdir@/nm-ppp-auth-dialog
-properties=@prefix@/${LIB}/libnm-ppp-properties
+auth-dialog=%libexecdir%/nm-ppp-auth-dialog
+properties=%prefix%/${LIB}/libnm-ppp-properties
