diff -ur ekiga-3.2.0-old/src/gui/assistant.cpp ekiga-3.2.0/src/gui/assistant.cpp
--- ekiga-3.2.0-old/src/gui/assistant.cpp	2009-03-07 12:09:14.000000000 +0100
+++ ekiga-3.2.0/src/gui/assistant.cpp	2009-04-10 17:31:01.000000000 +0200
@@ -1299,7 +1299,7 @@
   gchar **array = NULL;
   unsigned i;
 
-  array = (gchar**) malloc (sizeof(gchar*) * (list.size() + 1));
+  array = (gchar**) g_malloc (sizeof(gchar*) * (list.size() + 1));
   for (i = 0; i < list.size(); i++)
     array[i] = (gchar*) list[i].c_str();
   array[i] = NULL;
diff -ur ekiga-3.2.0-old/src/gui/preferences.cpp ekiga-3.2.0/src/gui/preferences.cpp
--- ekiga-3.2.0-old/src/gui/preferences.cpp	2009-01-07 11:02:09.000000000 +0100
+++ ekiga-3.2.0/src/gui/preferences.cpp	2009-04-16 14:28:08.000000000 +0200
@@ -862,7 +862,7 @@
   gchar **array = NULL;
   unsigned i;
 
-  array = (gchar**) malloc (sizeof(gchar*) * (list.size() + 1));
+  array = (gchar**) g_malloc (sizeof(gchar*) * (list.size() + 1));
   for (i = 0; i < list.size(); i++)
     array[i] = (gchar*) list[i].c_str();
   array[i] = NULL;
