--- lib/engine/components/opal/opal-call-manager.cpp.old	2009-04-17 23:32:21.000000000 -0400
+++ lib/engine/components/opal/opal-call-manager.cpp	2009-04-17 23:32:49.000000000 -0400
@@ -116,7 +116,7 @@
   forward_on_no_answer = false;
   forward_on_busy = false;
   unconditional_forward = false;
-  stun_enabled = false;
+  stun_enabled = true;
 
   // Create video devices
   PVideoDevice::OpenArgs video = GetVideoOutputDevice();
@@ -496,6 +496,7 @@
 void CallManager::set_stun_server (const std::string & server)
 {
   stun_server = server;
+	 start();
 }
 
 
