diff -Naur ../work/flamethrower-0.1.8/Makefile.PL flamethrower-0.1.8/Makefile.PL
--- ../work/flamethrower-0.1.8/Makefile.PL	2006-01-24 05:08:52.000000000 +0300
+++ flamethrower-0.1.8/Makefile.PL	2006-01-31 17:22:13.000000000 +0300
@@ -9,9 +9,9 @@
     my $string = <<EOF;
 
 set_ver::
-\t${install} \$(PREFIX)/bin/flamethrower
-\t${install} \$(PREFIX)/bin/flamethrowerd
-\t${install} \$(PREFIX)/lib/flamethrower/Flamethrower.pm
+\t${install} \$(DESTDIR)/\$(PREFIX)/bin/flamethrower
+\t${install} \$(DESTDIR)/\$(PREFIX)/bin/flamethrowerd
+\t${install} \$(DESTDIR)/\$(PREFIX)/lib/flamethrower/Flamethrower.pm
 
 install_configs::
 \tinstall -d \$(CONFDIR)/etc/init.d/
