--- GNOME_MacMenuApplet.server	2006-11-04 10:01:33.000000000 -0500
+++ GNOME_MacMenuApplet.server	2006-11-04 10:02:05.000000000 -0500
@@ -1,6 +1,6 @@
 <oaf_info>
   <oaf_server iid="OAFIID:GNOME_MacMenuApplet_Factory" type="exe"
-    location="/opt/gnome/libexec/gnome-macmenu-applet">
+    location="/usr/libexec/gnome-macmenu-applet">
     <oaf_attribute name="repo_ids" type="stringv">
       <item value="IDL:Bonobo/GenericFactory:1.0"/>
       <item value="IDL:Bonobo/Unknown:1.0"/>
@@ -16,7 +16,7 @@
       <item value="IDL:Bonobo/Unknown:1.0"/>
     </oaf_attribute>
     <oaf_attribute name="name" type="string" value="Mac Menu"/>
-    <oaf_attribute name="description" type="string" value="Mac Menubar applet for gtk2-aqd"/>
+    <oaf_attribute name="description" type="string" value="Mac Menubar applet"/>
     <oaf_attribute name="panel:icon" type="string" value="gnome-fs-home"/>
   </oaf_server>
 </oaf_info>
