diff -Naur ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/AppViewer/AppViewer.m GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/AppViewer/AppViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/AppViewer/AppViewer.m	2006-08-07 16:12:23.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/AppViewer/AppViewer.m	2006-08-07 16:17:04.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "AppViewer.h"
 
 @implementation AppViewer
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/FolderViewer/FolderViewer.m GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/FolderViewer/FolderViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/FolderViewer/FolderViewer.m	2006-08-07 16:15:54.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/FolderViewer/FolderViewer.m	2006-08-07 16:12:24.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "FolderViewer.h"
 
 #define byname 0
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/IBViewViewer/IBViewViewer.m GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/IBViewViewer/IBViewViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/IBViewViewer/IBViewViewer.m	2006-08-07 16:12:23.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/IBViewViewer/IBViewViewer.m	2006-08-07 16:18:46.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "IBViewViewer.h"
 
 @implementation CustomView 
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/ImageViewer/ImageViewer.m GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/ImageViewer/ImageViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/ImageViewer/ImageViewer.m	2006-08-07 16:12:23.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/ImageViewer/ImageViewer.m	2006-08-07 16:16:23.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "ImageViewer.h"
 #include <math.h>
 
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSColorViewer/NSColorViewer.m GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSColorViewer/NSColorViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSColorViewer/NSColorViewer.m	2006-08-07 16:12:23.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSColorViewer/NSColorViewer.m	2006-08-07 16:18:29.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include <math.h>
 #include "NSColorViewer.h"
 
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSRTFViewer/NSRTFViewer.m GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSRTFViewer/NSRTFViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSRTFViewer/NSRTFViewer.m	2006-08-07 16:12:23.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSRTFViewer/NSRTFViewer.m	2006-08-07 16:18:14.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "NSRTFViewer.h"
 
 #define STR  0
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSTIFFViewer/NSTIFFViewer.m GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSTIFFViewer/NSTIFFViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSTIFFViewer/NSTIFFViewer.m	2006-08-07 16:12:23.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/NSTIFFViewer/NSTIFFViewer.m	2006-08-07 16:17:57.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "NSTIFFViewer.h"
 
 @implementation NSTIFFViewer
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/PdfViewer/PdfViewer.m GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/PdfViewer/PdfViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/PdfViewer/PdfViewer.m	2006-08-07 16:12:23.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/PdfViewer/PdfViewer.m	2006-08-07 16:17:36.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include <PDFKit/PDFDocument.h>
 #include <PDFKit/PDFImageRep.h>
 #include <math.h>
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/RtfViewer/RtfViewer.m GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/RtfViewer/RtfViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/RtfViewer/RtfViewer.m	2006-08-07 16:12:23.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/RtfViewer/RtfViewer.m	2006-08-07 16:17:21.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "RtfViewer.h"
 
 #define MAXDATA 1000
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/SoundViewer/SoundViewer.m GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/SoundViewer/SoundViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/SoundViewer/SoundViewer.m	2006-08-07 16:12:23.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/Inspector/ContentViewers/SoundViewer/SoundViewer.m	2006-08-07 16:16:46.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "SoundViewer.h"
 
 @implementation SoundViewer
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/Desktop/GWDesktopManager.m GWorkspace-0.8.3_pre20060807/GWorkspace/Desktop/GWDesktopManager.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/Desktop/GWDesktopManager.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/Desktop/GWDesktopManager.m	2006-08-07 16:26:30.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "GWDesktopManager.h"
 #include "GWDesktopWindow.h"
 #include "GWDesktopView.h"
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWSpatialViewer.m GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWSpatialViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWSpatialViewer.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWSpatialViewer.m	2006-08-07 16:28:20.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include <math.h>
 #include "GWSpatialViewer.h"
 #include "GWViewersManager.h"
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerBrowser.m GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerBrowser.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerBrowser.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerBrowser.m	2006-08-07 16:36:17.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "GWViewerBrowser.h"
 #include "FSNBrowserColumn.h"
 #include "FSNBrowserMatrix.h"
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerIconsPath.m GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerIconsPath.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerIconsPath.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerIconsPath.m	2006-08-07 16:36:43.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include <math.h>
 #include <sys/types.h>
 #include <unistd.h>
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerIconsView.m GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerIconsView.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerIconsView.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerIconsView.m	2006-08-07 16:36:36.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "GWViewerIconsView.h"
 #include "FSNIcon.h"
 #include "GWSpatialViewer.h"
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerListView.m GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerListView.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerListView.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerListView.m	2006-08-07 16:36:59.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "GWViewerListView.h"
 #include "GWSpatialViewer.h"
 #include "GWViewer.h"
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewer.m GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewer.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewer.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewer.m	2006-08-07 16:27:55.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include <math.h>
 #include "GWViewer.h"
 #include "GWViewersManager.h"
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerPathsPopUp.m GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerPathsPopUp.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerPathsPopUp.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewerPathsPopUp.m	2006-08-07 16:37:07.000000000 +0400
@@ -24,6 +24,7 @@
 
 #include <Foundation/Foundation.h>
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "FSNode.h"
 #include "GWViewerPathsPopUp.h"
 
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewersManager.m GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewersManager.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewersManager.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/FileViewer/GWViewersManager.m	2006-08-07 16:27:32.000000000 +0400
@@ -23,6 +23,7 @@
  */
 
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "GWViewersManager.h"
 #include "GWViewer.h"
 #include "GWSpatialViewer.h"
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/GWorkspace.m GWorkspace-0.8.3_pre20060807/GWorkspace/GWorkspace.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/GWorkspace.m	2006-08-07 16:25:32.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/GWorkspace.m	2006-08-07 16:24:35.000000000 +0400
@@ -24,6 +24,7 @@
 
 #include <Foundation/Foundation.h>
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include <math.h>
 #include "GWFunctions.h"
 #include "FSNodeRep.h"
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/History/History.m GWorkspace-0.8.3_pre20060807/GWorkspace/History/History.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/History/History.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/History/History.m	2006-08-07 16:38:17.000000000 +0400
@@ -24,6 +24,7 @@
 
 #include <Foundation/Foundation.h>
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include "History.h"
 #include "GWViewersManager.h"
 #include "GWFunctions.h"
diff -Naur ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/WorkspaceApplication.m GWorkspace-0.8.3_pre20060807/GWorkspace/WorkspaceApplication.m
--- ../work/GWorkspace-0.8.3_pre20060807/GWorkspace/WorkspaceApplication.m	2006-08-07 16:20:27.000000000 +0400
+++ GWorkspace-0.8.3_pre20060807/GWorkspace/WorkspaceApplication.m	2006-08-07 16:26:05.000000000 +0400
@@ -24,6 +24,7 @@
 
 #include <Foundation/Foundation.h>
 #include <AppKit/AppKit.h>
+#include <AppKit/NSHelpManager.h>
 #include <math.h>
 #include "GWorkspace.h"
 #include "GWFunctions.h"
