--- FerrisUI/EditSQLColumns.cpp.orig	2006-10-02 22:32:09.000000000 +0300
+++ FerrisUI/EditSQLColumns.cpp	2006-10-02 22:32:32.000000000 +0300
@@ -90,7 +90,7 @@
         GtkTreeStore*   w_treemodel = esl->w_treemodel;
         GtkWidget*      w_treeview  = esl->w_treeview;
 
-        int cidx = (int)g_object_get_data( G_OBJECT( cell ), GOBJ_CELLID_K );
+        size_t cidx = (size_t)g_object_get_data( G_OBJECT( cell ), GOBJ_CELLID_K );
         cerr << "EditSQLColumns_edited_cb() cidx:" << cidx << " new_text:" << new_text << endl;
         GtkTreeModel *model = GTK_TREE_MODEL (w_treemodel);
         GtkTreeIter iter;
--- cc/capplets/auth/ferris-capplet-auth.cpp.orig	2006-10-03 01:01:39.000000000 +0300
+++ cc/capplets/auth/ferris-capplet-auth.cpp	2006-10-03 01:02:10.000000000 +0300
@@ -377,7 +377,7 @@
     cell_edited_cb (GtkCellRendererText *cell, gchar *path_string,
                     gchar *new_text, gpointer udata)
     {
-        int cidx = (int)udata;
+        size_t cidx = (size_t)udata;
         GtkTreeModel *model = GTK_TREE_MODEL (w_treemodel);
         GtkTreeIter iter;
         GtkTreePath *path = gtk_tree_path_new_from_string (path_string);
--- Ferris/TypeDecl.hh.orig	2006-10-03 01:07:11.000000000 +0300
+++ Ferris/TypeDecl.hh	2006-10-03 01:07:27.000000000 +0300
@@ -124,7 +124,7 @@
     /**
      * every emblem has a unique numeric identifier
      */
-    typedef guint32 emblemID_t;
+    typedef size_t emblemID_t;
     
     class Attribute;
     class AttributeCollection;
--- cc/capplets/medallion/ferris-capplet-medallion.cpp.orig	2006-10-03 01:04:09.000000000 +0300
+++ cc/capplets/medallion/ferris-capplet-medallion.cpp	2006-10-03 01:05:22.000000000 +0300
@@ -578,7 +578,7 @@
 {
     string v = new_text ? new_text : "";
     double dv = 0.0;
-    int cidx = (int)user_data;
+    size_t cidx = (size_t)user_data;
     cerr << "edited_cb() c:" << cidx << " new_text:" << new_text << endl;
 
     if( cidx >= C_EMBLEM_DLAT && cidx <= C_EMBLEM_DRANGE )
--- cc/capplets/agents/ferris-capplet-agents.cpp.orig	2006-10-03 01:55:59.000000000 +0300
+++ cc/capplets/agents/ferris-capplet-agents.cpp	2006-10-03 01:56:27.000000000 +0300
@@ -307,7 +307,7 @@
 edited_cb (GtkCellRendererText *cell, gchar *path_string, gchar *new_text, gpointer user_data )
 {
     string v = new_text ? new_text : "";
-    int cidx = (int)user_data;
+    size_t cidx = (size_t)user_data;
     cerr << "edited_cb() c:" << cidx << " new_text:" << new_text << endl;
 
     bool acceptNewValue = true;
--- cc/capplets/rdf/ferris-capplet-rdf.cpp.orig	2006-10-03 01:58:06.000000000 +0300
+++ cc/capplets/rdf/ferris-capplet-rdf.cpp	2006-10-03 01:58:25.000000000 +0300
@@ -356,7 +356,7 @@
 edited_cb (GtkCellRendererText *cell, gchar *path_string, gchar *new_text, gpointer user_data )
 {
     string v = new_text ? new_text : "";
-    int cidx = (int)user_data;
+    size_t cidx = (size_t)user_data;
     cerr << "edited_cb() c:" << cidx << " new_text:" << new_text << endl;
 
     bool acceptNewValue = true;
