--- configure.ac.orig	2006-10-04 03:05:14.330771472 +0300
+++ configure.ac	2006-10-04 03:07:47.376504984 +0300
@@ -3831,7 +3831,7 @@
 
 AC_DEFINE( FERRIS_USER_NAME_NOBODY, "nobody" )
 
-DESKTOP_FILE_DIR=`eval echo ${PREFIXSTR}/ferris/dot-desktop-files`
+DESKTOP_FILE_DIR=`eval echo ${PREFIXSTR}/share/applications`
 AC_SUBST(DESKTOP_FILE_DIR)
 
 FERRIS_AS_TEXT_PLUGIN_DIR="$LIBDIR/ferris/plugins/astext"
--- macros/ferrismacros.m4.orig	2006-10-03 23:48:05.000000000 +0300
+++ macros/ferrismacros.m4	2006-10-04 00:10:23.000000000 +0300
@@ -26,7 +26,7 @@
 dnl @author Angus Lees <gusl@cse.unsw.edu.au>
 dnl @version $Id: ferrismacros.m4,v 1.3 2005/12/25 09:35:03 ben Exp $
 
-AC_DEFUN(AC_PATH_GENERIC,
+AC_DEFUN([AC_PATH_GENERIC],
 [dnl
 dnl we're going to need uppercase, lowercase and user-friendly versions of the
 dnl string `LIBRARY'
@@ -145,7 +145,7 @@
 
 dnl Force the use of libtool and AC_TRY_LINK!
 dnl http://www.mail-archive.com/libtool@gnu.org/msg01271.html
-AC_DEFUN(AM_FERRIS_LIBTOOL_TRYLINK,
+AC_DEFUN([AM_FERRIS_LIBTOOL_TRYLINK],
 [dnl
 dnl
 	am_ferris_libtool_trylink_pass=no
@@ -188,7 +188,7 @@
 dnl 
 dnl the STLPORT_CFLAGS etc are all cleared after a failed test.
 dnl
-AC_DEFUN(AM_FERRIS_STLPORT_INTERNAL_TRYLINK,
+AC_DEFUN([AM_FERRIS_STLPORT_INTERNAL_TRYLINK],
 [dnl
 dnl
 	AC_LANG_CPLUSPLUS
@@ -264,7 +264,7 @@
 dnl    AC_DEFINE( HAVE_STLPORT )
 dnl    AC_DEFINE( STLPORT_IOSIZE )
 dnl
-AC_DEFUN(AM_FERRIS_STLPORT,
+AC_DEFUN([AM_FERRIS_STLPORT],
 [dnl 
 dnl Get the cflags and libraries from pkg-config, stlport-config or attempt to
 dnl detect the STLPort on the users system.
@@ -435,7 +435,7 @@
 dnl    AC_DEFINE( HAVE_STLPORT )
 dnl    AC_DEFINE( STLPORT_IOSIZE )
 dnl
-AC_DEFUN(AM_FERRIS_STLPORT_OPTIONAL,
+AC_DEFUN([AM_FERRIS_STLPORT_OPTIONAL],
 [dnl 
 dnl
 
@@ -488,7 +488,7 @@
 dnl FERRIS_CFLAGS and FERRIS_LIBS are set and AC_SUBST()ed when library is found.
 dnl LIBFERRIS_CFLAGS and LIBFERRIS_LIBS are set and AC_SUBST()ed when library is found.
 dnl
-AC_DEFUN(AM_FERRIS_FERRIS,
+AC_DEFUN([AM_FERRIS_FERRIS],
 [dnl 
 dnl Get the cflags and libraries from pkg-config, stlport-config or attempt to
 dnl detect the STLPort on the users system.
@@ -536,7 +536,7 @@
 dnl FERRISUI_CFLAGS and FERRISUI_LIBS are set and AC_SUBST()ed when library is found.
 dnl LIBFERRISUI_CFLAGS and LIBFERRISUI_LIBS are set and AC_SUBST()ed when library is found.
 dnl
-AC_DEFUN(AM_FERRIS_FERRISUI,
+AC_DEFUN([AM_FERRIS_FERRISUI],
 [dnl 
 dnl Get the cflags and libraries from pkg-config, stlport-config or attempt to
 dnl detect the STLPort on the users system.
@@ -583,7 +583,7 @@
 dnl
 dnl SIGC_CFLAGS and SIGC_LIBS are set and AC_SUBST()ed when library is found.
 dnl
-AC_DEFUN(AM_FERRIS_SIGC,
+AC_DEFUN([AM_FERRIS_SIGC],
 [dnl 
 dnl Get the cflags and libraries from pkg-config, stlport-config or attempt to
 dnl detect the STLPort on the users system.
@@ -634,7 +634,7 @@
 dnl
 dnl LOKI_CFLAGS and LOKI_LIBS are set and AC_SUBST()ed when library is found.
 dnl
-AC_DEFUN(AM_FERRIS_LOKI,
+AC_DEFUN([AM_FERRIS_LOKI],
 [dnl 
 dnl Get the cflags and libraries from pkg-config, stlport-config or attempt to
 dnl detect the STLPort on the users system.
@@ -676,7 +676,7 @@
 dnl STREAMS_CFLAGS and STREAMS_LIBS are set and AC_SUBST()ed when library is found.
 dnl FSTREAM_CFLAGS and FSTREAM_LIBS are set and AC_SUBST()ed when library is found.
 dnl
-AC_DEFUN(AM_FERRIS_STREAMS,
+AC_DEFUN([AM_FERRIS_STREAMS],
 [dnl 
 dnl Get the cflags and libraries from pkg-config, stlport-config or attempt to
 dnl detect the STLPort on the users system.
@@ -722,7 +722,7 @@
 dnl
 dnl STLDB4_CFLAGS and STLDB4_LIBS are set and AC_SUBST()ed when library is found.
 dnl
-AC_DEFUN(AM_FERRIS_STLDB4,
+AC_DEFUN([AM_FERRIS_STLDB4],
 [dnl 
 dnl Get the cflags and libraries from pkg-config, stlport-config or attempt to
 dnl detect the STLPort on the users system.
@@ -763,7 +763,7 @@
 dnl
 dnl FAMPP2_CFLAGS and FAMPP2_LIBS are set and AC_SUBST()ed when library is found.
 dnl
-AC_DEFUN(AM_FERRIS_FAMPP2,
+AC_DEFUN([AM_FERRIS_FAMPP2],
 [dnl 
 dnl Get the cflags and libraries from pkg-config or x-config
 dnl
@@ -801,7 +801,7 @@
 dnl Used internally to try to link to a library using C++ application
 dnl using the CFLAGS and LIBS. 
 dnl
-AC_DEFUN(AM_FERRIS_INTERNAL_TRYLINK,
+AC_DEFUN([AM_FERRIS_INTERNAL_TRYLINK],
 [dnl
 dnl
 	AC_LANG_CPLUSPLUS
@@ -836,7 +836,7 @@
 dnl Used internally to try to link to a library using C++ application
 dnl using the CFLAGS and LIBS. 
 dnl
-AC_DEFUN(AM_FERRIS_INTERNAL_TRYRUN,
+AC_DEFUN([AM_FERRIS_INTERNAL_TRYRUN],
 [dnl
 dnl
 	AC_LANG_CPLUSPLUS
@@ -877,7 +877,7 @@
 dnl XML4C_CFLAGS   and XML4C_LIBS   are set and AC_SUBST()ed when library is found.
 dnl AC_DEFINE(HAVE_XML4C) and AC_DEFINE(HAVE_XERCESC)
 dnl
-AC_DEFUN(AM_FERRIS_XERCESC,
+AC_DEFUN([AM_FERRIS_XERCESC],
 [dnl 
 dnl Get the cflags and libraries from pkg-config, stlport-config or attempt to
 dnl detect the STLPort on the users system.
@@ -1001,7 +1001,7 @@
 dnl XML4C_CFLAGS   and XML4C_LIBS   are set and AC_SUBST()ed when library is found.
 dnl AC_DEFINE(HAVE_XML4C) and AC_DEFINE(HAVE_XALAN)
 dnl
-AC_DEFUN(AM_FERRIS_XALAN,
+AC_DEFUN([AM_FERRIS_XALAN],
 [dnl 
 dnl Get the cflags and libraries from pkg-config, stlport-config or attempt to
 dnl detect the STLPort on the users system.
@@ -1156,7 +1156,7 @@
 dnl DTL_CFLAGS and DTL_LIBS are set and AC_SUBST()ed when library is found.
 dnl AC_DEFINE(HAVE_DTL) and AM_CONDITIONAL(HAVE_DTL)
 dnl
-AC_DEFUN(AM_FERRIS_DTL,
+AC_DEFUN([AM_FERRIS_DTL],
 [dnl 
 dnl
 required_version=$1
@@ -1199,7 +1199,7 @@
 dnl ######################################################################
 
 
-AC_DEFUN(AM_FERRIS_BOOST_INTERNAL_TRYLINK,
+AC_DEFUN([AM_FERRIS_BOOST_INTERNAL_TRYLINK],
 [dnl 
 dnl
 	AC_LANG_CPLUSPLUS
@@ -1249,7 +1249,7 @@
 dnl BOOST_CFLAGS and BOOST_LIBS are set and AC_SUBST()ed when library is found.
 dnl AC_DEFINE(HAVE_BOOST) and AM_CONDITIONAL(HAVE_BOOST)
 dnl
-AC_DEFUN(AM_FERRIS_BOOST,
+AC_DEFUN([AM_FERRIS_BOOST],
 [dnl 
 dnl
 required_version=$1
@@ -1320,7 +1320,7 @@
 dnl PATHAN_CFLAGS and PATHAN_LIBS are set and AC_SUBST()ed when library is found.
 dnl AC_DEFINE(HAVE_PATHAN)
 dnl
-AC_DEFUN(AM_FERRIS_PATHAN,
+AC_DEFUN([AM_FERRIS_PATHAN],
 [dnl 
 dnl
 have_package=no
@@ -1456,7 +1456,7 @@
 dnl PATHAN2_CFLAGS and PATHAN2_LIBS are set and AC_SUBST()ed when library is found.
 dnl AC_DEFINE(HAVE_PATHAN2)
 dnl
-AC_DEFUN(AM_FERRIS_PATHAN2,
+AC_DEFUN([AM_FERRIS_PATHAN2],
 [dnl 
 dnl
 have_package=no
@@ -1599,7 +1599,7 @@
 dnl AC_DEFINE(HAVE_SOCKETPP)
 dnl AM_CONDITIONAL(HAVE_SOCKETPP
 dnl
-AC_DEFUN(AM_FERRIS_SOCKETPP,
+AC_DEFUN([AM_FERRIS_SOCKETPP],
 [dnl 
 dnl
 have_package=no
@@ -7245,7 +7245,7 @@
 dnl
 dnl LIBPQXX_CFLAGS and LIBPQXX_LIBS are set and AC_SUBST()ed when library is found.
 dnl
-AC_DEFUN(AM_FERRIS_PQXX,
+AC_DEFUN([AM_FERRIS_PQXX],
 [dnl 
 dnl
 required_version=$1
@@ -7364,7 +7364,7 @@
 dnl
 dnl KDE3_CFLAGS and KDE3_LIBS are set and AC_SUBST()ed when library is found.
 dnl
-AC_DEFUN(AM_FERRIS_KDE,
+AC_DEFUN([AM_FERRIS_KDE],
 [dnl 
 dnl
 required_version=$1
--- plugins/fulltextindexers/external/Makefile.in.orig	2006-10-06 01:31:54.000000000 +0300
+++ plugins/fulltextindexers/external/Makefile.in	2006-10-06 01:35:34.000000000 +0300
@@ -1009,7 +1009,7 @@
 
 info-am:
 
-install-data-am: install-perlpluginSCRIPTS install-pluginLTLIBRARIES
+install-data-am: install-pluginLTLIBRARIES
 
 install-exec-am:
 
--- ./apps/ferris-mount-etagere-as-kml.sh.orig	2006-10-06 02:07:23.000000000 +0300
+++ ./apps/ferris-mount-etagere-as-kml.sh	2006-10-06 02:07:52.000000000 +0300
@@ -2,7 +2,7 @@
 
 mountpoint=${1:-~/ferrisfuse/etagere-as-kml};
 
-export LIBFERRIS_XSLTFS_SHEETS_URL="/usr/local/ferris/stylesheets"
+export LIBFERRIS_XSLTFS_SHEETS_URL="${datadir}/ferris/stylesheets"
 URL='single-file-filesystem:xsltfs?append-extension=.kml&stylesheet=ferris-geospatial-emblems-to-kml.xsl&reverse-stylesheet=ferris-kml-to-geospatial-emblems.xsl://context/etagere/libferris/libferris-geospatial.kml'
 
 echo "mountpoint:$mountpoint"
--- tests/ipc/shm/Makefile.in.orig	2006-10-06 02:44:19.000000000 +0300
+++ tests/ipc/shm/Makefile.in	2006-10-06 02:45:54.000000000 +0300
@@ -763,7 +763,7 @@
 MAINTAINERCLEANFILES = Makefile.in
 XML_FILES = 
 EXTRA_DIST = $(XML_FILES)
-xmldir = /usr/local/ferris/xml
+xmldir = ${datadir}/ferris/xml
 xml_DATA = $(XML_FILES)
 INCLUDES = \
 	@STLPORT_CFLAGS@                                \
--- configure.ac.orig	2006-10-06 13:17:21.521796784 +0300
+++ configure.ac	2006-10-06 13:17:59.950954664 +0300
@@ -3806,15 +3806,15 @@
 if test "x${prefix}" = 'xNONE'; then
         prefix=${ac_default_prefix};
 fi;
-if test "x${exec_prefix}" = 'xNONE'; then
-        exec_prefix=${ac_default_prefix};
-fi;
+#if test "x${exec_prefix}" = 'xNONE'; then
+#        exec_prefix=${ac_default_prefix};
+#fi;
 
 PREFIX=${prefix};  
 SYSCONFDIR=`eval echo ${sysconfdir}`
-EXECPREFIX=`eval echo ${exec_prefix}`
+#EXECPREFIX=`eval echo ${exec_prefix}`
 DOTFILE_ETC=`eval echo /etc`  # many places, but keep this one.
-BINDIR=`eval echo ${bindir}`
+#BINDIR=`eval echo ${bindir}`
 LIBDIR=`eval echo ${libdir}`  # ./Ferris/Makefile.am
 INCLUDEDIR=`eval echo ${includedir}`
 PREFIXSTR=`eval echo ${prefix}` # used in code like ./Ferris/Ferris.cpp
@@ -3823,8 +3823,8 @@
 AC_SUBST(PREFIXSTR)
 AC_SUBST(SYSCONFDIR)
 AC_SUBST(DOTFILE_ETC)
-AC_SUBST(EXECPREFIX)
-AC_SUBST(BINDIR)
+#AC_SUBST(EXECPREFIX)
+#AC_SUBST(BINDIR)
 AC_SUBST(LIBDIR)
 AC_SUBST(INCLUDEDIR)
 
--- Ferris/Makefile.in.orig	2006-10-06 16:09:55.216795896 +0300
+++ Ferris/Makefile.in	2006-10-06 16:05:23.792058728 +0300
@@ -1381,6 +1381,7 @@
 	    $(LIBTOOL) --mode=install $(libLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(libdir)/$$f"; \
 	  else :; fi; \
 	done
+	${LIBTOOL} --mode=finish ${libdir}
 
 uninstall-libLTLIBRARIES:
 	@$(NORMAL_UNINSTALL)
