diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrisa52_factory.cpp libferris-1.1.98/factories/eagenerators/libferrisa52_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrisa52_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrisa52_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -63,7 +63,7 @@
     {
         static bool r = RegisterEAGeneratorModule(
             Factory::ComposeEndsWithMatcher( EL() ),
-            "/usr/local/lib/ferris/plugins/eagenerators/libferrisa52.so",
+            "/usr/lib/ferris/plugins/eagenerators/libferrisa52.so",
             "DolbyA52",
             false,
             false,
@@ -84,7 +84,7 @@
 //     {
 //         return new GModuleMatchedEAGeneratorFactory(
 //              Factory::ComposeEndsWithMatcher( EL() ),
-//              "/usr/local/lib/ferris/plugins/eagenerators/libferrisa52.so" );
+//              "/usr/lib/ferris/plugins/eagenerators/libferrisa52.so" );
 //     }
     
 // }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrisdjvulibre_factory.cpp libferris-1.1.98/factories/eagenerators/libferrisdjvulibre_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrisdjvulibre_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrisdjvulibre_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -45,7 +45,7 @@
     {
         const char* ext       = "djvu";
         bool        writable  = true;
-        const char* libname   = "/usr/local/lib/ferris/plugins/eagenerators/libferrisdjvulibre.so";
+        const char* libname   = "/usr/lib/ferris/plugins/eagenerators/libferrisdjvulibre.so";
         const char* shortname = "djvu";
         
         bool r = Context::RegisterImageEAGeneratorModule(
@@ -77,7 +77,7 @@
 //     {
 //         static bool r = RegisterEAGeneratorModule(
 //             Factory::ComposeEndsWithMatcher( EL() ),
-//             "/usr/local/lib/ferris/plugins/eagenerators/libferrisdjvulibre.so",
+//             "/usr/lib/ferris/plugins/eagenerators/libferrisdjvulibre.so",
 //             "djvu",
 //             false,
 //             true );
@@ -97,7 +97,7 @@
 //         {
 //             return new GModuleMatchedEAGeneratorFactory(
 //                 Factory::ComposeEndsWithMatcher( EL() ),
-//                 "/usr/local/lib/ferris/plugins/eagenerators/libferrisdjvulibre.so" );
+//                 "/usr/lib/ferris/plugins/eagenerators/libferrisdjvulibre.so" );
 //         }
     
 //     }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrisexifea_factory.cpp libferris-1.1.98/factories/eagenerators/libferrisexifea_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrisexifea_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrisexifea_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -62,7 +62,7 @@
     {
         static bool r = RegisterEAGeneratorModule(
             Factory::ComposeEndsWithMatcher( EL() ),
-            "/usr/local/lib/ferris/plugins/eagenerators/libferrisexifea.so",
+            "/usr/lib/ferris/plugins/eagenerators/libferrisexifea.so",
             "exif",
             false, // dynamic
             true,  // hasState
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrisextractor_factory.cpp libferris-1.1.98/factories/eagenerators/libferrisextractor_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrisextractor_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrisextractor_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -40,7 +40,7 @@
     {
         static bool r = RegisterEAGeneratorModule(
             Factory::MakeAlwaysTrueMatcher(),
-            "/usr/local/lib/ferris/plugins/eagenerators/libferrisextractor.so",
+            "/usr/lib/ferris/plugins/eagenerators/libferrisextractor.so",
             "libextractor",
             false, // dyn
             false, // state
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrisid3_factory.cpp libferris-1.1.98/factories/eagenerators/libferrisid3_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrisid3_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrisid3_factory.cpp	2006-10-10 18:14:53.000000000 +0300
@@ -61,7 +61,7 @@
     {
         static bool r = RegisterEAGeneratorModule(
              Factory::ComposeEndsWithMatcher( EL() ),
-             "/usr/local/lib/ferris/plugins/eagenerators/libferrisid3.so",
+             "/usr/lib/ferris/plugins/eagenerators/libferrisid3.so",
             "ID3",
             false,
             true );
@@ -79,7 +79,7 @@
 //     {
 //         return new GModuleMatchedEAGeneratorFactory(
 //              Factory::ComposeEndsWithMatcher( EL() ),
-//              "/usr/local/lib/ferris/plugins/eagenerators/libferrisid3.so" );
+//              "/usr/lib/ferris/plugins/eagenerators/libferrisid3.so" );
 //     }
     
 // }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrisjasper_factory.cpp libferris-1.1.98/factories/eagenerators/libferrisjasper_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrisjasper_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrisjasper_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -44,7 +44,7 @@
     {
         const char* ext       = "jp2";
         bool        writable  = false;
-        const char* libname   = "/usr/local/lib/ferris/plugins/eagenerators/libferrisjasper.so";
+        const char* libname   = "/usr/lib/ferris/plugins/eagenerators/libferrisjasper.so";
         const char* shortname = "jasper";
         
         bool r = Context::RegisterImageEAGeneratorModule(
@@ -95,7 +95,7 @@
 //     {
 //         static bool r = RegisterEAGeneratorModule(
 //             Factory::ComposeEndsWithMatcher( EL() ),
-//             "/usr/local/lib/ferris/plugins/eagenerators/libferrisjasper.so",
+//             "/usr/lib/ferris/plugins/eagenerators/libferrisjasper.so",
 //             "jasper",
 //             false,
 //             true );
@@ -116,7 +116,7 @@
 //         {
 //             return new GModuleMatchedEAGeneratorFactory(
 //                 Factory::ComposeEndsWithMatcher( EL() ),
-//                 "/usr/local/lib/ferris/plugins/eagenerators/libferrisjasper.so" );
+//                 "/usr/lib/ferris/plugins/eagenerators/libferrisjasper.so" );
 //         }
     
 //     }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrisjpeg_factory.cpp libferris-1.1.98/factories/eagenerators/libferrisjpeg_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrisjpeg_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrisjpeg_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -45,7 +45,7 @@
     {
         const char* ext       = "jpeg";
         bool        writable  = false;
-        const char* libname   = "/usr/local/lib/ferris/plugins/eagenerators/libferrisjpeg.so";
+        const char* libname   = "/usr/lib/ferris/plugins/eagenerators/libferrisjpeg.so";
         const char* shortname = "jpeg";
         
         bool r = Context::RegisterImageEAGeneratorModule(
@@ -80,7 +80,7 @@
 //     {
 //         static bool r = RegisterEAGeneratorModule(
 //             Factory::ComposeEndsWithMatcher( EL() ),
-//             "/usr/local/lib/ferris/plugins/eagenerators/libferrisjpeg.so",
+//             "/usr/lib/ferris/plugins/eagenerators/libferrisjpeg.so",
 //             "jpeg",
 //             false,
 //             true );
@@ -99,13 +99,13 @@
 //         {
 //             return new GModuleMatchedEAGeneratorFactory(
 //                 Factory::ComposeEndsWithMatcher( EL() ),
-//                 "/usr/local/lib/ferris/plugins/eagenerators/libferrisjpeg.so" );
+//                 "/usr/lib/ferris/plugins/eagenerators/libferrisjpeg.so" );
 // //             return new GModuleMatchedEAGeneratorFactory(
 // //                 Factory::MakeAndMatcher(
 // //                     Factory::MakeEqualMatcher( "is-native", "1" ),
 // //                     Factory::ComposeEndsWithMatcher( EL() )
 // //                     ),
-// //                 "/usr/local/lib/ferris/plugins/eagenerators/libferrisjpeg.so" );
+// //                 "/usr/lib/ferris/plugins/eagenerators/libferrisjpeg.so" );
 //         }
     
 //     }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferriskde3metadata_factory.cpp libferris-1.1.98/factories/eagenerators/libferriskde3metadata_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferriskde3metadata_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferriskde3metadata_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -40,7 +40,7 @@
     {
         static bool r = RegisterEAGeneratorModule(
             Factory::MakeAlwaysTrueMatcher(),
-            "/usr/local/lib/ferris/plugins/eagenerators/libferriskde3metadata.so",
+            "/usr/lib/ferris/plugins/eagenerators/libferriskde3metadata.so",
             "kde3_metadata",
             false, // dyn
             true, // state
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrismpeg2_factory.cpp libferris-1.1.98/factories/eagenerators/libferrismpeg2_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrismpeg2_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrismpeg2_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -62,7 +62,7 @@
     {
         static bool r = RegisterEAGeneratorModule(
              Factory::ComposeEndsWithMatcher( EL() ),
-             "/usr/local/lib/ferris/plugins/eagenerators/libferrismpeg2.so",
+             "/usr/lib/ferris/plugins/eagenerators/libferrismpeg2.so",
             "mpeg2",
             false,
             false,
@@ -82,7 +82,7 @@
 //     {
 //         return new GModuleMatchedEAGeneratorFactory(
 //              Factory::ComposeEndsWithMatcher( EL() ),
-//              "/usr/local/lib/ferris/plugins/eagenerators/libferrismpeg2.so" );
+//              "/usr/lib/ferris/plugins/eagenerators/libferrismpeg2.so" );
 //     }
     
 // }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrismpeg3_factory.cpp libferris-1.1.98/factories/eagenerators/libferrismpeg3_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrismpeg3_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrismpeg3_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -71,7 +71,7 @@
     {
         return new GModuleMatchedEAGeneratorFactory(
              Factory::ComposeEndsWithMatcher( EL() ),
-             "/usr/local/lib/ferris/plugins/eagenerators/libferrismpeg3.so" );
+             "/usr/lib/ferris/plugins/eagenerators/libferrismpeg3.so" );
     }
     
 }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrispng_factory.cpp libferris-1.1.98/factories/eagenerators/libferrispng_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrispng_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrispng_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -44,7 +44,7 @@
     {
         const char* ext       = "png";
         bool        writable  = false;
-        const char* libname   = "/usr/local/lib/ferris/plugins/eagenerators/libferrispng.so";
+        const char* libname   = "/usr/lib/ferris/plugins/eagenerators/libferrispng.so";
         const char* shortname = "png";
         
         bool r = Context::RegisterImageEAGeneratorModule(
@@ -78,7 +78,7 @@
 //     {
 //         static bool r = RegisterEAGeneratorModule(
 //             Factory::ComposeEndsWithMatcher( EL() ),
-//             "/usr/local/lib/ferris/plugins/eagenerators/libferrispng.so",
+//             "/usr/lib/ferris/plugins/eagenerators/libferrispng.so",
 //             "png",
 //             false,
 //             true );
@@ -98,7 +98,7 @@
 //         {
 //             return new GModuleMatchedEAGeneratorFactory(
 //                 Factory::ComposeEndsWithMatcher( EL() ),
-//                 "/usr/local/lib/ferris/plugins/eagenerators/libferrispng.so" );
+//                 "/usr/lib/ferris/plugins/eagenerators/libferrispng.so" );
 //         }
     
 //     }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrisredland_factory.cpp libferris-1.1.98/factories/eagenerators/libferrisredland_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrisredland_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrisredland_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -62,7 +62,7 @@
         static bool r = RegisterEAGeneratorModule(
 //            Factory::ComposeEndsWithMatcher( EL() ),
             Factory::MakeAlwaysTrueMatcher(),
-            "/usr/local/lib/ferris/plugins/eagenerators/libferrisredland.so",
+            "/usr/lib/ferris/plugins/eagenerators/libferrisredland.so",
             "rdf",
             false, // dynamic
             true,  // hasState
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrisxineea_factory.cpp libferris-1.1.98/factories/eagenerators/libferrisxineea_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrisxineea_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrisxineea_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -64,7 +64,7 @@
 
         static bool r1 = RegisterEAGeneratorModule(
             getMatcher(),
-            "/usr/local/lib/ferris/plugins/eagenerators/libferrisxineea.so",
+            "/usr/lib/ferris/plugins/eagenerators/libferrisxineea.so",
             "xine_metadata",
             false, // dyn
             true,  // state
@@ -72,7 +72,7 @@
 
 
         bool        writable  = false;
-        const char* libname   = "/usr/local/lib/ferris/plugins/eagenerators/libferrisxineea.so";
+        const char* libname   = "/usr/lib/ferris/plugins/eagenerators/libferrisxineea.so";
         const char* shortname = "xine";
         bool ri1 = Context::RegisterImageEAGeneratorModule(
             "avi", writable, libname, shortname );
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libferrisxmp_factory.cpp libferris-1.1.98/factories/eagenerators/libferrisxmp_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libferrisxmp_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libferrisxmp_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -86,7 +86,7 @@
     {
         static bool r = RegisterEAGeneratorModule(
             getMatcher(),
-            "/usr/local/lib/ferris/plugins/eagenerators/libferrisxmp.so",
+            "/usr/lib/ferris/plugins/eagenerators/libferrisxmp.so",
             "xmp",
             false,
             true,
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libgimp_factory.cpp libferris-1.1.98/factories/eagenerators/libgimp_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libgimp_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libgimp_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -43,7 +43,7 @@
     {
         const char* ext       = "xcf";
         bool        writable  = false;
-        const char* libname   = "/usr/local/lib/ferris/plugins/eagenerators/libgimp.so";
+        const char* libname   = "/usr/lib/ferris/plugins/eagenerators/libgimp.so";
         const char* shortname = "gimp";
         
         bool r = Context::RegisterImageEAGeneratorModule(
@@ -75,7 +75,7 @@
 //     {
 //         return new GModuleMatchedEAGeneratorFactory(
 //              ::Factory::ComposeEndsWithMatcher( EL() ),
-//              "/usr/local/lib/ferris/plugins/eagenerators/libgimp.so" );
+//              "/usr/lib/ferris/plugins/eagenerators/libgimp.so" );
 //     }
     
 // }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libimlib2_factory.cpp libferris-1.1.98/factories/eagenerators/libimlib2_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libimlib2_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libimlib2_factory.cpp	2006-10-10 18:14:53.000000000 +0300
@@ -44,7 +44,7 @@
     {
         const char* ext       = "imlib2";
         bool        writable  = false;
-        const char* libname   = "/usr/local/lib/ferris/plugins/eagenerators/libimlib2.so";
+        const char* libname   = "/usr/lib/ferris/plugins/eagenerators/libimlib2.so";
         const char* shortname = "imlib2";
         
         bool r = Context::RegisterImageEAGeneratorModule(
@@ -77,7 +77,7 @@
 //     {
 //         static bool r = RegisterEAGeneratorModule(
 //              Factory::ComposeEndsWithMatcher( EL() ),
-//              "/usr/local/lib/ferris/plugins/eagenerators/libimlib2.so",
+//              "/usr/lib/ferris/plugins/eagenerators/libimlib2.so",
 //             "Imlib2",
 //             false,
 //             true );
@@ -96,7 +96,7 @@
 //     {
 //         return new GModuleMatchedEAGeneratorFactory(
 //              Factory::ComposeEndsWithMatcher( EL() ),
-//              "/usr/local/lib/ferris/plugins/eagenerators/libimlib2.so" );
+//              "/usr/lib/ferris/plugins/eagenerators/libimlib2.so" );
 //     }
     
 // }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libmagick_factory.cpp libferris-1.1.98/factories/eagenerators/libmagick_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libmagick_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libmagick_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -44,7 +44,7 @@
     {
         const char* ext       = "jp2";
         bool        writable  = true;
-        const char* libname   = "/usr/local/lib/ferris/plugins/eagenerators/libmagick.so";
+        const char* libname   = "/usr/lib/ferris/plugins/eagenerators/libmagick.so";
         const char* shortname = "magick";
         
         bool r = Context::RegisterImageEAGeneratorModule(
@@ -267,7 +267,7 @@
 //     {
 //         static bool r = RegisterEAGeneratorModule(
 //             Factory::ComposeEndsWithMatcher( EL() ),
-//             "/usr/local/lib/ferris/plugins/eagenerators/libmagick.so",
+//             "/usr/lib/ferris/plugins/eagenerators/libmagick.so",
 //             "ImageMagick",
 //             false,
 //             true );
@@ -285,7 +285,7 @@
 //         {
 //             return new GModuleMatchedEAGeneratorFactory(
 //                 Factory::ComposeEndsWithMatcher( EL() ),
-//                 "/usr/local/lib/ferris/plugins/eagenerators/libmagick.so" );
+//                 "/usr/lib/ferris/plugins/eagenerators/libmagick.so" );
 //         }
     
 //     }
diff -Nuar libferris-1.1.98-orig/factories/eagenerators/libxfsnative_factory.cpp libferris-1.1.98/factories/eagenerators/libxfsnative_factory.cpp
--- libferris-1.1.98-orig/factories/eagenerators/libxfsnative_factory.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/factories/eagenerators/libxfsnative_factory.cpp	2006-10-10 18:14:54.000000000 +0300
@@ -43,7 +43,7 @@
     {
         static bool r = RegisterEAGeneratorModule(
             Factory::MakeAlwaysTrueMatcher(),
-            "/usr/local/lib/ferris/plugins/eagenerators/libxfsnative.so",
+            "/usr/lib/ferris/plugins/eagenerators/libxfsnative.so",
             "KernelEA",
             true,
             true,
@@ -73,7 +73,7 @@
 //         {
 //             return new GModuleMatchedEAGeneratorFactory(
 //                 Factory::MakeAlwaysTrueMatcher(),
-//                 "/usr/local/lib/ferris/plugins/eagenerators/libxfsnative.so" );
+//                 "/usr/lib/ferris/plugins/eagenerators/libxfsnative.so" );
 //         }
     
 //     }
diff -Nuar libferris-1.1.98-orig/Ferris/EAIndexerMetaInterface.cpp libferris-1.1.98/Ferris/EAIndexerMetaInterface.cpp
--- libferris-1.1.98-orig/Ferris/EAIndexerMetaInterface.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/Ferris/EAIndexerMetaInterface.cpp	2006-10-10 18:14:52.000000000 +0300
@@ -969,7 +969,7 @@
         MetaEAIndexerInterface* CreateEAIndexerFromLibrary(
             const std::string& implnameraw )
         {
-            std::string libname = "/usr/local/lib/ferris/plugins/eaindexers/"
+            std::string libname = "/usr/lib/ferris/plugins/eaindexers/"
                 + implnameraw;
 
             typedef ::Loki::Functor< MetaEAIndexerInterface*,
diff -Nuar libferris-1.1.98-orig/Ferris/FullTextIndexerMetaInterface.cpp libferris-1.1.98/Ferris/FullTextIndexerMetaInterface.cpp
--- libferris-1.1.98-orig/Ferris/FullTextIndexerMetaInterface.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/Ferris/FullTextIndexerMetaInterface.cpp	2006-10-10 18:14:51.000000000 +0300
@@ -632,7 +632,7 @@
         MetaFullTextIndexerInterface* CreateFullTextIndexerFromLibrary(
             const std::string& implnameraw )
         {
-            std::string libname = "/usr/local/lib/ferris/plugins/fulltextindexers/"
+            std::string libname = "/usr/lib/ferris/plugins/fulltextindexers/"
                 + implnameraw;
 
             typedef ::Loki::Functor< MetaFullTextIndexerInterface*,
diff -Nuar libferris-1.1.98-orig/Ferris/LinkContextSchemeClients.cpp libferris-1.1.98/Ferris/LinkContextSchemeClients.cpp
--- libferris-1.1.98-orig/Ferris/LinkContextSchemeClients.cpp	2006-10-10 18:13:32.000000000 +0300
+++ libferris-1.1.98/Ferris/LinkContextSchemeClients.cpp	2006-10-10 18:16:48.000000000 +0300
@@ -61,7 +61,7 @@
 
         
         const std::string link_ctx_icons_name = "icons";
-        const std::string link_ctx_icons_url  = PREFIX + "/ferris/icons";
+        const std::string link_ctx_icons_url  = PREFIX + "/share/ferris/icons";
         typedef Loki::SingletonHolder<
             LinkContextSchemeVFS_RootContextDropper<
             link_ctx_icons_name, link_ctx_icons_url > > link_ctx_icons_singleton;
