diff -Nuar Ferris.orig/BranchContext_Clients.cpp Ferris/BranchContext_Clients.cpp
--- Ferris.orig/BranchContext_Clients.cpp	2006-10-02 11:53:51.000000000 +0300
+++ Ferris/BranchContext_Clients.cpp	2006-10-02 11:47:35.000000000 +0300
@@ -38,11 +38,11 @@
 #include <Ferris/Runner.hh>
 #include <Ferris/xfsutil.hh>
 
-#pragma GCC visibility push(hidden)
+//#pragma GCC visibility push(hidden)
 #define BOOST_SPIRIT_RULE_SCANNERTYPE_LIMIT 3
 #include <boost/spirit.hpp>
 using namespace boost::spirit;
-#pragma GCC visibility pop
+//#pragma GCC visibility pop
 
 
 
diff -Nuar Ferris.orig/ContextSetCompare.cpp Ferris/ContextSetCompare.cpp
--- Ferris.orig/ContextSetCompare.cpp	2006-10-02 11:53:51.000000000 +0300
+++ Ferris/ContextSetCompare.cpp	2006-10-02 11:46:58.000000000 +0300
@@ -33,7 +33,7 @@
 #include <Context_private.hh>
 #include <Trimming.hh>
 
-#pragma GCC visibility push(hidden)
+//#pragma GCC visibility push(hidden)
 #define BOOST_SPIRIT_RULE_SCANNERTYPE_LIMIT 3
 
 #include <boost/spirit.hpp>
@@ -43,7 +43,7 @@
 #include <boost/function.hpp>
 #include <boost/lambda/lambda.hpp>
 using namespace boost::lambda;
-#pragma GCC visibility pop
+//#pragma GCC visibility pop
 
 using namespace std;
 
diff -Nuar Ferris.orig/SpiritTime.cpp Ferris/SpiritTime.cpp
--- Ferris.orig/SpiritTime.cpp	2006-10-02 11:53:51.000000000 +0300
+++ Ferris/SpiritTime.cpp	2006-10-02 11:47:15.000000000 +0300
@@ -42,7 +42,7 @@
 #include <time.h>
 #include <config.h>
 
-#pragma GCC visibility push(hidden)
+//#pragma GCC visibility push(hidden)
 #define BOOST_SPIRIT_RULE_SCANNERTYPE_LIMIT 3
 
 #include <boost/spirit.hpp>
@@ -51,7 +51,7 @@
 #include <boost/function.hpp>
 #include <boost/lambda/lambda.hpp>
 using namespace boost::lambda;
-#pragma GCC visibility pop
+//#pragma GCC visibility pop
 
 /**********/
 /**********/
