--- libjava/Makefile.am.orig	2006-09-04 17:22:31.000000000 +0200
+++ libjava/Makefile.am	2006-09-04 17:24:34.000000000 +0200
@@ -68,7 +68,7 @@
 ## For now, only on native systems.  FIXME.
 if NATIVE
 bin_PROGRAMS = jv-convert gij grmic grmiregistry gcj-dbtool \
-	gappletviewer gjarsigner gkeytool
+	gappletviewer gjarsigner gkeytool gjar
 
 ## It is convenient to actually build and install the default database
 ## when gcj-dbtool is available.
@@ -785,6 +785,17 @@
 gjarsigner_DEPENDENCIES = libgcj-tools.la
 
 ## This is a dummy definition.
+gjar_SOURCES =
+gjar_LDFLAGS = --main=gnu.classpath.tools.jar.Main \
+	-rpath $(toolexeclibdir) -shared-libgcc $(THREADLDFLAGS)
+gjar_LINK = $(GCJLINK)
+## See jv_convert_LDADD.
+gjar_LDADD = -L$(here)/.libs libgcj-tools.la
+## Depend on the spec file to make sure it is up to date before
+## linking this program.
+gjar_DEPENDENCIES = libgcj-tools.la
+
+## This is a dummy definition.
 gkeytool_SOURCES =
 gkeytool_LDFLAGS = --main=gnu.classpath.tools.keytool.Main \
 	-rpath $(toolexeclibdir) -shared-libgcc $(THREADLDFLAGS)
--- libjava/Makefile.in.orig	2006-09-04 17:32:46.000000000 +0200
+++ libjava/Makefile.in	2006-09-04 17:35:34.000000000 +0200
@@ -46,7 +46,8 @@
 @NATIVE_TRUE@bin_PROGRAMS = jv-convert$(EXEEXT) gij$(EXEEXT) \
 @NATIVE_TRUE@	grmic$(EXEEXT) grmiregistry$(EXEEXT) \
 @NATIVE_TRUE@	gcj-dbtool$(EXEEXT) gappletviewer$(EXEEXT) \
-@NATIVE_TRUE@	gjarsigner$(EXEEXT) gkeytool$(EXEEXT)
+@NATIVE_TRUE@	gjarsigner$(EXEEXT) gkeytool$(EXEEXT) \
+@NATIVE_TRUE@	gjar$(EXEEXT)
 @BASH_JAR_TRUE@am__append_4 = scripts/jar
 @USING_GCC_TRUE@am__append_5 = $(WARNINGS)
 @USING_BOEHMGC_TRUE@am__append_6 = boehm.cc
@@ -391,6 +392,8 @@
 gen_from_JIS_LDADD = $(LDADD)
 am_gij_OBJECTS =
 gij_OBJECTS = $(am_gij_OBJECTS)
+am_gjar_OBJECTS =
+gjar_OBJECTS = $(am_gjar_OBJECTS)
 am_gjarsigner_OBJECTS =
 gjarsigner_OBJECTS = $(am_gjarsigner_OBJECTS)
 am_gkeytool_OBJECTS =
@@ -430,16 +433,17 @@
 	$(libgcj_la_SOURCES) $(libgcj_bc_la_SOURCES) \
 	$(libgij_la_SOURCES) $(libjvm_la_SOURCES) \
 	$(gappletviewer_SOURCES) $(gcj_dbtool_SOURCES) \
-	$(gen_from_JIS_SOURCES) $(gij_SOURCES) $(gjarsigner_SOURCES) \
-	$(gkeytool_SOURCES) $(grmic_SOURCES) $(grmiregistry_SOURCES) \
-	$(jv_convert_SOURCES)
+	$(gen_from_JIS_SOURCES) $(gij_SOURCES) $(gjar_SOURCES) \
+	$(gjarsigner_SOURCES) $(gkeytool_SOURCES) $(grmic_SOURCES) \
+	$(grmiregistry_SOURCES) $(jv_convert_SOURCES)
 DIST_SOURCES = $(lib_gnu_awt_xlib_la_SOURCES) \
 	$(libgcj_tools_la_SOURCES) $(am__libgcj_la_SOURCES_DIST) \
 	$(libgcj_bc_la_SOURCES) $(libgij_la_SOURCES) \
 	$(libjvm_la_SOURCES) $(gappletviewer_SOURCES) \
 	$(gcj_dbtool_SOURCES) $(am__gen_from_JIS_SOURCES_DIST) \
-	$(gij_SOURCES) $(gjarsigner_SOURCES) $(gkeytool_SOURCES) \
-	$(grmic_SOURCES) $(grmiregistry_SOURCES) $(jv_convert_SOURCES)
+	$(gij_SOURCES) $(gjar_SOURCES) $(gjarsigner_SOURCES) \
+	$(gkeytool_SOURCES) $(grmic_SOURCES) $(grmiregistry_SOURCES) \
+	$(jv_convert_SOURCES)
 RECURSIVE_TARGETS = all-recursive check-recursive dvi-recursive \
 	html-recursive info-recursive install-data-recursive \
 	install-exec-recursive install-info-recursive \
@@ -7359,6 +7363,13 @@
 gjarsigner_LINK = $(GCJLINK)
 gjarsigner_LDADD = -L$(here)/.libs libgcj-tools.la
 gjarsigner_DEPENDENCIES = libgcj-tools.la
+gjar_SOURCES = 
+gjar_LDFLAGS = --main=gnu.classpath.tools.jar.Main \
+	-rpath $(toolexeclibdir) -shared-libgcc $(THREADLDFLAGS)
+
+gjar_LINK = $(GCJLINK)
+gjar_LDADD = -L$(here)/.libs libgcj-tools.la
+gjar_DEPENDENCIES = libgcj-tools.la
 gkeytool_SOURCES = 
 gkeytool_LDFLAGS = --main=gnu.classpath.tools.keytool.Main \
 	-rpath $(toolexeclibdir) -shared-libgcc $(THREADLDFLAGS)
@@ -8048,6 +8059,9 @@
 gij$(EXEEXT): $(gij_OBJECTS) $(gij_DEPENDENCIES) 
 	@rm -f gij$(EXEEXT)
 	$(gij_LINK) $(gij_LDFLAGS) $(gij_OBJECTS) $(gij_LDADD) $(LIBS)
+gjar$(EXEEXT): $(gjar_OBJECTS) $(gjar_DEPENDENCIES) 
+	@rm -f gjar$(EXEEXT)
+	$(gjar_LINK) $(gjar_LDFLAGS) $(gjar_OBJECTS) $(gjar_LDADD) $(LIBS)
 gjarsigner$(EXEEXT): $(gjarsigner_OBJECTS) $(gjarsigner_DEPENDENCIES) 
 	@rm -f gjarsigner$(EXEEXT)
 	$(gjarsigner_LINK) $(gjarsigner_LDFLAGS) $(gjarsigner_OBJECTS) $(gjarsigner_LDADD) $(LIBS)
