diff -urN mc-4.6.1-pre1.org/syntax/spec.syntax mc-4.6.1-pre1/syntax/spec.syntax
--- mc-4.6.1-pre1.org/syntax/spec.syntax	2004-06-06 23:14:14.290331920 +0200
+++ mc-4.6.1-pre1/syntax/spec.syntax	2004-06-06 23:17:52.591145152 +0200
@@ -2,24 +2,24 @@
     keyword whole Auto\{Rr\}eq\{Pp\}rov: green
     keyword whole Auto\{Rr\}eq: green
     keyword whole Build\{Aa\}rch: green
+    keyword whole Build\{Cc\}onflicts: green
     keyword whole Build\{Pp\}re\{Rr\}eq: green
     keyword whole Build\{Rr\}oot: green
     keyword whole Build\{Rr\}equires: green
     keyword whole Conflicts: green
     keyword whole Copyright: white
     keyword whole Description: green    
     keyword whole Distribution: green    
     keyword whole Doc\{Dd\}ir: green    
     keyword whole Epoch: green
     keyword whole Exclude\{Aa\}rch: green
     keyword whole Exclusive\{Aa\}rch: green
     keyword whole Exclusive\{Oo\}\{Ss\}: green
     keyword whole Group: green
-    keyword whole Group(\[abcdefghijklmnopqrstuvwxyz\]): green
-    keyword whole Group(\[abcdefghijklmnopqrstuvwxyz\]_\[ABCDEFGHIJKLMNOPQRSTUVWXYZ\]): green
     keyword whole Icon: green
     keyword whole License: green
     keyword whole Name: green    
+    keyword whole NoSource\[0123456789\]: green
     keyword whole Obsoletes: green
     keyword whole Packager: green
     keyword whole Patch\[0123456789\]: green    
@@ -32,9 +32,6 @@
     keyword whole Root: green    
     keyword whole Source\[0123456789\]: green
     keyword whole Summary: green
-    keyword whole Summary(\[abcdefghijklmnopqrstuvwxyz\]): green
-    keyword whole Summary(\[abcdefghijklmnopqrstuvwxyz\]_\[ABCDEFGHIJKLMNOPQRSTUVWXYZ\]): green
-    keyword whole Summary(\[abcdefghijklmnopqrstuvwxyz\]_\[ABCDEFGHIJKLMNOPQRSTUVWXYZ\].\[ABCDEFGHIJKLMNOPQRSTUVWXYZ-1234567890\]): green
     keyword whole Vendor: green
     keyword whole Version: green
     keyword whole URL: green
@@ -80,6 +77,7 @@
     keyword wholeright $+ brightgreen
     keyword ftp://+ brightgreen
     keyword http://+ brightgreen
+    keyword cvs://+ brightgreen
 
     keyword whole missingok brightblue
     keyword whole noreplace brightblue
@@ -87,6 +85,189 @@
     keyword whole PATCH\[0123456789\] cyan
     keyword whole SOURCE\[0123456789\] cyan
 
+context Group( ): green
+	keyword whole af yellow
+	keyword whole az yellow
+	keyword whole bg yellow
+	keyword whole br yellow
+	keyword whole ca yellow
+	keyword whole ca_ES yellow
+	keyword whole cs yellow
+	keyword whole cs_CZ yellow
+	keyword whole da yellow
+	keyword whole da_DK yellow
+	keyword whole de yellow
+	keyword whole de_AT yellow
+	keyword whole de_DE yellow
+	keyword whole el yellow
+	keyword whole el_GR yellow
+	keyword whole en yellow
+	keyword whole en_GB yellow
+	keyword whole eo yellow
+	keyword whole es yellow
+	keyword whole es_ES yellow
+	keyword whole et yellow
+	keyword whole et_EE yellow
+	keyword whole eu yellow
+	keyword whole fi yellow
+	keyword whole fi_FI yellow
+	keyword whole fr yellow
+	keyword whole fr_FR yellow
+	keyword whole gl yellow
+	keyword whole gl_ES yellow
+	keyword whole gr yellow
+	keyword whole he yellow
+	keyword whole he_IL yellow
+	keyword whole hi yellow
+	keyword whole hr yellow
+	keyword whole hr_HR yellow
+	keyword whole hu yellow
+	keyword whole hu_HU yellow
+	keyword whole id yellow
+	keyword whole is yellow
+	keyword whole is_IS yellow
+	keyword whole it yellow
+	keyword whole it_IT yellow
+	keyword whole ja yellow
+	keyword whole ja_JP yellow
+	keyword whole ja_JP.SJIS yellow
+	keyword whole ko yellow
+	keyword whole ko_KR yellow
+	keyword whole lt yellow
+	keyword whole lt_LT yellow
+	keyword whole lv yellow
+	keyword whole ms yellow
+	keyword whole nb yellow
+	keyword whole nl yellow
+	keyword whole nl_NL yellow
+	keyword whole nn yellow
+	keyword whole nn_NO yellow
+	keyword whole no yellow
+	keyword whole no_NO yellow
+	keyword whole pl yellow
+	keyword whole pl_PL yellow
+	keyword whole pt yellow
+	keyword whole pt_BR yellow
+	keyword whole pt_PT yellow
+	keyword whole ro yellow
+	keyword whole ro_RO yellow
+	keyword whole ru yellow
+	keyword whole ru_RU yellow
+	keyword whole sk yellow
+	keyword whole sk_SK yellow
+	keyword whole sl yellow
+	keyword whole sl_SI yellow
+	keyword whole sr yellow
+	keyword whole sv yellow
+	keyword whole sv_SE yellow
+	keyword whole ta yellow
+	keyword whole th_TH yellow
+	keyword whole tr yellow
+	keyword whole tr_TR yellow
+	keyword whole uk yellow
+	keyword whole vi yellow
+	keyword whole wa yellow
+	keyword whole zh yellow
+	keyword whole zh_CN yellow
+	keyword whole zh_CN.GB2312 yellow
+	keyword whole zh_TW yellow
+	keyword whole zh_TW.Big5 yellow
+
+context Requires( ): green
+	keyword whole post yellow
+	keyword whole pre yellow
+	keyword whole postun yellow
+	keyword whole preun yellow
+	keyword , lightgray
+
+context Summary( ): green
+	keyword whole af yellow
+	keyword whole az yellow
+	keyword whole bg yellow
+	keyword whole br yellow
+	keyword whole ca yellow
+	keyword whole ca_ES yellow
+	keyword whole cs yellow
+	keyword whole cs_CZ yellow
+	keyword whole da yellow
+	keyword whole da_DK yellow
+	keyword whole de yellow
+	keyword whole de_AT yellow
+	keyword whole de_DE yellow
+	keyword whole el yellow
+	keyword whole el_GR yellow
+	keyword whole en yellow
+	keyword whole en_GB yellow
+	keyword whole eo yellow
+	keyword whole es yellow
+	keyword whole es_ES yellow
+	keyword whole et yellow
+	keyword whole et_EE yellow
+	keyword whole eu yellow
+	keyword whole fi yellow
+	keyword whole fi_FI yellow
+	keyword whole fr yellow
+	keyword whole fr_FR yellow
+	keyword whole gl yellow
+	keyword whole gl_ES yellow
+	keyword whole gr yellow
+	keyword whole he yellow
+	keyword whole he_IL yellow
+	keyword whole hi yellow
+	keyword whole hr yellow
+	keyword whole hr_HR yellow
+	keyword whole hu yellow
+	keyword whole hu_HU yellow
+	keyword whole id yellow
+	keyword whole is yellow
+	keyword whole is_IS yellow
+	keyword whole it yellow
+	keyword whole it_IT yellow
+	keyword whole ja yellow
+	keyword whole ja_JP yellow
+	keyword whole ja_JP.SJIS yellow
+	keyword whole ko yellow
+	keyword whole ko_KR yellow
+	keyword whole lt yellow
+	keyword whole lt_LT yellow
+	keyword whole lv yellow
+	keyword whole ms yellow
+	keyword whole nb yellow
+	keyword whole nl yellow
+	keyword whole nl_NL yellow
+	keyword whole nn yellow
+	keyword whole nn_NO yellow
+	keyword whole no yellow
+	keyword whole no_NO yellow
+	keyword whole pl yellow
+	keyword whole pl_PL yellow
+	keyword whole pt yellow
+	keyword whole pt_BR yellow
+	keyword whole pt_PT yellow
+	keyword whole ro yellow
+	keyword whole ro_RO yellow
+	keyword whole ru yellow
+	keyword whole ru_RU yellow
+	keyword whole sk yellow
+	keyword whole sk_SK yellow
+	keyword whole sl yellow
+	keyword whole sl_SI yellow
+	keyword whole sr yellow
+	keyword whole sv yellow
+	keyword whole sv_SE yellow
+	keyword whole ta yellow
+	keyword whole th_TH yellow
+	keyword whole tr yellow
+	keyword whole tr_TR yellow
+	keyword whole uk yellow
+	keyword whole vi yellow
+	keyword whole wa yellow
+	keyword whole zh yellow
+	keyword whole zh_CN yellow
+	keyword whole zh_CN.GB2312 yellow
+	keyword whole zh_TW yellow
+	keyword whole zh_TW.Big5 yellow
+
 context linestart # \n brown
 context %( ) brightblue
 
@@ -95,7 +95,7 @@
     keyword Revision*\n brightgreen
     keyword \*\s\{MWTFS\}*\n brightgreen
     keyword \s\s[*] brightgreen
-    keyword $Log*\n brightred
+    keyword $Log*$\n brightred

 #sections
 context exclusive linestart %build \n yellow
