diff -ur mc-4.6.2-pre1.orig/syntax/Syntax mc-4.6.2-pre1/syntax/Syntax
--- mc-4.6.2-pre1.orig/syntax/Syntax	2007-08-27 09:56:47.000000000 +0300
+++ mc-4.6.2-pre1/syntax/Syntax	2008-04-27 08:48:32.000000000 +0300
@@ -148,5 +148,8 @@
 file Don_t_match_me Mail\sfolder ^From\s
 include mail.syntax
 
+file .\*\\.(ebuild|eclass)$ Gentoo\sEbuild
+include ebuild.syntax
+
 file .\* unknown
 include unknown.syntax
