diff -Nurwd ./mc-4.6.2~pre1/edit/editcmd.c ./mc-4.6.2~pre1.patched/edit/editcmd.c
--- ./mc-4.6.2~pre1/edit/editcmd.c	2008-02-12 19:58:24.000000000 +0100
+++ ./mc-4.6.2~pre1.patched/edit/editcmd.c	2008-02-12 19:57:11.000000000 +0100
@@ -243,7 +243,7 @@
     this_save_mode = option_save_mode;
     if (this_save_mode != EDIT_QUICK_SAVE) {
 	if (!vfs_file_is_local (filename) ||
-	    (fd = mc_open (filename, O_WRONLY | O_BINARY)) == -1) {
+	    (fd = mc_open (filename, O_RDONLY | O_BINARY)) == -1) {
 	    /*
 	     * The file does not exists yet, so no safe save or
 	     * backup are necessary.
