diff -Nurwd mc-4.6.2~pre1.orig/vfs/fish.c mc-4.6.2~pre1/vfs/fish.c
--- mc-4.6.2~pre1.orig/vfs/fish.c	2008-02-27 16:52:15.000000000 +0100
+++ mc-4.6.2~pre1/vfs/fish.c	2008-02-27 16:56:23.000000000 +0100
@@ -618,16 +618,28 @@
     g_free (name);
     name = quoted_name;
     fh->u.fish.append = 0;
+
+    /*
+     * Check whether the remote file is readable by using `dd' to copy 
+     * a single byte from the remote file to /dev/null. If `dd' completes
+     * with exit status of 0 use `cat' to send the file contents to the
+     * standard output (i.e. over the network).
+     */
     offset = fish_command (me, FH_SUPER, WANT_STRING,
 		"#RETR /%s\n"
+		"if dd if=/%s of=/dev/null bs=1 count=1 2>/dev/null ;\n"
+		"then\n"
 		"ls -ln /%s 2>/dev/null | (\n"
 		  "read p l u g s r\n"
 		  "echo \"$s\"\n"
 		")\n"
 		"echo '### 100'\n"
 		"cat /%s\n"
-		"echo '### 200'\n", 
-		name, name, name );
+		"echo '### 200'\n"
+		"else\n"
+		"echo '### 500'\n" 
+		"fi\n",
+		name, name, name, name );
     g_free (name);
     if (offset != PRELIM) ERRNOR (E_REMOTE, 0);
     fh->linear = LS_LINEAR_OPEN;
