diff -ur a/wengophone/src/presentation/qt/QtWengoStyle.h b/wengophone/src/presentation/qt/QtWengoStyle.h
--- a/wengophone/src/presentation/qt/QtWengoStyle.h	2007-02-21 15:13:12.000000000 +0100
+++ b/wengophone/src/presentation/qt/QtWengoStyle.h	2008-03-18 22:46:26.111284321 +0100
@@ -24,10 +24,10 @@
 
 #ifdef OS_MACOSX
 	#include <QtGui/QMacStyle>
-	class SystemStyle : public QMacStyle { };
+	typedef QMacStyle BaseStyle;
 #else
-	#include <QtGui/QStyle>
-	class SystemStyle : public QStyle { };
+	#include <QtGui/QCommonStyle>
+	typedef QCommonStyle BaseStyle;
 #endif
 
 /**
@@ -39,7 +39,7 @@
  *
  * @author Tanguy Krotoff
  */
-class QtWengoStyle : public SystemStyle {
+class QtWengoStyle : public BaseStyle {
 public:
 
 	QtWengoStyle();
