Index: lib/sound/sound.c
===================================================================
--- lib/sound/sound.c	(revision 218)
+++ lib/sound/sound.c	(working copy)
@@ -170,6 +170,10 @@
 		return;
 
 	alutInit(NULL, NULL);
+	alGetError();
+	ALCcontext *pContext = alcGetCurrentContext();
+	ALCdevice *pDevice = alcGetContextsDevice(pContext);
+	log_message(TRACE, "Initializing ALCdevice: %s ", alcGetString(pDevice, ALC_DEVICE_SPECIFIER));
 }
 
 void sound_uninit(void)
@@ -186,9 +190,9 @@
 	log_message(TRACE, "Play sound sample %s (use OpenAL library)", path);
 
 	ALuint AlutBuffer = alutCreateBufferFromFile(path);
+	free(path);
 	if (!AlutBuffer)
 	{
-		free(path);
 		log_message(ERROR, "Failed to create OpenAL buffer");
 		return NULL;
 	}
@@ -197,16 +201,21 @@
 	alGenSources(1, &AlutSource);
 	alSourcei(AlutSource, AL_BUFFER, AlutBuffer);
 	alSourcePlay(AlutSource);
+	
+	ALint result;
+	alGetSourcei(AlutSource, AL_SOURCE_STATE, &result);
+	if ( result == AL_PLAYING) {
+		sleep(1);
+		alGetSourcei(AlutSource, AL_SOURCE_STATE, &result);
+	}
 
 	do
 		alDeleteSources(1, &AlutSource);
 	while (alGetError() != AL_NO_ERROR);
-
 	do
 		alDeleteBuffers(1, &AlutBuffer);
 	while (alGetError() != AL_NO_ERROR);
 
-	free(path);
 	return NULL;
 }
 
