diff -Nur glibc-2.3.5.orig/Makerules glibc-2.3.5/Makerules
--- glibc-2.3.5.orig/Makerules	2005-07-27 17:03:47.631632561 +0200
+++ glibc-2.3.5/Makerules	2005-07-27 17:04:34.594100229 +0200
@@ -605,6 +605,7 @@
 			 $(common-objpfx)libc_pic.os \
 			 $(elfobjdir)/sofini.os \
 			 $(elfobjdir)/interp.os $(elfobjdir)/ld.so \
+			 $(common-objpfx)debug/stack_chk_fail_local.oS \
 			 $(common-objpfx)shlib.lds
 	$(build-shlib)
 ifeq ($(versioning),yes)
diff -Nur glibc-2.3.5.orig/elf/Makefile glibc-2.3.5/elf/Makefile
--- glibc-2.3.5.orig/elf/Makefile	2005-07-27 15:55:26.978635224 +0200
+++ glibc-2.3.5/elf/Makefile	2005-07-27 15:36:08.364203540 +0200
@@ -383,6 +383,7 @@
 CFLAGS-cache.c = $(SYSCONF-FLAGS)
 
 CPPFLAGS-.os += $(if $(filter $(@F),$(patsubst %,%.os,$(all-rtld-routines))),-DNOT_IN_libc=1 -DIS_IN_rtld=1)
+CFLAGS-.os += $(if $(filter $(@F),$(patsubst %,%.os,$(all-rtld-routines))),-fno-stack-protector)
 
 test-modules = $(addprefix $(objpfx),$(addsuffix .so,$(strip $(modules-names))))
 generated += $(addsuffix .so,$(strip $(modules-names)))
diff -Nur glibc-2.3.5.orig/elf/rtld-Rules glibc-2.3.5/elf/rtld-Rules
--- glibc-2.3.5.orig/elf/rtld-Rules	2005-07-27 15:55:26.983633997 +0200
+++ glibc-2.3.5/elf/rtld-Rules	2005-07-27 15:34:40.293839104 +0200
@@ -121,7 +121,7 @@
 
 # This here is the whole point of all the shenanigans.
 CPPFLAGS-rtld := -DNOT_IN_libc=1 -DIS_IN_rtld=1
-CFLAGS-rtld := # blah
+CFLAGS-rtld := -fno-stack-protector
 
 
 endif
diff -Nur glibc-2.3.5.orig/csu/Makefile glibc-2.3.5/csu/Makefile
--- glibc-2.3.5.orig/csu/Makefile	2005-07-27 17:08:15.000000000 +0200
+++ glibc-2.3.5/csu/Makefile	2005-07-27 18:33:05.206235275 +0200
@@ -90,7 +90,7 @@
 
 ifeq ($(have-initfini),yes)
 
-CPPFLAGS += -DHAVE_INITFINI
+CPPFLAGS += -fno-stack-protector -DHAVE_INITFINI
 
 # These are the special initializer/finalizer files.  They are always the
 # first and last file in the link.  crti.o ... crtn.o define the global
diff -Nur glibc-2.3.5.orig/nptl/Makefile glibc-2.3.5/nptl/Makefile
--- glibc-2.3.5.orig/nptl/Makefile	2005-07-27 17:08:18.000000000 +0200
+++ glibc-2.3.5/nptl/Makefile	2005-07-27 18:32:25.096095270 +0200
@@ -132,10 +132,10 @@
 # even with asynchronous unwind tables.
 
 # init.c contains sigcancel_handler().
-CFLAGS-init.c = -fexceptions -fasynchronous-unwind-tables
+CFLAGS-init.c = -fexceptions -fasynchronous-unwind-tables -fno-stack-protector
 # The unwind code itself,
-CFLAGS-unwind.c = -fexceptions
-CFLAGS-unwind-forcedunwind.c = -fexceptions -fasynchronous-unwind-tables
+CFLAGS-unwind.c = -fexceptions -fno-stack-protector
+CFLAGS-unwind-forcedunwind.c = -fexceptions -fasynchronous-unwind-tables -fno-stack-protector
 
 # The following three functions must be async-cancel safe.
 CFLAGS-pthread_cancel.c = -fexceptions -fasynchronous-unwind-tables
