diff --git a/filter/source/config/fragments/filters/MS_Excel_5_0_95.xcu b/filter/source/config/fragments/filters/MS_Excel_5_0_95.xcu
index a6106a1..3bceb75 100644
--- filter/source/config/fragments/filters/MS_Excel_5_0_95.xcu
+++ filter/source/config/fragments/filters/MS_Excel_5_0_95.xcu
@@ -1,5 +1,5 @@
 <node oor:name="MS Excel 5.0/95" oor:op="replace">
-	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN PREFERRED</value></prop>
+	<prop oor:name="Flags"><value>IMPORT ALIEN PREFERRED</value></prop>
 	<prop oor:name="UIComponent"/>
 	<prop oor:name="FilterService"/>
 	<prop oor:name="UserData"/>
diff --git a/filter/source/config/fragments/filters/MS_Excel_5_0_95_Vorlage_Template.xcu b/filter/source/config/fragments/filters/MS_Excel_5_0_95_Vorlage_Template.xcu
index 5d66e9c..fc2421b 100644
--- filter/source/config/fragments/filters/MS_Excel_5_0_95_Vorlage_Template.xcu
+++ filter/source/config/fragments/filters/MS_Excel_5_0_95_Vorlage_Template.xcu
@@ -1,5 +1,5 @@
 <node oor:name="MS Excel 5.0/95 Vorlage/Template" oor:op="replace">
-	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN TEMPLATE TEMPLATEPATH</value></prop>
+	<prop oor:name="Flags"><value>IMPORT ALIEN TEMPLATE TEMPLATEPATH</value></prop>
 	<prop oor:name="UIComponent"/>
 	<prop oor:name="FilterService"/>
 	<prop oor:name="UserData"/>
diff --git a/filter/source/config/fragments/filters/MS_Excel_95.xcu b/filter/source/config/fragments/filters/MS_Excel_95.xcu
index dd792a1..4dbe69f 100644
--- filter/source/config/fragments/filters/MS_Excel_95.xcu
+++ filter/source/config/fragments/filters/MS_Excel_95.xcu
@@ -1,5 +1,5 @@
 <node oor:name="MS Excel 95" oor:op="replace">
-	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN PREFERRED</value></prop>
+	<prop oor:name="Flags"><value>IMPORT ALIEN PREFERRED</value></prop>
 	<prop oor:name="UIComponent"/>
 	<prop oor:name="FilterService"/>
 	<prop oor:name="UserData"/>
diff --git a/filter/source/config/fragments/filters/MS_Excel_95_Vorlage_Template.xcu b/filter/source/config/fragments/filters/MS_Excel_95_Vorlage_Template.xcu
index c04ddc6..7342d21 100644
--- filter/source/config/fragments/filters/MS_Excel_95_Vorlage_Template.xcu
+++ filter/source/config/fragments/filters/MS_Excel_95_Vorlage_Template.xcu
@@ -1,5 +1,5 @@
 <node oor:name="MS Excel 95 Vorlage/Template" oor:op="replace">
-	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN TEMPLATE TEMPLATEPATH</value></prop>
+	<prop oor:name="Flags"><value>IMPORT ALIEN TEMPLATE TEMPLATEPATH</value></prop>
 	<prop oor:name="UIComponent"/>
 	<prop oor:name="FilterService"/>
 	<prop oor:name="UserData"/>
diff --git a/filter/source/config/fragments/filters/MS_WinWord_6_0.xcu b/filter/source/config/fragments/filters/MS_WinWord_6_0.xcu
index 19be31c..3acc4ff 100644
--- filter/source/config/fragments/filters/MS_WinWord_6_0.xcu
+++ filter/source/config/fragments/filters/MS_WinWord_6_0.xcu
@@ -1,5 +1,5 @@
 	<node oor:name="MS WinWord 6.0" oor:op="replace">
-		<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN</value></prop>
+		<prop oor:name="Flags"><value>IMPORT ALIEN</value></prop>
 		<prop oor:name="UIComponent"/>
 		<prop oor:name="FilterService"/>
 		<prop oor:name="UserData"><value>CWW6</value></prop>
diff --git a/filter/source/config/fragments/filters/MS_Word_95.xcu b/filter/source/config/fragments/filters/MS_Word_95.xcu
index 2bc919b..b2d867a 100644
--- filter/source/config/fragments/filters/MS_Word_95.xcu
+++ filter/source/config/fragments/filters/MS_Word_95.xcu
@@ -1,5 +1,5 @@
 	<node oor:name="MS Word 95" oor:op="replace">
-		<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN</value></prop>
+		<prop oor:name="Flags"><value>IMPORT ALIEN</value></prop>
 		<prop oor:name="UIComponent"/>
 		<prop oor:name="FilterService"/>
 		<prop oor:name="UserData"><value>CWW6</value></prop>
