--- sc/source/ui/vba/vbaaxis.cxx.orig	2007-05-26 23:43:22.000000000 +0400
+++ sc/source/ui/vba/vbaaxis.cxx	2007-05-26 23:43:49.000000000 +0400
@@ -39,6 +39,8 @@
 #include <org/openoffice/excel/XlScaleType.hpp>
 #include "vbaaxistitle.hxx"
 #include "vbachart.hxx"
+#undef MIN
+#undef MAX
 using namespace ::com::sun::star;
 using namespace ::org::openoffice;
 using namespace ::org::openoffice::excel::XlAxisCrosses;
