=== modified file 'solenv/bin/modules/installer/globals.pm'
--- solenv/bin/modules/installer/globals.pm	2008-09-25 12:35:58 +0000
+++ solenv/bin/modules/installer/globals.pm	2008-09-25 12:36:22 +0000
@@ -39,7 +39,7 @@
 {	
 	$prog="make_installer.pl";
 
-	@noMSLocaleLangs = ( "br", "bs", "dz", "gu", "km", "nr", "ns", "rw", "ss", "st", "tg", "ts", "tn", "ve", "xh", "zu" , "ne" , "bn", "bn-BD", "bn-IN", "lo", "cy" , "ku" , "as-IN" , "te-IN" , "ml-IN" , "mr-IN", "ur-IN", "ta-IN", "or-IN" , "ti-ER" , "eo" , "ka" ,"ga" , "gl" ,"uk", "gd" );
+	@noMSLocaleLangs = ( "br", "bs", "dz", "gu", "km", "nr", "ns", "rw", "ss", "st", "tg", "ts", "tn", "ve", "xh", "zu" , "ne" , "bn", "bn-BD", "bn-IN", "lo", "cy" , "ku" , "as-IN" , "te-IN" , "ml-IN" , "mr-IN", "ur-IN", "ta-IN", "or-IN" , "ti-ER" , "eo" , "ka" ,"ga" , "gl" ,"gd" );
 	@items_at_modules = ("Files", "Dirs", "Unixlinks");
 	@asianlanguages = ("ja", "ko", "zh-CN", "zh-TW");
 	

