--- officecfg/registry/schema/org/openoffice/Office/Writer.xcs.orig	2005-11-17 21:35:46.000000000 +0300
+++ officecfg/registry/schema/org/openoffice/Office/Writer.xcs	2005-12-09 19:33:17.589871727 +0300
@@ -1741,6 +1741,7 @@
 					<desc>Specifies the font type to be used for the Standard paragraph style.</desc>
 					<label>Standard</label>
 				</info>
+				<value>Arial</value>
 			</prop>
             <prop oor:name="StandardHeight" oor:type="xs:int">
                 <!-- UIHints: Tools - Options - Text document  Default Fonts (Western) -->
@@ -1749,6 +1750,7 @@
                     <desc>Specifies the font size to be used for the Standard paragraph style.</desc>
                     <label>StandardHeight</label>
                 </info>
+				<value>353</value>
             </prop>
             <prop oor:name="Heading" oor:type="xs:string">
 				<!-- OldPath: Writer/Default_Fonts -->
@@ -1759,6 +1761,7 @@
 					<desc>Specifies the font type to be used for the paragraph&apos;s headings.</desc>
 					<label>Heading</label>
 				</info>
+				<value>Arial</value>
 			</prop>
             <prop oor:name="HeadingHeight" oor:type="xs:int">
                 <!-- UIHints: Tools - Options - Text document  Default Fonts (Western) -->
@@ -1777,6 +1780,7 @@
 					<desc>Specifies the font to be used in the paragraph styles for lists and numbering.</desc>
 					<label>List</label>
 				</info>
+				<value>Arial</value>
 			</prop>
             <prop oor:name="ListHeight" oor:type="xs:int">
                 <!-- UIHints: Tools - Options - Text document  Default Fonts (Western) -->
@@ -1785,6 +1789,7 @@
                     <desc>Specifies the font size to be used in the paragraph styles for lists and numberings.</desc>
                     <label>ListHeight</label>
                 </info>
+				<value>353</value>
             </prop>
             <prop oor:name="Caption" oor:type="xs:string">
 				<!-- OldPath: Writer/Default_Fonts -->
@@ -1795,6 +1800,7 @@
 					<desc>Specifies the font to be used for the labels of images and tables.</desc>
 					<label>Caption</label>
 				</info>
+				<value>Arial</value>
 			</prop>
             <prop oor:name="CaptionHeight" oor:type="xs:int">
                 <!-- UIHints: Tools - Options - Text document  Default Fonts (Western) -->
@@ -1803,6 +1809,7 @@
                     <desc>Specifies the font size to be used for the labels of images and tables.</desc>
                     <label>CaptionHeight</label>
                 </info>
+				<value>353</value>
             </prop>
             <prop oor:name="Index" oor:type="xs:string">
 				<!-- OldPath: Writer/Default_Fonts -->
@@ -1813,6 +1820,7 @@
 					<desc>Specifies the fonts to be applied to the paragraph styles for user-defined indexes, alphabetical indexes, tables of contents and to all of the paragraph styles derived from this.</desc>
 					<label>Index</label>
 				</info>
+				<value>Arial</value>
 			</prop>
             <prop oor:name="IndexHeight" oor:type="xs:int">
                 <!-- UIHints: Tools - Options - Text document  Default Fonts (Western) -->
@@ -1821,6 +1829,7 @@
                     <desc>Specifies the font size to be applied to the paragraph styles for user-defined indexes, alphabetical indexes, tables of contents and to all of the paragraph styles derived from this.</desc>
                     <label>IndexHeight</label>
                 </info>
+				<value>353</value>
             </prop>
             <prop oor:name="Document" oor:type="xs:boolean">
 				<!-- OldPath: Writer/Default_Fonts -->
