diff -urN extras.orig/source/templates/educate/lang/delzip extras/source/templates/educate/lang/delzip
--- extras.orig/source/templates/educate/lang/delzip	1970-01-01 03:00:00.000000000 +0300
+++ extras/source/templates/educate/lang/delzip	2008-08-01 09:23:42.000000000 +0400
@@ -0,0 +1 @@
+ECHO is OFF
diff -urN extras.orig/source/templates/educate/lang/makefile.mk extras/source/templates/educate/lang/makefile.mk
--- extras.orig/source/templates/educate/lang/makefile.mk	1970-01-01 03:00:00.000000000 +0300
+++ extras/source/templates/educate/lang/makefile.mk	2008-09-03 09:10:06.000000000 +0400
@@ -0,0 +1,53 @@
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: makefile.mk,v $
+#
+# $Revision: 1.3 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+PRJ=..$/..$/..$/..
+
+
+
+PRJNAME=extras
+TARGET=templateseducate
+
+.INCLUDE : settings.mk
+
+.INCLUDE : $(PRJ)$/util$/target.pmk
+
+
+
+ZIP1TARGET      = $(TEMPLATE_EDUCATE_WRITER_TARGET)
+
+ZIP1LIST	= $(LANGDIR)$/*
+
+
+
+.INCLUDE : target.mk
+
+
+
diff -urN extras.orig/source/templates/misc/lang/delzip extras/source/templates/misc/lang/delzip
--- extras.orig/source/templates/misc/lang/delzip	1970-01-01 03:00:00.000000000 +0300
+++ extras/source/templates/misc/lang/delzip	2008-08-01 09:23:42.000000000 +0400
@@ -0,0 +1 @@
+ECHO is OFF
diff -urN extras.orig/source/templates/misc/lang/makefile.mk extras/source/templates/misc/lang/makefile.mk
--- extras.orig/source/templates/misc/lang/makefile.mk	1970-01-01 03:00:00.000000000 +0300
+++ extras/source/templates/misc/lang/makefile.mk	2008-09-03 09:14:03.000000000 +0400
@@ -0,0 +1,53 @@
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: makefile.mk,v $
+#
+# $Revision: 1.3 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+PRJ=..$/..$/..$/..
+
+
+
+PRJNAME=extras
+TARGET=templatesmisc
+
+.INCLUDE : settings.mk
+
+.INCLUDE : $(PRJ)$/util$/target.pmk
+
+
+
+ZIP1TARGET      = $(TEMPLATE_MISC_WRITER_TARGET)
+
+ZIP1LIST	= $(LANGDIR)$/*
+
+
+
+.INCLUDE : target.mk
+
+
+
diff -urN extras.orig/source/templates/officor/lang/delzip extras/source/templates/officorr/lang/delzip
--- extras.orig/source/templates/officorr/lang/delzip	1970-01-01 03:00:00.000000000 +0300
+++ extras/source/templates/officorr/lang/delzip	2008-08-01 09:23:42.000000000 +0400
@@ -0,0 +1 @@
+ECHO is OFF
diff -urN extras.orig/source/templates/officorr/lang/makefile.mk extras/source/templates/officor/lang/makefile.mk
--- extras.orig/source/templates/officorr/lang/makefile.mk	1970-01-01 03:00:00.000000000 +0300
+++ extras/source/templates/officorr/lang/makefile.mk	2008-09-03 09:14:29.000000000 +0400
@@ -0,0 +1,53 @@
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: makefile.mk,v $
+#
+# $Revision: 1.3 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+PRJ=..$/..$/..$/..
+
+
+
+PRJNAME=extras
+TARGET=templatesofficorr
+
+.INCLUDE : settings.mk
+
+.INCLUDE : $(PRJ)$/util$/target.pmk
+
+
+
+ZIP1TARGET      = $(TEMPLATE_OFFICORR_WRITER_TARGET)
+
+ZIP1LIST	= $(LANGDIR)$/*
+
+
+
+.INCLUDE : target.mk
+
+
+
diff -urN extras.orig/source/templates/offimisc/lang/delzip extras/source/templates/offimisc/lang/delzip
--- extras.orig/source/templates/offimisc/lang/delzip	1970-01-01 03:00:00.000000000 +0300
+++ extras/source/templates/offimisc/lang/delzip	2008-08-01 09:23:42.000000000 +0400
@@ -0,0 +1 @@
+ECHO is OFF
diff -urN extras.orig/source/templates/offimisc/lang/makefile.mk extras/source/templates/offimisc/lang/makefile.mk
--- extras.orig/source/templates/offimisc/lang/makefile.mk	1970-01-01 03:00:00.000000000 +0300
+++ extras/source/templates/offimisc/lang/makefile.mk	2008-09-03 09:14:44.000000000 +0400
@@ -0,0 +1,53 @@
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: makefile.mk,v $
+#
+# $Revision: 1.3 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+PRJ=..$/..$/..$/..
+
+
+
+PRJNAME=extras
+TARGET=templatesoffimisc
+
+.INCLUDE : settings.mk
+
+.INCLUDE : $(PRJ)$/util$/target.pmk
+
+
+
+ZIP1TARGET      = $(TEMPLATE_OFFIMISC_WRITER_TARGET)
+
+ZIP1LIST	= $(LANGDIR)$/*
+
+
+
+.INCLUDE : target.mk
+
+
+
--- extras/prj/build.lst.orig	2008-09-03 09:11:07.000000000 +0400
+++ extras/prj/build.lst	2008-09-03 09:21:09.000000000 +0400
@@ -18,6 +18,10 @@
 er	extras\source\palettes					nmake	-	all	er_palet NULL
 er	extras\source\symbols					nmake	-	all	er_symb NULL
 er      extras\source\templates\layout\lang                     nmake   -       all     er_tlal NULL
+er      extras\source\templates\educate\lang                     nmake   -       all     er_tled NULL
+er      extras\source\templates\officorr\lang                     nmake   -       all     er_tloc NULL
+er      extras\source\templates\misc\lang                     nmake   -       all     er_tlms NULL
+er      extras\source\templates\offimisc\lang                     nmake   -       all     er_tlom NULL
 er      extras\source\templates\presnt\lang                     nmake   -       all     er_tprel NULL
 er      extras\source\templates\wizard\agenda\lang              nmake   -       all     er_twagl NULL
 er      extras\source\templates\wizard\bitmap                   nmake   -       all     er_twbmp NULL
--- scp2/source/ooo/file_extra_ooo.scp.orig	2008-09-03 09:53:26.000000000 +0400
+++ scp2/source/ooo/file_extra_ooo.scp	2008-09-03 09:56:31.000000000 +0400
@@ -360,6 +360,34 @@
     EXTRA_ALL_LANG(tpllayoutimpr);
 End
 
+File gid_File_Extra_Tpleduwrit_Lang
+    Dir = gid_Dir_Template_Educate;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(tpleduwrit);
+End
+
+File gid_File_Extra_Tplmiscwrit_Lang
+    Dir = gid_Dir_Template_Misc;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(tplmiscwrit);
+End
+
+File gid_File_Extra_Tplofficorrwrit_Lang
+    Dir = gid_Dir_Template_Officorr;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(tplofficorrwrit);
+End
+
+File gid_File_Extra_Tploffimiscwrit_Lang
+    Dir = gid_Dir_Template_Offimisc;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(tploffimiscwrit);
+End
+
 File gid_File_Extra_Tplpresntimpr_Lang
     Dir = gid_Dir_Template_Presnt;
     TXT_FILE_BODY;
--- scp2/source/ooo/module_lang_template.scp.orig	2008-09-03 10:02:09.000000000 +0400
+++ scp2/source/ooo/module_lang_template.scp	2008-09-03 10:02:34.000000000 +0400
@@ -36,6 +36,10 @@
     Files = (gid_File_Extra_Autotextshare_Lang,
              gid_File_Extra_Palettes_Lang,
              gid_File_Extra_Tpllayoutimpr_Lang,
+             gid_File_Extra_Tpleduwrit_Lang,
+             gid_File_Extra_Tplmiscwrit_Lang,
+             gid_File_Extra_Tplofficorrwrit_Lang,
+             gid_File_Extra_Tploffimiscwrit_Lang,
              gid_File_Extra_Tplpresntimpr_Lang,
              gid_File_Extra_Tplwizagenda_Lang,
              gid_File_Extra_Tplwizbitmap,
