--- officecfg/registry/schema/org/openoffice/Office/Common.xcs.orig	2008-03-04 02:26:09.000000000 +0300
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2008-03-04 02:26:35.000000000 +0300
@@ -4877,7 +4877,7 @@
 								</info>
 							</enumeration>
 						</constraints>
-						<value>4</value>
+						<value>0</value>
 					</prop>
 					<prop oor:name="Basic" oor:type="xs:boolean">
 						<!-- OldPath: Browser/HTML/Export -->
