--- solenv/bin/modules/installer/globals.pm.orig        2008-08-27 23:38:16.000000000 +0400
+++ solenv/bin/modules/installer/globals.pm     2008-08-27 23:38:47.000000000 +0400
@@ -309,7 +309,8 @@
        $one_cab_file = 0;
        $fix_number_of_cab_files = 1;
        $cab_file_per_component = 0;
-       $cabfilecompressionlevel = 2;
+       $cabfilecompressionlevel = 7;
+       $cabfilecompressionmemory = 21;
        $number_of_cabfiles = 1;        # only for $fix_number_of_cab_files = 1
        $include_cab_in_msi = 0;
        $use_packages_for_cabs = 0;
--- solenv/bin/modules/installer/windows/msiglobal.pm.orig      2008-08-27 23:42:04.000000000 +0400
+++ solenv/bin/modules/installer/windows/msiglobal.pm   2008-08-27 23:42:31.000000000 +0400
@@ -70,6 +70,8 @@
        push(@{$ddffileref} ,$oneline);
        $oneline = ".Set CompressionLevel=$installer::globals::cabfilecompressionlevel\n";
        push(@{$ddffileref} ,$oneline);
+       $oneline = ".Set CompressionMemory=$installer::globals::cabfilecompressionmemory\n";
+       push(@{$ddffileref} ,$oneline);
        $oneline = ".Set Cabinet=ON\n";
        push(@{$ddffileref} ,$oneline);
        $oneline = ".Set DiskDirectoryTemplate=" . $installdir . "\n";
