--- instsetoo_native/util/makefile.mk.orig	2006-04-10 11:27:07.004875742 +0400
+++ instsetoo_native/util/makefile.mk	2006-04-10 11:28:03.629552386 +0400
@@ -109,7 +109,10 @@
 .ELSE			# "$(GUI)"!="WNT" && "$(EPM)"=="NO" && "$(USE_PACKAGER)"==""
 .IF "$(UPDATER)"=="" || "$(USE_PACKAGER)"==""
 .IF "$(BUILD_TYPE)"=="$(BUILD_TYPE:s/ODK//)"
-ALLTAR : openoffice
+ALLTAR : openoffice ooolanguagepack
+.IF "$(GUI)"=="WNT"
+ALLTAR : openoffice ooolanguagepack openofficewithjre
+.ENDIF
 .ELSE
 ALLTAR : openoffice sdkoo_en-US ure_en-US
 .ENDIF
