--- officecfg/registry/data/org/openoffice/Inet.xcu.orig	2005-10-02 18:55:59.000000000 +0400
+++ officecfg/registry/data/org/openoffice/Inet.xcu	2005-12-10 14:38:36.170143296 +0300
@@ -354,5 +354,93 @@
         </prop>
       </node>
     </node>
+    <node oor:name="Rambler" oor:op="replace">
+     <node oor:name="And">
+      <prop oor:name="ooInetCaseMatch" oor:type="xs:int">
+       <value>0</value>
+      </prop>
+      <prop oor:name="ooInetPrefix" oor:type="xs:string">
+       <value>http://www.rambler.ru/srch?words=</value>
+      </prop>
+      <prop oor:name="ooInetSeparator" oor:type="xs:string">
+       <value>+%26+</value>
+      </prop>
+      <prop oor:name="ooInetSuffix" oor:type="xs:string">
+         <value>&amp;btnG=%CD%E0%E9%F2%E8%21</value>
+      </prop>
+     </node>
+     <node oor:name="Exact">
+      <prop oor:name="ooInetCaseMatch" oor:type="xs:int">
+       <value>0</value>
+      </prop>
+      <prop oor:name="ooInetPrefix" oor:type="xs:string">
+       <value>http://www.rambler.ru/srch?words=%22</value>
+      </prop>
+      <prop oor:name="ooInetSeparator" oor:type="xs:string">
+       <value>%20</value>
+      </prop>
+      <prop oor:name="ooInetSuffix" oor:type="xs:string">
+       <value>%22&amp;btnG=%CD%E0%E9%F2%E8%21</value>
+      </prop>
+     </node>
+     <node oor:name="Or">
+      <prop oor:name="ooInetCaseMatch" oor:type="xs:int">
+       <value>0</value>
+      </prop>
+      <prop oor:name="ooInetPrefix" oor:type="xs:string">
+       <value>http://www.rambler.ru/srch?words=</value>
+      </prop>
+      <prop oor:name="ooInetSeparator" oor:type="xs:string">
+       <value>+%7C+</value>
+      </prop>
+      <prop oor:name="ooInetSuffix" oor:type="xs:string">
+       <value>&amp;btnG=%CD%E0%E9%F2%E8%21</value>
+      </prop>
+     </node>
+    </node>
+    <node oor:name="Яndex" oor:op="replace">
+     <node oor:name="And">
+      <prop oor:name="ooInetCaseMatch" oor:type="xs:int">
+       <value>0</value>
+      </prop>
+      <prop oor:name="ooInetPrefix" oor:type="xs:string">
+       <value>http://www.yandex.ru/yandsearch?rpt=rad&amp;text=</value>
+      </prop>
+      <prop oor:name="ooInetSeparator" oor:type="xs:string">
+       <value>+</value>
+      </prop>
+      <prop oor:name="ooInetSuffix" oor:type="xs:string">
+       <value/>
+      </prop>
+     </node>
+     <node oor:name="Exact">
+      <prop oor:name="ooInetCaseMatch" oor:type="xs:int">
+       <value>0</value>
+      </prop>
+      <prop oor:name="ooInetPrefix" oor:type="xs:string">
+       <value>http://www.yandex.ru/yandsearch?rpt=rad&amp;text=%22</value>
+      </prop>
+      <prop oor:name="ooInetSeparator" oor:type="xs:string">
+       <value>%20</value>
+      </prop>
+      <prop oor:name="ooInetSuffix" oor:type="xs:string">
+       <value>%22</value>
+      </prop>
+     </node>
+     <node oor:name="Or">
+      <prop oor:name="ooInetCaseMatch" oor:type="xs:int">
+       <value>0</value>
+      </prop>
+      <prop oor:name="ooInetPrefix" oor:type="xs:string">
+       <value>http://www.yandex.ru/yandsearch?rpt=rad&amp;text=</value>
+      </prop>
+      <prop oor:name="ooInetSeparator" oor:type="xs:string">
+       <value>+</value>
+      </prop>
+      <prop oor:name="ooInetSuffix" oor:type="xs:string">
+       <value/>
+      </prop>
+     </node>
+    </node>
   </node>
 </oor:component-data>
