diff --git a/desktop/source/app/appfirststart.cxx b/desktop/source/app/appfirststart.cxx
index b687cff..48b5300 100644
--- desktop/source/app/appfirststart.cxx
+++ desktop/source/app/appfirststart.cxx
@@ -154,7 +154,21 @@
         if ((result >>= bCompleted) && bCompleted)
             return sal_False;  // wizard was already completed
         else
+        {
+            sal_Bool bQuickstart( sal_True );
+            sal_Bool bAutostart( sal_True );
+            Sequence< Any > aSeq( 2 );
+            aSeq[0] <<= bQuickstart;
+            aSeq[1] <<= bAutostart;
+
+            Reference < XInitialization > xQuickstart( ::comphelper::getProcessServiceFactory()->createInstance(
+                                                 OUString::createFromAscii( "com.sun.star.office.Quickstart" )),
+                                                UNO_QUERY );
+            if ( xQuickstart.is() )
+                xQuickstart->initialize( aSeq );
+
             return sal_True;
+        }
     } catch (const Exception&)
     {
         return sal_True;
