=== modified file 'solenv/bin/modules/installer/ziplist.pm'
--- solenv/bin/modules/installer/ziplist.pm	2008-09-08 04:53:25 +0000
+++ solenv/bin/modules/installer/ziplist.pm	2008-09-08 04:54:02 +0000
@@ -679,12 +679,6 @@
 		$allvariables->{'PRODUCTVERSION'} = 1;	# FAKE
 	}
 	
-	# Creating differentiators for BUILD_SPECIAL in layering
-	if ( ! $ENV{'BUILD_SPECIAL'} )
-	{
-		if ( $allvariables->{'REGISTRYLAYERNAME'} ) { $allvariables->{'REGISTRYLAYERNAME'} = $allvariables->{'REGISTRYLAYERNAME'} . "_"; }
-		if (( $installer::globals::iswindowsbuild ) && ( $allvariables->{'BASISROOTNAME'} )) { $allvariables->{'BASISROOTNAME'} = $allvariables->{'BASISROOTNAME'} . "_"; }
-	} 
 }
 
 ####################################################

