--- officecfg/registry/schema/org/openoffice/Office/Common.xcs.orig	2007-08-22 17:49:47.000000000 +0400
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2007-08-22 17:51:01.000000000 +0400
@@ -2092,7 +2092,7 @@
 						<desc>Specifies if files saved in the OOo 6.0 XML file formats should be in pretty printing format. Saving and loading the document takes more time in pretty printing format.</desc>
 						<label>Open windows</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
                                 <prop oor:name="WarnAlienFormat" oor:type="xs:boolean">
                                         <!-- UIHints: Tools  Options - General  Save - -->
