=== modified file 'sc/source/ui/view/tabview3.cxx'
--- sc/source/ui/view/tabview3.cxx	2008-09-06 03:41:08 +0000
+++ sc/source/ui/view/tabview3.cxx	2008-09-06 05:19:50 +0000
@@ -1747,7 +1747,6 @@
 		SfxBindings& rBindings = aViewData.GetBindings();
 		ScMarkData& rMark = aViewData.GetMarkData();
 
-        BOOL bSelectOneTable = FALSE;
         if (bExtendSelection)
         {
             // #i6327# if all tables are selected, a selection event (#i6330#) will deselect all
@@ -1757,21 +1756,15 @@
             if( bAllSelected )
             {
                 bExtendSelection = FALSE;
-                bSelectOneTable = TRUE;
             }
         }
-        else
-        {
-            // move from multi-selection to unselected table
-            bSelectOneTable = !rMark.GetTableSelect( nTab );
-        }
 
         if (bExtendSelection)
         {
             // #i6330# multi-selection with keyboard
             rMark.SelectTable( nTab, TRUE );
         }
-        else if (bSelectOneTable)
+        else
 		{
 			rMark.SelectOneTable( nTab );
 			rBindings.Invalidate( FID_FILL_TAB );

