diff --git a/sc/source/core/tool/ddelink.cxx b/sc/source/core/tool/ddelink.cxx
index 29df108..e55c0a9 100644
--- sc/source/core/tool/ddelink.cxx
+++ sc/source/core/tool/ddelink.cxx
@@ -59,11 +59,7 @@
 
 TYPEINIT2(ScDdeLink,::sfx2::SvBaseLink,SfxBroadcaster);
 
-#ifdef PM2
-#define DDE_TXT_ENCODING    RTL_TEXTENCODING_IBM_850
-#else
-#define DDE_TXT_ENCODING 	RTL_TEXTENCODING_MS_1252
-#endif
+#define DDE_TXT_ENCODING 	gsl_getSystemTextEncoding()
 
 BOOL ScDdeLink::bIsInUpdate = FALSE;
 
diff --git a/sw/source/core/fields/ddefld.cxx b/sw/source/core/fields/ddefld.cxx
index c3d38fb..d24d63f 100644
--- sw/source/core/fields/ddefld.cxx
+++ sw/source/core/fields/ddefld.cxx
@@ -79,11 +79,7 @@
 using namespace rtl;
 using namespace ::com::sun::star;
 
-#ifdef PM2
-#define DDE_TXT_ENCODING    RTL_TEXTENCODING_IBM_850
-#else
-#define DDE_TXT_ENCODING 	RTL_TEXTENCODING_MS_1252
-#endif
+#define DDE_TXT_ENCODING 	gsl_getSystemTextEncoding()
 
 /*--------------------------------------------------------------------
 	Beschreibung: Globale Variablen
diff --git a/sw/source/ui/dochdl/swdtflvr.cxx b/sw/source/ui/dochdl/swdtflvr.cxx
index 6ad6f1d..de5c397 100644
--- sw/source/ui/dochdl/swdtflvr.cxx
+++ sw/source/ui/dochdl/swdtflvr.cxx
@@ -333,11 +333,7 @@ using namespace nsTransferBufferType;
 
 #ifdef DDE_AVAILABLE
 
-#ifdef PM2
-#define DDE_TXT_ENCODING    RTL_TEXTENCODING_IBM_850
-#else
-#define DDE_TXT_ENCODING 	RTL_TEXTENCODING_MS_1252
-#endif
+#define DDE_TXT_ENCODING 	gsl_getSystemTextEncoding()
 
 //---------------------------------------------
 // this struct conforms to the Microsoft
