--- /usr/share/portage/sys-fs/udev/udev-044.ebuild	2004-11-11 01:14:25.000000000 +0300
+++ udev-044.ebuild	2004-11-15 10:19:04.937370368 +0300
@@ -11,7 +11,7 @@
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~arm ~alpha ~amd64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86"
-IUSE="static"
+IUSE="static selinux"
 
 DEPEND="virtual/libc
 	sys-apps/hotplug-base"
@@ -62,6 +62,11 @@
 src_compile() {
 	local myconf=
 	local extras="extras/scsi_id extras/volume_id"
+	
+	if ( use selinux )
+	    then
+	    myconf="${myconf} USE_SELINUX=true"
+	fi
 
 	# Do not work with emake
 	make EXTRAS="${extras}" \
