diff -Naur ../work/jabbin-2.0beta2a/3party/jrtplib/rtprandom.cpp jabbin-2.0beta2a/3party/jrtplib/rtprandom.cpp
--- ../work/jabbin-2.0beta2a/3party/jrtplib/rtprandom.cpp	2006-08-10 14:09:46.000000000 +0400
+++ jabbin-2.0beta2a/3party/jrtplib/rtprandom.cpp	2006-12-05 10:11:44.000000000 +0300
@@ -53,7 +53,7 @@
 	x = (u_int32_t)getpid();
 	x += (u_int32_t)time(0);
 	x -= (u_int32_t)clock();
-	x ^= (u_int32_t)(this);
+	x ^= (u_int64_t)(this);
 
 #ifdef RTP_SUPPORT_GNUDRAND
 	srand48_r(x,&drandbuffer);
diff -Naur ../work/jabbin-2.0beta2a/src/libpsi/psiwidgets/iconaction.cpp jabbin-2.0beta2a/src/libpsi/psiwidgets/iconaction.cpp
--- ../work/jabbin-2.0beta2a/src/libpsi/psiwidgets/iconaction.cpp	2006-08-10 14:09:52.000000000 +0400
+++ jabbin-2.0beta2a/src/libpsi/psiwidgets/iconaction.cpp	2006-12-05 10:11:11.000000000 +0300
@@ -72,7 +72,7 @@
 
 			QMap<QPopupMenu *, int>::Iterator it = popups.begin();
 			for ( ; it != popups.end(); ++it ) {
-				if ( (int)it.key() == (int)obj ) {
+				if ( (int64_t)it.key() == (int64_t)obj ) {
 					dirty = true;
 					popups.remove( it );
 					break;
diff -Naur ../work/jabbin-2.0beta2a/voip/stun.cpp jabbin-2.0beta2a/voip/stun.cpp
--- ../work/jabbin-2.0beta2a/voip/stun.cpp	2006-08-10 14:09:58.000000000 +0400
+++ jabbin-2.0beta2a/voip/stun.cpp	2006-12-05 10:10:22.000000000 +0300
@@ -673,12 +673,12 @@
       asm("rdtsc" : "=A" (tick));
 #elif defined (__SUNPRO_CC) || defined( __sparc__ )	
       tick = gethrtime();
-#elif defined(__MACH__) 
+#else /* #elif defined(__MACH__) */
       int fd=open("/dev/random",O_RDONLY);
       read(fd,&tick,sizeof(tick));
       closesocket(fd);
-#else
-#     error Need some way to seed the random number generator 
+//#else
+//#     error Need some way to seed the random number generator 
 #endif 
       int seed = int(tick);
 #ifdef WIN32
