--- configure.orig	2008-07-01 16:25:13.000000000 +0200
+++ configure	2008-07-01 16:25:33.000000000 +0200
@@ -243,9 +243,9 @@
     "x")
         if test $ARG_WITH_GNUSTEP = 1; then
           DARG_IS_FHS=0;
-          ARG_PREFIX="$GNUSTEP_LOCAL_ROOT"
+          ARG_PREFIX="$GNUSTEP_SYSTEM_ROOT"
           if test $ARG_BEQUIET != 1; then
-	    echo "Note: will install in GNUSTEP_LOCAL_ROOT: $ARG_PREFIX"
+	    echo "Note: will install in GNUSTEP_SYSTEM_ROOT: $ARG_PREFIX"
 	    echo ""
 	  fi
         elif test "x$ARG_FRAMEWORK_DIR" != "x"; then
