--- sope-mime/NGImap4/NGImap4Client.m.orig	Tue Dec  4 11:00:36 2007
+++ sope-mime/NGImap4/NGImap4Client.m	Tue Jan 15 07:16:07 2008
@@ -53,18 +53,6 @@
 
 @end /* NGImap4Client(ConnectionRegistration); */
 
-#if GNUSTEP_BASE_LIBRARY
-/* FIXME: TODO: move someplace better (hh: NGExtensions...) */
-@implementation NSException(setUserInfo)
-
-- (id)setUserInfo:(NSDictionary *)_userInfo {
-  ASSIGN(self->_e_info, _userInfo);
-  return self;
-}
-
-@end /* NSException(setUserInfo) */
-#endif
-
 @interface NGImap4Client(Private)
 
 - (NSString *)_folder2ImapFolder:(NSString *)_folder;
@@ -967,9 +955,10 @@ static NSArray      *Imap4SystemFlags  = nil;
   descr = @"Could not process qualifier for imap search "; 
   descr = [descr stringByAppendingString:reason];           
   
-  exception = [[NGImap4SearchException alloc] initWithFormat:@"%@", descr];    
   ui = [NSDictionary dictionaryWithObject:_q forKey:@"qualifier"];
-  [exception setUserInfo:ui];
+  exception = [[NSException alloc] initWithName:@"NGImap4SearchException" 
+	reason:descr userInfo:ui];
+
   [self->context setLastException:exception];
   [exception release];
 }

