diff -Naur ../work/freetype-2.1.5/src/autohint/ahhint.c freetype-2.1.5/src/autohint/ahhint.c
--- ../work/freetype-2.1.5/src/autohint/ahhint.c	2003-08-18 03:00:59.000000000 +0400
+++ freetype-2.1.5/src/autohint/ahhint.c	2004-10-05 17:07:30.066871360 +0400
@@ -1851,7 +1851,7 @@
     hinter->do_stem_adjust   = FT_BOOL( hint_mode != FT_RENDER_MODE_LIGHT );
 
 
-    load_flags  |= FT_LOAD_NO_SCALE
+    load_flags  = FT_LOAD_NO_SCALE
                  | FT_LOAD_IGNORE_TRANSFORM ;
 
     error = ah_hinter_load( hinter, glyph_index, load_flags, 0 );
