diff -bdruN avidemux-9999.orig/avidemux/ADM_libraries/ADM_mplex/inputstrm.cpp avidemux-9999/avidemux/ADM_libraries/ADM_mplex/inputstrm.cpp
--- avidemux-9999.orig/avidemux/ADM_libraries/ADM_mplex/inputstrm.cpp	2008-04-12 23:28:37.000000000 +0200
+++ avidemux-9999/avidemux/ADM_libraries/ADM_mplex/inputstrm.cpp	2008-04-12 23:31:08.000000000 +0200
@@ -29,6 +29,8 @@
 #include "inputstrm.hpp"
 #include "multiplexor.hpp"
 
+#include <climits>
+
 MuxStream::MuxStream() : init(false) 
 {
 }
diff -bdruN avidemux-9999.orig/avidemux/ADM_libraries/ADM_mplex/multiplexor.cpp avidemux-9999/avidemux/ADM_libraries/ADM_mplex/multiplexor.cpp
--- avidemux-9999.orig/avidemux/ADM_libraries/ADM_mplex/multiplexor.cpp	2008-04-12 23:28:37.000000000 +0200
+++ avidemux-9999/avidemux/ADM_libraries/ADM_mplex/multiplexor.cpp	2008-04-12 23:31:08.000000000 +0200
@@ -36,7 +36,7 @@
 #include "zalphastrm.hpp"
 #endif
 #include "multiplexor.hpp"
-
+#include <cstring>
 
 /****************
  *
diff -bdruN avidemux-9999.orig/avidemux/ADM_libraries/ADM_mplex/padstrm.cpp avidemux-9999/avidemux/ADM_libraries/ADM_mplex/padstrm.cpp
--- avidemux-9999.orig/avidemux/ADM_libraries/ADM_mplex/padstrm.cpp	2008-04-12 23:28:37.000000000 +0200
+++ avidemux-9999/avidemux/ADM_libraries/ADM_mplex/padstrm.cpp	2008-04-12 23:31:08.000000000 +0200
@@ -27,6 +27,7 @@
 
 #include "padstrm.hpp"
 
+#include <cstring>
 
 
 //
diff -bdruN avidemux-9999.orig/avidemux/ADM_script/DirectorySearch.h avidemux-9999/avidemux/ADM_script/DirectorySearch.h
--- avidemux-9999.orig/avidemux/ADM_script/DirectorySearch.h	2008-04-12 23:28:37.000000000 +0200
+++ avidemux-9999/avidemux/ADM_script/DirectorySearch.h	2008-04-12 23:31:08.000000000 +0200
@@ -24,6 +24,7 @@
 #include <string.h>
 #endif
 #include "StdFile.h"
+#include <cstring>
 
 // create a non file bit for unix
 #define _A_NONFILE	0x03
diff -bdruN avidemux-9999.orig/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/ADM_jogshuttle.cpp avidemux-9999/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/ADM_jogshuttle.cpp
--- avidemux-9999.orig/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/ADM_jogshuttle.cpp	2008-04-12 23:28:37.000000000 +0200
+++ avidemux-9999/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/ADM_jogshuttle.cpp	2008-04-12 23:31:08.000000000 +0200
@@ -41,6 +41,8 @@
 #include "config.h"
 
 #ifdef USE_JOG
+#include <cstdlib>
+#include <cstring>
 #include <string>
 using std::string;
 #include <iostream>
diff -bdruN avidemux-9999.orig/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/mediactrl.h avidemux-9999/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/mediactrl.h
--- avidemux-9999.orig/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/mediactrl.h	2008-04-12 23:28:37.000000000 +0200
+++ avidemux-9999/avidemux/ADM_userInterfaces/ADM_GTK/ADM_toolkit_gtk/mediactrl.h	2008-04-12 23:31:08.000000000 +0200
@@ -97,6 +97,8 @@
 extern "C" {
 #endif
 
+#include <time.h>
+
 struct media_ctrl_device;
 
 struct media_ctrl_key
diff -bdruN avidemux-9999.orig/avidemux/gui_action.cpp avidemux-9999/avidemux/gui_action.cpp
--- avidemux-9999.orig/avidemux/gui_action.cpp	2008-04-12 23:28:36.000000000 +0200
+++ avidemux-9999/avidemux/gui_action.cpp	2008-04-12 23:34:40.000000000 +0200
@@ -1,5 +1,5 @@
-#include <stdio.h>
-#include <strings.h>
+#include <cstdio>
+#include <cstring>
 #include <vector>
 using std::vector;
 #include <utility>
