diff -Naur php-4.3.1/sapi/caudium/config.m4 php-4.3.1.new/sapi/caudium/config.m4
--- php-4.3.1/sapi/caudium/config.m4	2002-03-07 17:19:50.000000000 +0300
+++ php-4.3.1.new/sapi/caudium/config.m4	2003-03-03 13:50:16.000000000 +0300
@@ -83,7 +83,7 @@
     PIKE_VERSION=`$PIKE -e 'string v; int rel;sscanf(version(), "Pike v%s release %d", v, rel); write(v+"."+rel);'`   
 	AC_DEFINE(HAVE_CAUDIUM,1,[Whether to compile with Caudium support])
     PHP_SELECT_SAPI(caudium, shared, caudium.c)
-	INSTALL_IT="\$(INSTALL) -m 0755 $SAPI_SHARED $withval/lib/$PIKE_VERSION/PHP4.so"
+	INSTALL_IT="\$(mkinstalldirs) \$(INSTALL_ROOT)/$withval/lib/$PIKE_VERSION; \$(INSTALL) -m 0755 $SAPI_SHARED \${INSTALL_ROOT}/$withval/lib/$PIKE_VERSION/PHP4.so"
 	RESULT="	*** Pike binary used:         $PIKE
 	*** Pike include dir(s) used: $PIKE_INCLUDE_DIR
 	*** Pike version:             $PIKE_VERSION"
