diff -Nuar qtiplot-0.9.8.9.orig/qtiplot/src/analysis/dialogs/FitDialog.cpp qtiplot-0.9.8.9/qtiplot/src/analysis/dialogs/FitDialog.cpp
--- qtiplot-0.9.8.9.orig/qtiplot/src/analysis/dialogs/FitDialog.cpp	2011-10-17 13:38:07.000000000 +0300
+++ qtiplot-0.9.8.9/qtiplot/src/analysis/dialogs/FitDialog.cpp	2017-09-28 22:56:51.174119689 +0300
@@ -1020,9 +1020,9 @@
     for (int i = 0; i<parameters; i++){
         QTableWidgetItem *it = new QTableWidgetItem(paramList[i]);
 #ifdef Q_CC_MSVC
-        it->setFlags(it->flags() & (~Qt::ItemIsEditable));
+        it->setFlags(it->flags() & Qt::ItemFlags(~Qt::ItemIsEditable));
 #else
-        it->setFlags(!Qt::ItemIsEditable);
+        it->setFlags(Qt::ItemFlags(!Qt::ItemIsEditable));
 #endif
         it->setBackground(QBrush(Qt::lightGray));
         it->setForeground(Qt::black);
@@ -1049,7 +1049,7 @@
         boxParams->setCellWidget(i, 2, sb);
 
         it = new QTableWidgetItem();
-		it->setFlags(!Qt::ItemIsEditable);
+		it->setFlags(Qt::ItemFlags(!Qt::ItemIsEditable));
 		it->setText("--");
 		boxParams->setItem(i, 5, it);
 	}
@@ -1062,9 +1062,9 @@
 		for (int i = 0; i<boxParams->rowCount(); i++ ){
             QTableWidgetItem *it = new QTableWidgetItem();
 #ifdef Q_CC_MSVC
-            it->setFlags(it->flags() & (~Qt::ItemIsEditable));
+            it->setFlags(it->flags() & Qt::ItemFlags(~Qt::ItemIsEditable));
 #else
-            it->setFlags(!Qt::ItemIsEditable);
+            it->setFlags(Qt::ItemFlags(!Qt::ItemIsEditable));
 #endif
             it->setBackground(QBrush(Qt::lightGray));
             boxParams->setItem(i, 4, it);
diff -Nuar qtiplot-0.9.8.9.orig/qtiplot/src/core/ConfigDialog.cpp qtiplot-0.9.8.9/qtiplot/src/core/ConfigDialog.cpp
--- qtiplot-0.9.8.9.orig/qtiplot/src/core/ConfigDialog.cpp	2011-10-25 12:50:22.000000000 +0300
+++ qtiplot-0.9.8.9/qtiplot/src/core/ConfigDialog.cpp	2017-09-28 22:57:11.911109684 +0300
@@ -1390,7 +1390,7 @@
 	colorsList->setRowCount(colors);
 	for (int i = 0; i < colors; i++){
 		QTableWidgetItem *it = new QTableWidgetItem();
-		it->setFlags(!Qt::ItemIsEditable);
+		it->setFlags(Qt::ItemFlags(!Qt::ItemIsEditable));
 		it->setBackground(QBrush(colList[i]));
 		colorsList->setItem(i, 0, it);
 
diff -Nuar qtiplot-0.9.8.9.orig/qtiplot/src/lib/src/ColorMapEditor.cpp qtiplot-0.9.8.9/qtiplot/src/lib/src/ColorMapEditor.cpp
--- qtiplot-0.9.8.9.orig/qtiplot/src/lib/src/ColorMapEditor.cpp	2011-07-02 10:23:52.000000000 +0300
+++ qtiplot-0.9.8.9/qtiplot/src/lib/src/ColorMapEditor.cpp	2017-09-28 22:55:36.008780653 +0300
@@ -156,9 +156,9 @@
 
 		QTableWidgetItem *it = new QTableWidgetItem(c.name());
 	#ifdef Q_CC_MSVC
-		it->setFlags(it->flags() & (~Qt::ItemIsEditable));
+		it->setFlags(it->flags() & Qt::ItemFlags(~Qt::ItemIsEditable));
 	#else
-		it->setFlags(!Qt::ItemIsEditable);
+		it->setFlags(Qt::ItemFlags(!Qt::ItemIsEditable));
 	#endif
 		it->setBackground(QBrush(c));
 		it->setForeground(QBrush(c));
@@ -247,9 +247,9 @@
 
 	QTableWidgetItem *it = new QTableWidgetItem(c.name());
 #ifdef Q_CC_MSVC
-	it->setFlags(it->flags() & (~Qt::ItemIsEditable));
+	it->setFlags(it->flags() & Qt::ItemFlags(~Qt::ItemIsEditable));
 #else
-	it->setFlags(!Qt::ItemIsEditable);
+	it->setFlags(Qt::ItemFlags(!Qt::ItemIsEditable));
 #endif
 	it->setBackground(QBrush(c));
 	it->setForeground(QBrush(c));
diff -Nuar qtiplot-0.9.8.9.orig/qtiplot/src/matrix/MatrixModel.cpp qtiplot-0.9.8.9/qtiplot/src/matrix/MatrixModel.cpp
--- qtiplot-0.9.8.9.orig/qtiplot/src/matrix/MatrixModel.cpp	2011-07-01 03:06:08.000000000 +0300
+++ qtiplot-0.9.8.9/qtiplot/src/matrix/MatrixModel.cpp	2017-09-28 22:58:56.303025225 +0300
@@ -110,9 +110,9 @@
 Qt::ItemFlags MatrixModel::flags(const QModelIndex & index ) const
 {
 	if (index.isValid())
-		return Qt::ItemIsEnabled | Qt::ItemIsSelectable | Qt::ItemIsEditable;
+		return Qt::ItemFlags(Qt::ItemIsEnabled | Qt::ItemIsSelectable | Qt::ItemIsEditable);
 	else
-		return Qt::ItemIsEnabled;
+		return Qt::ItemFlags(Qt::ItemIsEnabled);
 }
 
 int MatrixModel::rowCount(const QModelIndex & /* parent */) const
diff -Nuar qtiplot-0.9.8.9.orig/qtiplot/src/plot2D/dialogs/AssociationsDialog.cpp qtiplot-0.9.8.9/qtiplot/src/plot2D/dialogs/AssociationsDialog.cpp
--- qtiplot-0.9.8.9.orig/qtiplot/src/plot2D/dialogs/AssociationsDialog.cpp	2010-10-01 14:17:22.000000000 +0300
+++ qtiplot-0.9.8.9/qtiplot/src/plot2D/dialogs/AssociationsDialog.cpp	2017-09-28 22:59:51.913316699 +0300
@@ -225,7 +225,7 @@
 		for (int i=0; i<table->rowCount(); i++ ){
 			QTableWidgetItem *cell = new QTableWidgetItem(colNames[i].replace(",", "."));
 			cell->setBackground (QBrush(Qt::lightGray));
-			cell->setFlags (Qt::ItemIsEnabled);
+			cell->setFlags (Qt::ItemFlags(Qt::ItemIsEnabled));
 			table->setItem(i, 0, cell);
 			}
 
diff -Nuar qtiplot-0.9.8.9.orig/qtiplot/src/plot2D/dialogs/PlotDialog.cpp qtiplot-0.9.8.9/qtiplot/src/plot2D/dialogs/PlotDialog.cpp
--- qtiplot-0.9.8.9.orig/qtiplot/src/plot2D/dialogs/PlotDialog.cpp	2011-09-13 14:50:34.000000000 +0300
+++ qtiplot-0.9.8.9/qtiplot/src/plot2D/dialogs/PlotDialog.cpp	2017-09-28 23:00:42.121871267 +0300
@@ -5211,7 +5211,7 @@
     : QTreeWidgetItem( parent, QStringList(s), PlotCurveTreeItem ),
       d_curve(curve)
 {
-	setFlags(Qt::ItemIsSelectable | Qt::ItemIsUserCheckable | Qt::ItemIsEnabled);
+	setFlags(Qt::ItemFlags(Qt::ItemIsSelectable | Qt::ItemIsUserCheckable | Qt::ItemIsEnabled));
 	setIcon(0, QPixmap(":/graph_disabled.png"));
 	setCheckState(0, curve->isVisible() ? Qt::Checked : Qt::Unchecked);
 }
@@ -5264,7 +5264,7 @@
 	d_widget(w)
 {
 	setActive(false);
-	setFlags(Qt::ItemIsSelectable | Qt::ItemIsUserCheckable | Qt::ItemIsEnabled);
+	setFlags(Qt::ItemFlags(Qt::ItemIsSelectable | Qt::ItemIsUserCheckable | Qt::ItemIsEnabled));
 	setCheckState(0, d_widget->isVisible() ? Qt::Checked : Qt::Unchecked);
 }
 
