diff --git a/Make.inc b/Make.inc
index b8b04c9..9e5a3c6 100644
--- a/Make.inc
+++ b/Make.inc
@@ -384,7 +384,7 @@ ifneq ($(OS), WINNT)
 JCXXFLAGS += -pedantic
 endif
 DEBUGFLAGS := -O0 -ggdb2 -DJL_DEBUG_BUILD -fstack-protector-all
-SHIPFLAGS := -O3 -ggdb2 -falign-functions
+SHIPFLAGS := GENTOOCFLAGS
 endif
 
 ifeq ($(USECLANG),1)
@@ -395,7 +395,7 @@ JCFLAGS := -pipe $(fPIC) -fno-strict-aliasing -D_FILE_OFFSET_BITS=64
 JCPPFLAGS := -fasynchronous-unwind-tables
 JCXXFLAGS := -pipe $(fPIC) -fno-rtti -pedantic
 DEBUGFLAGS := -O0 -g -DJL_DEBUG_BUILD -fstack-protector-all
-SHIPFLAGS := -O3 -g
+SHIPFLAGS := GENTOOCFLAGS
 ifeq ($(OS), Darwin)
 ifeq ($(USE_LIBCPP), 1)
 CC += -stdlib=libc++ -mmacosx-version-min=10.7
@@ -422,7 +422,7 @@ JCFLAGS := -std=gnu11 -pipe $(fPIC) -fno-strict-aliasing -D_FILE_OFFSET_BITS=64
 JCPPFLAGS :=
 JCXXFLAGS := -pipe $(fPIC) -fno-rtti
 DEBUGFLAGS := -O0 -g -DJL_DEBUG_BUILD -fstack-protector-all
-SHIPFLAGS := -O3 -g -falign-functions
+SHIPFLAGS := GENTOOCFLAGS
 endif
 
 ifeq ($(USECCACHE), 1)
diff --git a/doc/Makefile b/doc/Makefile
index 0749d73..f4d49ec 100644
--- a/doc/Makefile
+++ b/doc/Makefile
@@ -62,7 +62,7 @@ help:
 	@echo "  doctest    to run all doctests embedded in the documentation (if enabled)"
 
 clean:
-	-rm -rf _build/*
+	@echo "Do not clean doc/_build/html. Just use it..."
 
 cleanall: clean
 
diff --git a/src/Makefile b/src/Makefile
index d7ec864..09cf8e0 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -75,7 +75,7 @@ else
 ifeq ($(OS),WINNT)
 LLVMLINK += $(shell $(LLVM_CONFIG_HOST) --ldflags) -lLLVM-$(LLVM_VER_SHORT)
 else
-LLVMLINK += $(shell $(LLVM_CONFIG_HOST) --ldflags) -lLLVM-$(shell $(LLVM_CONFIG_HOST) --version)
+LLVMLINK += $(shell $(LLVM_CONFIG_HOST) --ldflags) -lLLVM
 endif # OS == WINNT
 endif # LLVM_USE_CMAKE == 1
 FLAGS += -DLLVM_SHLIB
