diff -ruN a/doc/example.conf b/doc/example.conf
--- a/doc/example.conf	2008-04-24 17:29:34.000000000 +0400
+++ b/doc/example.conf	2008-05-24 23:28:14.000000000 +0400
@@ -163,6 +163,8 @@
 	#
 	# If you give "" no chroot is performed. The path must not end in a /.
 	# chroot: "/usr/local/etc/unbound"
+	# GENTOO SPECIFIC - do not change
+chroot: ""
 
 	# if given, user privileges are dropped (after binding port),
 	# and the given username is assumed. Default is user "unbound".
@@ -184,6 +186,8 @@
 
 	# the pid file.
 	# pidfile: "/usr/local/etc/unbound/unbound.pid"
+	# GENTOO SPECIFIC - do not change
+pidfile: "/var/run/unbound.pid"
 	
 	# file to read root hints from.
 	# get one from ftp://FTP.INTERNIC.NET/domain/named.cache
