diff -uNr ldapserver.org/ldap/nsldap.mk ldapserver/ldap/nsldap.mk
--- ldapserver.org/ldap/nsldap.mk	2005-06-02 17:28:37.000000000 +0200
+++ ldapserver/ldap/nsldap.mk	2005-06-02 17:29:29.000000000 +0200
@@ -1493,7 +1493,7 @@
 #
 
 # compiler to use, e.g. CC=cc or CC=gcc
-CC=/usr/bin/gcc -fwritable-strings
+CC=/usr/bin/gcc
 
 # give full path to hostname since it may not be in user's path
 HOSTNAME=/bin/hostname
diff -uNr ldapserver.org/ldap/servers/plugins/replication/repl5_replica.c ldapserver/ldap/servers/plugins/replication/repl5_replica.c
--- ldapserver.org/ldap/servers/plugins/replication/repl5_replica.c	2005-06-02 17:28:37.000000000 +0200
+++ ldapserver/ldap/servers/plugins/replication/repl5_replica.c	2005-06-02 17:30:40.000000000 +0200
@@ -108,9 +108,9 @@
 static char * _replica_get_config_dn (const Slapi_DN *root);
 static char * _replica_type_as_string (const Replica *r);
 /* DBDB, I think this is probably bogus : */
-int replica_create_ruv_tombstone(Replica *r);
-void assign_csn_callback(const CSN *csn, void *data);
-void abort_csn_callback(const CSN *csn, void *data);
+static int replica_create_ruv_tombstone(Replica *r);
+static void assign_csn_callback(const CSN *csn, void *data);
+static void abort_csn_callback(const CSN *csn, void *data);
 static void eq_cb_reap_tombstones(time_t when, void *arg);
 static CSN *_replica_get_purge_csn_nolock (const Replica *r);
 static void replica_get_referrals_nolock (const Replica *r, char ***referrals);
diff -uNr ldapserver.org/nsconfig.mk ldapserver/nsconfig.mk
--- ldapserver.org/nsconfig.mk	2005-06-02 17:28:37.000000000 +0200
+++ ldapserver/nsconfig.mk	2005-06-02 17:29:08.000000000 +0200
@@ -665,8 +665,8 @@
 else 
 ifeq ($(ARCH), Linux)
 OSVERSION	:= $(basename $(shell uname -r))
-CC=/usr/bin/gcc -fwritable-strings
-CXX=/usr/bin/g++ -fwritable-strings
+CC=/usr/bin/gcc
+CXX=/usr/bin/g++
 CCC=$(CXX)
 LD=$(CXX)
 ARCH_DEBUG=-g
