diff -Pur OpenModelica/Compiler/Ceval.mo OpenModelica-patched/Compiler/Ceval.mo
--- OpenModelica/Compiler/Ceval.mo	2008-06-26 00:11:32.000000000 +0000
+++ OpenModelica-patched/Compiler/Ceval.mo	2008-07-03 20:50:00.000000000 +0000
@@ -1916,7 +1916,7 @@
         pd = System.pathDelimiter();
         executableSuffixedExe = stringAppend(executable, System.getExeExt());
         sim_call = Util.stringAppendList(
-          {cit,executableSuffixedExe,cit," > output.log 2>&1"});
+          {cit,"./",executableSuffixedExe,cit," > output.log 2>&1"});
         0 = System.systemCall(sim_call);
         result_file = Util.stringAppendList({executable,"_res.plt"});
         simValue = Values.RECORD(Absyn.IDENT("SimulationResult"),
diff -Pur OpenModelica/Compiler/omc_debug/Makefile.in OpenModelica-patched/Compiler/omc_debug/Makefile.in
--- OpenModelica/Compiler/omc_debug/Makefile.in	2008-06-26 00:11:32.000000000 +0000
+++ OpenModelica-patched/Compiler/omc_debug/Makefile.in	2008-06-25 05:30:20.000000000 +0000
@@ -36,9 +36,9 @@
 
 ifeq ($(OSTYP),$(OS_MSYS))
 	LIBSOCKET=
-	CORBALIB=-L$(CORBAHOME)/lib -lmico -lwsock32
+	CORBALIB=-L$(CORBAHOME)/lib -lmico2.3.12 -lwsock32
 else
-	CORBALIB=-L$(CORBAHOME)/lib -lmico -lcrypto -lssl -lpthread -lreadline
+	CORBALIB=-L$(CORBAHOME)/lib -lmico2.3.12 -lcrypto -lssl -lpthread -lreadline
 endif
 
 ifdef USE_CORBA
diff -Pur OpenModelica/Compiler/omc_release/Makefile.in OpenModelica-patched/Compiler/omc_release/Makefile.in
--- OpenModelica/Compiler/omc_release/Makefile.in	2008-06-26 00:11:32.000000000 +0000
+++ OpenModelica-patched/Compiler/omc_release/Makefile.in	2008-06-25 05:30:04.000000000 +0000
@@ -36,9 +36,9 @@
 
 ifeq ($(OSTYP),$(OS_MSYS))
 	LIBSOCKET=
-	CORBALIB=-L$(CORBAHOME)/lib -lmico -lwsock32
+	CORBALIB=-L$(CORBAHOME)/lib -lmico2.3.12 -lwsock32
 else
-	CORBALIB=-L$(CORBAHOME)/lib -lmico -lcrypto -lssl -lpthread
+	CORBALIB=-L$(CORBAHOME)/lib -lmico2.3.12 -lcrypto -lssl -lpthread
 endif
 
 ifdef USE_CORBA
diff -Pur OpenModelica/Compiler/runtime/corbaimpl_stub.cpp OpenModelica-patched/Compiler/runtime/corbaimpl_stub.cpp
--- OpenModelica/Compiler/runtime/corbaimpl_stub.cpp	2008-06-26 00:11:32.000000000 +0000
+++ OpenModelica-patched/Compiler/runtime/corbaimpl_stub.cpp	2008-06-25 05:29:15.000000000 +0000
@@ -25,13 +25,13 @@
 {	
 }
 
-void errmsg() {
+void errmsg_CORBA() {
   cerr << "CORBA disabled. Configure with --with-CORBA and recompile for enabling." << endl;
 }
 
 RML_BEGIN_LABEL(Corba__initialize)
 {
-  errmsg();
+  errmsg_CORBA();
   RML_TAILCALLK(rmlFC);
 }
 RML_END_LABEL
@@ -39,21 +39,21 @@
 
 RML_BEGIN_LABEL(Corba__waitForCommand)
 {
-  errmsg();
+  errmsg_CORBA();
   RML_TAILCALLK(rmlFC);
 }
 RML_END_LABEL
 
 RML_BEGIN_LABEL(Corba__sendreply)
 {
-  errmsg();
+  errmsg_CORBA();
   RML_TAILCALLK(rmlFC);
 }
 RML_END_LABEL
 
 RML_BEGIN_LABEL(Corba__close)
 {
-  errmsg();
+  errmsg_CORBA();
   RML_TAILCALLK(rmlFC);
 }
 RML_END_LABEL
diff -Pur OpenModelica/Compiler/SimCodegen.mo OpenModelica-patched/Compiler/SimCodegen.mo
--- OpenModelica/Compiler/SimCodegen.mo	2008-06-26 00:11:32.000000000 +0000
+++ OpenModelica-patched/Compiler/SimCodegen.mo	2008-07-03 20:30:59.000000000 +0000
@@ -142,6 +142,7 @@
           " $(CFLAGS)",
           " -I.",
           " -o ",cname,"$(EXEEXT) ",cpp_file,
+          " -static",
           " -lsim",
           " $(LDFLAGS)",
           " -lf2c",
diff -Pur OpenModelica/configure OpenModelica-patched/configure
--- OpenModelica/configure	2008-06-26 00:11:33.000000000 +0000
+++ OpenModelica-patched/configure	2008-06-27 06:06:47.000000000 +0000
@@ -3232,7 +3232,7 @@
 
 echo "$as_me:$LINENO: checking for rml includefiles in $RMLHOME" >&5
 echo $ECHO_N "checking for rml includefiles in $RMLHOME... $ECHO_C" >&6
-rmlinc=`find $RMLHOME -follow -type d -name plain |grep include/plain`
+rmlinc=`find ${RMLHOME}/include -follow -type d -name plain |grep include/plain`
 if test -z "$rmlinc" ; then
   { { echo "$as_me:$LINENO: error: rml headers not found" >&5
 echo "$as_me: error: rml headers not found" >&2;}
@@ -3246,7 +3246,7 @@
 
 echo "$as_me:$LINENO: checking for antlr header files" >&5
 echo $ECHO_N "checking for antlr header files... $ECHO_C" >&6
-antlrinc=`find $ANTLRHOME /usr/local/include /usr/include -type f -name TreeParser.hpp -print | head -1 | awk -F/ '{ total="";for (i = 1 ; i < NF-1; i++) total= total $i "/"; print total}'`
+antlrinc=`find ${ANTLRHOME}/include /usr/local/include /usr/include -type f -name TreeParser.hpp -print | head -1 | awk -F/ '{ total="";for (i = 1 ; i < NF-1; i++) total= total $i "/"; print total}'`
 if test -z "$antlrinc" ; then
   { { echo "$as_me:$LINENO: error: Could not find antlr include files" >&5
 echo "$as_me: error: Could not find antlr include files" >&2;}
@@ -3258,7 +3258,7 @@
 
 echo "$as_me:$LINENO: checking for antlr libs" >&5
 echo $ECHO_N "checking for antlr libs... $ECHO_C" >&6
-antlrlib=`find $ANTLRHOME /usr/local/lib /usr/lib -type f -name libantlr.a -print | head -1 | awk -F/ '{ total="";for (i = 1 ; i < NF; i++) total= total $i "/"; print total}'`
+antlrlib=`find -L ${ANTLRHOME}/lib /usr/local/lib /usr/lib -type f -name libantlr.a -print | head -1 | awk -F/ '{ total="";for (i = 1 ; i < NF; i++) total= total $i "/"; print total}'`
 if test -z "$antlrlib" ; then
   { { echo "$as_me:$LINENO: error: Could not find antlr lib file" >&5
 echo "$as_me: error: Could not find antlr lib file" >&2;}
@@ -3310,7 +3310,7 @@
   echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
 else
-  readlinelib=`find /usr/lib /usr/local/lib -name libreadline.a | tail -1 | awk -F/ '{ total="";for (i = 1 ; i < NF; i++) total= total $i "/"; print total}'`
+  readlinelib=`find -L /usr/lib /usr/local/lib -name libreadline.a | tail -1 | awk -F/ '{ total="";for (i = 1 ; i < NF; i++) total= total $i "/"; print total}'`
   if test -z "$readlinelib" ; then
      echo "$as_me:$LINENO: result: no" >&5
 echo "${ECHO_T}no" >&6
diff -Pur OpenModelica/c_runtime/Makefile OpenModelica-patched/c_runtime/Makefile
--- OpenModelica/c_runtime/Makefile	2008-06-26 00:11:33.000000000 +0000
+++ OpenModelica-patched/c_runtime/Makefile	2008-07-03 07:36:01.000000000 +0000
@@ -6,7 +6,7 @@
 CC = gcc
 FC = g77
 AR = ar -ru 
-CFLAGS = -O -fexceptions -Wall -ansi -pedantic -I$(top_builddir)/mosh/src/ $(EXTRA_CFLAGS)
+CFLAGS = -O -fexceptions -fPIC -Wall -ansi -pedantic -I$(top_builddir)/mosh/src/ $(EXTRA_CFLAGS)
 CPPFLAGS = $(CFLAGS)
 FFLAGS  = -O -fexceptions
 # P.A: before, g77 had -O3 or -O2 but that caused a bug in DDASRT, giving infinite loop.
diff -Pur OpenModelica/OMShell/OMShell.pro OpenModelica-patched/OMShell/OMShell.pro
--- OpenModelica/OMShell/OMShell.pro	2008-06-26 00:11:33.000000000 +0000
+++ OpenModelica-patched/OMShell/OMShell.pro	2008-07-02 18:11:57.000000000 +0000
@@ -1,8 +1,8 @@
 TEMPLATE = app
 
 DEPENDPATH += .
-INCLUDEPATH += /home/adrpo/dev/mico-2.3.12/include
-LIBS+=-L/home/adrpo/dev/mico-2.3.12/lib -lmico2.3.12   -lssl -lcrypto -ldl -lbsd -lm  -lpthread
+INCLUDEPATH += .
+LIBS+= $$system(mico-config --libs)
 CONFIG += warn_on
 QT += network xml 
 
@@ -18,3 +18,5 @@
            oms.cpp \
            main.cpp
 
+RESOURCES += oms.qrc
+
diff -Pur OpenModelica/OMShell/oms.qrc OpenModelica-patched/OMShell/oms.qrc
--- OpenModelica/OMShell/oms.qrc	2008-06-26 00:11:33.000000000 +0000
+++ OpenModelica-patched/OMShell/oms.qrc	2008-07-02 18:12:50.000000000 +0000
@@ -1,14 +1,14 @@
 <RCC>
     <qresource prefix="/" >
-        <file>Resources\OMS.bmp</file>
-        <file>Resources\OMS.ico</file>
-        <file>Resources\clear.bmp</file>
-        <file>Resources\copy.bmp</file>
-        <file>Resources\cut.bmp</file>
-        <file>Resources\help.bmp</file>
-        <file>Resources\paste.bmp</file>
-        <file>Resources\print.bmp</file>
-        <file>Resources\start.bmp</file>
-        <file>Resources\stop.bmp</file>
+        <file>Resources/OMS.bmp</file>
+        <file>Resources/OMS.ico</file>
+        <file>Resources/clear.bmp</file>
+        <file>Resources/copy.bmp</file>
+        <file>Resources/cut.bmp</file>
+        <file>Resources/help.bmp</file>
+        <file>Resources/paste.bmp</file>
+        <file>Resources/print.bmp</file>
+        <file>Resources/start.bmp</file>
+        <file>Resources/stop.bmp</file>
     </qresource>
 </RCC>
