--- linphone-3.4.1/configure.ac.orig	2011-03-01 12:39:43.061643811 +0300
+++ linphone-3.4.1/configure.ac	2011-03-01 12:43:09.803556308 +0300
@@ -267,15 +267,6 @@
 
 AC_WORDS_BIGENDIAN
 
-dnl normaly this should only by done by mediastreamer2/configure.ac
-dnl but to workaround bugs when cross-compiling for arm-linux,
-dnl we need to have SPEEX_LIBS defined
-dnl Furthermore it is good to repeat here all mediastreamer2 toggles
-dnl since top-level configure --help will not print them.
-
-PKG_CHECK_MODULES(SPEEX, speex >= 1.1.6, build_speex=yes)
-AC_SUBST(SPEEX_LIBS)
-
 dnl conditionnal build of video support
 AC_ARG_ENABLE(video,
 		[  --enable-video    Turn on video support compiling],
