--- TeX.orig/texk/kpathsea/texmf.in	2007-02-02 23:26:57.000000000 +0100
+++ TeX/texk/kpathsea/texmf.in	2007-05-27 18:44:00.074408384 +0200
@@ -56,31 +56,34 @@
 
 % The tree containing the runtime files closely related to the specific
 % program version used:
-TEXMFMAIN = @texmf@
+TEXMFMAIN = /usr/share/texmf
 
 % The main distribution tree:
-TEXMFDIST = @texmf@-dist
+TEXMFDIST = /usr/share/texmf-dist
+
+% The Gentoo site tree
+TEXMFSITE = /usr/share/texmf-site
 
 % A place for local additions to a "standard" texmf tree.
 % This tree is not used for local configuration maintained by
 % texconfig, it uses TEXMFCONFIG below.
-TEXMFLOCAL = @texmf@-local
+TEXMFLOCAL = /usr/local/share/texmf
 
 % TEXMFSYSVAR, where texconfig-sys stores variable runtime data.
 % With teTeX-3.0 or later, this must be set.
 % For sharing this tree with $TEXMFMAIN:
 %   TEXMFSYSVAR = $TEXMFMAIN
 % For using a separate tree:
-%   TEXMFSYSVAR = @texmf@-var
-TEXMFSYSVAR = @texmf@-var
+%   TEXMFSYSVAR = /var/lib/texmf-var
+TEXMFSYSVAR = /var/lib/texmf
 
 % TEXMFSYSCONFIG, where texconfig-sys stores configuration data.
 % With teTeX-3.0 or later, this must be set.
 % For sharing this tree with $TEXMFMAIN:
 %   TEXMFSYSCONFIG = $TEXMFMAIN
 % For using a separate tree:
-%   TEXMFSYSCONFIG = @texmf@-config
-TEXMFSYSCONFIG = @texmf@-config
+%   TEXMFSYSCONFIG = /var/lib/texmf-config
+TEXMFSYSCONFIG = /etc/texmf
 
 % User texmf trees are allowed as follows.
 % This used to be HOMETEXMF.
@@ -111,25 +114,28 @@
 %
 % For texconfig to work properly, TEXMFCONFIG and TEXMFVAR should be named
 % explicitly and before all other trees.
-TEXMF = {$TEXMFCONFIG,$TEXMFVAR,$TEXMFHOME,!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFMAIN,!!$TEXMFLOCAL,!!$TEXMFDIST}
+TEXMF = {$TEXMFCONFIG,$TEXMFVAR,$TEXMFHOME,!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFMAIN,!!$TEXMFLOCAL,!!$TEXMFSITE,!!$TEXMFDIST}
 
 % The system trees.  These are the trees that are shared by all the users.
 % If a tree appears in this list, the mktex* scripts will use
 % VARTEXFONTS for generated files, if the original tree isn't writable;
 % otherwise the current working directory is used.
-SYSTEXMF = $TEXMFLOCAL;$TEXMFMAIN;$TEXMFDIST
+SYSTEXMF = $TEXMFLOCAL;$TEXMFMAIN;$TEXMFDIST;$TEXMFSITE
 
 % Where generated fonts may be written.  This tree is used when the sources
 % were found in a system tree and either that tree wasn't writable, or the
 % varfonts feature was enabled in MT_FEATURES in mktex.cnf.
-VARTEXFONTS = @vartexfonts@
+VARTEXFONTS = /var/tmp/texfonts
 
 % Where to look for ls-R files.  There need not be an ls-R in the
 % directories in this path, but if there is one, Kpathsea will use it.
 %
-% Remove $VARTEXFONTS from TEXMFDBS if the VARTEXFONTS directory is below
-% one of the TEXMF directories (avoids overlapping ls-R files).
-TEXMFDBS = $TEXMF;$VARTEXFONTS
+% By default, this is only the !! elements of TEXMF, so that mktexlsr
+% does not create ls-R files in the non-!! elements -- because if an
+% ls-R is present, it will be used, and the disk will not be searched.
+% This is arguably a bug in kpathsea, but we will not think about it now.
+% 
+TEXMFDBS = {!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFMAIN,!!$TEXMFLOCAL,!!$TEXMFSITE,!!$TEXMFDIST}
 
 % It may be convenient to define TEXMF like this:
 %   TEXMF = {$TEXMFHOME,!!$TEXMFLOCAL,!!$TEXMFMAIN,$HOME}
@@ -149,6 +155,9 @@
 % is used on many Linux systems.  In this case, set VARTEXFONTS like this
 % VARTEXFONTS = $TEXMFVAR/fonts
 % and do not mention it in TEXMFDBS (but _do_ mention TEXMFVAR).
+%
+% Remove $VARTEXFONTS from TEXMFDBS if the VARTEXFONTS directory is below
+% one of the TEXMF directories (avoids overlapping ls-R files).
 
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
@@ -286,7 +295,7 @@
 
 % BibTeX bibliographies and style files.
 BIBINPUTS = .;$TEXMF/bibtex/bib//
-BSTINPUTS = .;$TEXMF/bibtex/bst//
+BSTINPUTS = .;$TEXMF/bibtex/{bst,csf}//
 
 % MFT style files.
 MFTINPUTS = .;$TEXMF/mft//
@@ -296,14 +305,22 @@
 TEXPSHEADERS = .;$TEXMF/{dvips,fonts/{enc,type1,type42,type3}}//
 TEXPSHEADERS.gsftopk = .;$TEXMF/{dvips,fonts/{enc,type1,type42,type3,truetype}}//
 
+% OSFONTDIR is to provide a convenient hook for allowing TeX to find
+% fonts installed on the system (outside of TeX).  An empty default
+% value would add "//" to the search paths, so we give it a dummy value.
+OSFONTDIR = /please/set/osfontdir/in/the/environment
+
 % PostScript Type 1 outline fonts.
-T1FONTS = .;$TEXMF/fonts/type1//
+T1FONTS = .;$TEXMF/fonts/type1//;$OSFONTDIR//
 
 % PostScript AFM metric files.
-AFMFONTS = .;$TEXMF/fonts/afm//
+AFMFONTS = .;$TEXMF/fonts/afm//;$OSFONTDIR//
 
 % TrueType outline fonts.
-TTFONTS = .;$TEXMF/fonts/truetype//
+TTFONTS = .;$TEXMF/fonts/truetype//;$OSFONTDIR//
+
+% Opentype outline fonts.
+OPENTYPEFONTS = .;$TEXMF/fonts/opentype//;$OSFONTDIR//
 
 % Type 42 outline fonts.
 T42FONTS = .;$TEXMF/fonts/type42//
@@ -339,8 +356,15 @@
 % For xdvi to find mime.types and .mailcap, if they do not exist in
 % $HOME.  These are single directories, not paths.
 % (But the default mime.types, at least, may well suffice.)
-MIMELIBDIR = @prefix@/etc
-MAILCAPLIBDIR = @prefix@/etc
+MIMELIBDIR = /usr/etc
+MAILCAPLIBDIR = /usr/etc
+
+% Default settings for fontconfig library, used by Win32 versions of  
+% xetex/xdvipdfmx (note that fontconfig on Linux/Unix-like systems 
+% will ignore settings in this file)
+FONTCONFIG_FILE=fonts.conf
+FONTCONFIG_PATH=$SELFAUTOLOC/conf
+FC_CACHEDIR=$SELFAUTOLOC/cache
 
 % TeX documentation and source files, for use with kpsewhich.
 TEXDOCS = .;$TEXMF/doc//
@@ -364,7 +388,7 @@
 T4HTINPUTS   = .;$TEXMF/tex4ht//
 
 % Architecture independent executables.
-TEXMFSCRIPTS = $TEXMF/scripts//
+TEXMFSCRIPTS = $TEXMF/scripts/{$engine,$progname,}//
 
 %% The mktex* scripts rely on KPSE_DOT. Do not set it in the environment.
 % KPSE_DOT = .
@@ -373,13 +397,13 @@
 % paradoxical), but the compile-time default in paths.h is built from it.
 % The SELFAUTO* variables are set automatically from the location of
 % argv[0], in kpse_set_progname.
-%
-% The TETEXDIR stuff isn't likely to relevant unless you're using teTeX,
-% but it doesn't hurt.
+% 
+% The TETEXDIR stuff isn't likely to be relevant unless you're using
+% teTeX, but it doesn't hurt.
 %
 % For security reasons, it is better not to have . part of the path.
 %
-TEXMFCNF = {$SELFAUTOLOC,$SELFAUTODIR,$SELFAUTOPARENT}{,{/share,}/texmf{-local,}/web2c};/.$TETEXDIR;$TEXMF/web2c;@web2c@
+TEXMFCNF = {$SELFAUTOLOC,$SELFAUTODIR,$SELFAUTOPARENT}{,{/share,}/texmf{-local,}/web2c};/.$TETEXDIR;$TEXMF/web2c
 
 # $progname: kpathsea v. 3.5.3 or later overwrites this at runtime. To
 # avoid empty expansions from binaries linked against an earlier
@@ -419,7 +443,13 @@
 
 % Allow TeX, MF, and MP to parse the first line of an input file for
 % the %&format construct.
-parse_first_line = f
+parse_first_line = t
+
+% But don't parse the first line if invoked as "tex", since we want that
+% to remain Knuth-compatible.  The src_specials and
+% file_line_error_style settings, as well as the options -enctex,
+% -mltex, -8bit, etc., also affect this, but they are all off by default.
+parse_first_line.tex = f
 
 % Control file:line:error style messages.
 file_line_error_style = f
@@ -445,6 +475,10 @@
 % if MP is in troff mode.  Set to `0' to disable this feature.
 MPXCOMMAND = makempx
 
+% Used by makempx to run TeX.  We use "etex" because MetaPost is
+% expecting DVI, and not "tex" because we want first line parsing.
+TEX = etex
+
 
 %  Part 3: Array and other sizes for TeX (and Metafont and MetaPost).
 %
@@ -498,19 +532,19 @@
 % min pool space left after loading .fmt
 pool_free = 47500
 
-% Hyphenation trie.  The maximum possible is 4194303 (ssup_trie_size in
-% the sources), but we don't need that much.  The value here suffices
-% for all known free hyphenation patterns to be loaded simultaneously.
-%
-trie_size = 300000
-
 % Buffer size.  TeX uses the buffer to contain input lines, but macro
 % expansion works by writing material into the buffer and reparsing the
 % line.  As a consequence, certain constructs require the buffer to be
-% very large.  As distributed, the size is 50000; most documents can be
-% handled within a tenth of this size.
+% very large, even though most documents can be handled with a small value.
 buf_size = 200000
 
+% Hyphenation trie.  The maximum possible is 4194303 (ssup_trie_size in
+% the sources), but we don't need that much.  The value here suffices
+% for all known free hyphenation patterns to be loaded simultaneously
+% (as TeX Live does).
+%
+trie_size = 300000
+
 hyph_size = 8191        % prime number of hyphenation exceptions, >610, <32767.
                         % http://primes.utm.edu/curios/page.php/8191.html
 nest_size = 500         % simultaneous semantic levels (e.g., groups)
@@ -542,3 +576,47 @@
 error_line = 79
 half_error_line = 50
 max_print_line = 79
+
+
+%  Part 4: extra format definitions for TeX Live
+
+TEXINPUTS.elambda       = .;$TEXMF/tex/{lambda,latex,generic,}//
+TEXINPUTS.eomega        = .;$TEXMF/tex/{plain,generic,}//
+TEXINPUTS.pdfxmex       = .;$TEXMF/tex/{mex,plain,generic,}//
+TEXINPUTS.frpdflatex    = .;$TEXMF/tex/{french,latex,generic,}//
+TEXINPUTS.frpdftex      = .;$TEXMF/tex/{french,plain,generic,}//
+TEXINPUTS.xmltex        = .;$TEXMF/tex/{xmltex,latex,generic,}//
+TEXINPUTS.pdfxmltex     = .;$TEXMF/tex/{xmltex,latex,generic,}//
+
+TTF2TFMINPUTS =                 .;$TEXMF/ttf2pk//
+
+% For bibtex8. Do we really need to repeat the BIBINPUTS setting?
+BIBINPUTS.bibtex8 =             .;$TEXMF/bibtex/{bib,}//
+BSTINPUTS.bibtex8 =             .;$TEXMF/bibtex/{bst,csf,}//
+
+% allow for compressed files, and various extenions
+TEXDOCSSUFFIX   = ;.pdf;.ps;.dvi;.html;.txt;.tex
+TEXDOCSCOMPRESS = ;.gz;.bz2;.zip;.Z
+TEXDOCEXT       = {$TEXDOCSSUFFIX}{$TEXDOCSCOMPRESS}
+
+%% t4ht utility, sharing files with TeX4ht
+TEX4HTFONTSET=alias,iso8859,unicode
+TEX4HTINPUTS = .;$TEXMF/tex4ht/base//;$TEXMF/tex4ht/ht-fonts/{$TEX4HTFONTSET}//
+
+% Suggestions for editor settings
+
+% vi, vim, NEdit, (X)Emacs, pico, jed
+% TEXEDIT =                     vi     +%d %s
+% TEXEDIT =                     vim    +%d %s
+% TEXEDIT =                     nedit  +%d %s
+% TEXEDIT =                     xemacs +%d %s
+% TEXEDIT =                     emacs +%d %s
+
+%(x)fte:
+% TEXEDIT =                     xfte   -l%d %s
+
+% Disable search on multiple suffixes filenames. In many case, when `foo.bar'
+% is looked for, you do not want to look for `foo.bar.tex' before. This flag
+% disables searching for standard suffixes if the file name has already an
+% extension of 3 characters. Default value is true (old behaviour).
+allow_multiple_suffixes =                       f
