diff -NaurwB heimdal-1.2.1rc1.orig/cf/krb-ipv6.m4 heimdal-1.2.1rc1/cf/krb-ipv6.m4
--- heimdal-1.2.1rc1.orig/cf/krb-ipv6.m4	2008-05-31 16:38:43.000000000 +0200
+++ heimdal-1.2.1rc1/cf/krb-ipv6.m4	2008-07-19 00:12:39.000000000 +0200
@@ -2,12 +2,14 @@
 dnl
 dnl test for IPv6
 dnl
+ac_cv_lib_ipv6=yes
+
 AC_DEFUN([AC_KRB_IPV6], [
 AC_ARG_WITH(ipv6,
 	AS_HELP_STRING([--without-ipv6],[do not enable IPv6 support]),[
-if test "$withval" = "no"; then
-	ac_cv_lib_ipv6=no
-fi])
+        ac_cv_lib_ipv6="$withval"
+])
+if test $ac_cv_lib_ipv6 != "no"; then
 save_CFLAGS="${CFLAGS}"
 AC_CACHE_CHECK([for IPv6 stack type], rk_cv_v6type,
 [dnl check for different v6 implementations (by itojun)
@@ -146,4 +148,5 @@
 			[Define if you have the in6addr_loopback variable])
 	fi
 fi
+fi
 ])
\ No newline at end of file
diff -NaurwB heimdal-1.2.1rc1.orig/lib/roken/mini_inetd.c heimdal-1.2.1rc1/lib/roken/mini_inetd.c
--- heimdal-1.2.1rc1.orig/lib/roken/mini_inetd.c	2008-05-31 16:38:40.000000000 +0200
+++ heimdal-1.2.1rc1/lib/roken/mini_inetd.c	2008-07-19 00:13:50.000000000 +0200
@@ -134,7 +134,11 @@
     memset (&hints, 0, sizeof(hints));
     hints.ai_flags    = AI_PASSIVE;
     hints.ai_socktype = SOCK_STREAM;
-    hints.ai_family   = PF_UNSPEC;
+#ifdef HAVE_IPV6
+    hints.ai_family   = AF_UNSPEC;
+#else
+    hints.ai_family   = AF_INET;
+#endif
 
     snprintf (portstr, sizeof(portstr), "%d", ntohs(port));
 
