diff -Nru xc.old/programs/Xserver/hw/xfree86/loader/xf86sym.c xc/programs/Xserver/hw/xfree86/loader/xf86sym.c
--- xc.old/programs/Xserver/hw/xfree86/loader/xf86sym.c	2004-08-09 17:03:23.712128392 -0400
+++ xc/programs/Xserver/hw/xfree86/loader/xf86sym.c	2004-08-09 17:04:05.029847144 -0400
@@ -233,6 +233,10 @@
 extern unsigned short ldw_brx(volatile unsigned char *, int);
 #endif
 
+/* Propolice! */
+extern long __guard[];
+extern void __stack_smash_handler(char [], int);
+
 /* XFree86 things */
 
 LOOKUP xfree86LookupTab[] = {
@@ -1093,6 +1097,10 @@
 #endif
 #endif
 
+    /* propolice */
+    SYMFUNC(__stack_smash_handler)
+    SYMVAR(__guard)
+
     /* Some variables. */
 
     SYMVAR(xf86stdin)
