--- src/main.c	2006-08-24 13:56:32.000000000 +0200
+++ src/main.c.oden	2006-08-24 13:53:29.000000000 +0200
@@ -146,7 +146,7 @@
 int pause_after_run = pause_on_dumb_terminals;
 
 /* It true saves the setup when quitting */
-int auto_save_setup = 1;
+int auto_save_setup = 0;
 
 #ifndef HAVE_CHARSET
 /* If true, allow characters in the range 160-255 */
@@ -201,7 +201,7 @@
 static int force_xterm = 0;
 
 /* Controls screen clearing before an exec */
-int clear_before_exec = 1;
+int clear_before_exec = 0;
 
 /* Asks for confirmation before deleting a file */
 int confirm_delete = 1;
@@ -213,7 +213,7 @@
 int confirm_execute = 0;
 
 /* Asks for confirmation before leaving the program */
-int confirm_exit = 1;
+int confirm_exit = 0;
 
 /* Asks for confirmation when using F3 to view a directory and there
    are tagged files */
