diff -purN kdebase-3.5.3.orig/konqueror/listview/konq_listviewwidget.cc kdebase-3.5.3.pertty/konqueror/listview/konq_listviewwidget.cc
--- kdebase-3.5.3.orig/konqueror/listview/konq_listviewwidget.cc	2005-10-10 10:04:14.000000000 -0500
+++ kdebase-3.5.3.pertty/konqueror/listview/konq_listviewwidget.cc	2006-06-04 10:32:52.000000000 -0500
@@ -405,7 +405,6 @@ void KonqBaseListViewWidget::contentsMou
 {
    if ( m_rubber )
    {
-      drawRubber();
       delete m_rubber;
       m_rubber = 0;
    }
@@ -422,13 +421,11 @@ void KonqBaseListViewWidget::contentsMou
    else {
       if ( e->button() == LeftButton )
       {
-         if ( !( e->state() & ControlButton ) )
-            setSelected( itemAt( vp ), false );
          m_rubber = new QRect( e->x(), e->y(), 0, 0 );
          m_fileTip->setItem( 0 );
+         clearSelection();
+         emit selectionChanged();
       }
-      if ( e->button() != RightButton )
-         QListView::contentsMousePressEvent( e );
    }
    // Store list of selected items at mouse-press time.
    // This is used when autoscrolling (why?)
@@ -440,9 +437,10 @@ void KonqBaseListViewWidget::contentsMou
 {
    if ( m_rubber )
    {
-      drawRubber();
+      QRect r( m_rubber->normalize() );
       delete m_rubber;
       m_rubber = 0;
+      repaintContents( r, FALSE );
    }
 
    if ( m_scrollTimer )
@@ -534,23 +532,21 @@ void KonqBaseListViewWidget::leaveEvent(
    KListView::leaveEvent( e );
 }
 
-void KonqBaseListViewWidget::drawRubber()
+void KonqBaseListViewWidget::drawRubber( QPainter *p )
 {
    if ( !m_rubber )
       return;
 
-   QPainter p;
-   p.begin( viewport() );
-   p.setRasterOp( NotROP );
-   p.setPen( QPen( color0, 1 ) );
-   p.setBrush( NoBrush );
+   p->setRasterOp( NotROP );
+   p->setPen( QPen( color0, 1 ) );
+   p->setBrush( NoBrush );
 
    QPoint pt( m_rubber->x(), m_rubber->y() );
    pt = contentsToViewport( pt );
-   style().drawPrimitive( QStyle::PE_FocusRect, &p,
+   style().drawPrimitive( QStyle::PE_RubberBand, p,
                           QRect( pt.x(), pt.y(), m_rubber->width(), m_rubber->height() ),
                           colorGroup(), QStyle::Style_Default, colorGroup().base() );
-   p.end();
+   p->end();
 }
 
 void KonqBaseListViewWidget::slotAutoScroll()
@@ -566,10 +562,11 @@ void KonqBaseListViewWidget::slotAutoScr
    if ( vc == m_rubber->bottomRight() )
       return;
 
+   QRect oldRubber = *m_rubber;
+
    const int oldTop = m_rubber->normalize().top();
    const int oldBottom = m_rubber->normalize().bottom();
 
-   drawRubber();
    m_rubber->setBottomRight( vc );
 
    QListViewItem *cur = itemAt( QPoint(0,0) );
@@ -577,7 +574,9 @@ void KonqBaseListViewWidget::slotAutoScr
    bool block = signalsBlocked();
    blockSignals( true );
 
+   QRect rr;
    QRect nr = m_rubber->normalize();
+   bool changed = FALSE;
    if ( cur )
    {
       QRect rect = itemRect( cur );
@@ -606,9 +605,25 @@ void KonqBaseListViewWidget::slotAutoScr
          if ( rect.intersects( nr ) )
          {
             if ( !cur->isSelected() && cur->isSelectable() )
+            {
                setSelected( cur, true );
-         } else if ( !m_selected || !m_selected->contains( (KonqBaseListViewItem*)cur ) )
-            setSelected( cur, false );
+               changed = TRUE;
+               rr = rr.unite( itemRect( cur ) );
+            }
+         }
+         else 
+         {
+            if ( cur->isSelected() )
+            {
+               changed = TRUE;
+               rr = rr.unite( itemRect( cur ) );
+            }
+
+            if ( !m_selected || !m_selected->contains( (KonqBaseListViewItem*)cur ) )
+            {
+               setSelected( cur, false );
+            }
+         }
 
          cur = cur->itemBelow();
          if (cur && !allColumnsShowFocus())
@@ -625,9 +640,25 @@ void KonqBaseListViewWidget::slotAutoScr
          if ( rect.intersects( nr ) )
          {
             if ( !cur->isSelected() && cur->isSelectable() )
+            {
                setSelected( cur, true );
-         } else if ( !m_selected || !m_selected->contains( (KonqBaseListViewItem*)cur ) )
-            setSelected( cur, false );
+               changed = TRUE;
+               rr = rr.unite( itemRect( cur ) );
+            }
+         }
+         else 
+         {
+            if ( cur->isSelected() )
+            {
+               changed = TRUE;
+               rr = rr.unite( itemRect( cur ) );
+            }
+
+            if ( !m_selected || !m_selected->contains( (KonqBaseListViewItem*)cur ) )
+            {
+               setSelected( cur, false );
+            }
+         }
 
          cur = cur->itemAbove();
          if (cur && !allColumnsShowFocus())
@@ -639,7 +670,31 @@ void KonqBaseListViewWidget::slotAutoScr
    blockSignals( block );
    emit selectionChanged();
 
-   drawRubber();
+   QRect allRect = oldRubber.normalize();
+   if ( changed ) 
+   {
+       allRect |= rr.normalize();
+   }
+   allRect |= m_rubber->normalize();
+   QPoint point = contentsToViewport( allRect.topLeft() );
+   allRect = QRect( point.x(), point.y(), allRect.width(), allRect.height() );
+   allRect &= viewport()->rect();
+   allRect.addCoords( -2, -2, 2, 2 );
+
+   QPixmap backrubber( viewport()->rect().size() );
+   backrubber.fill( viewport(), viewport()->rect().topLeft() );
+
+   QPainter p( &backrubber );
+   p.save();
+   drawContentsOffset( &p,
+      contentsX(),
+      contentsY(),
+      contentsX() + allRect.left(), contentsY() + allRect.top(),
+      allRect.width(), allRect.height() );
+   p.restore();
+   drawRubber( &p );
+   p.end();
+   bitBlt( viewport(), allRect.topLeft(), &backrubber, allRect );
 
    const int scroll_margin = 40;
    ensureVisible( vc.x(), vc.y(), scroll_margin, scroll_margin );
@@ -669,9 +724,11 @@ void KonqBaseListViewWidget::slotAutoScr
 
 void KonqBaseListViewWidget::viewportPaintEvent( QPaintEvent *e )
 {
-   drawRubber();
    KListView::viewportPaintEvent( e );
-   drawRubber();
+
+   QPainter p( viewport() );
+   drawRubber( &p );
+   p.end();
 }
 
 void KonqBaseListViewWidget::viewportResizeEvent(QResizeEvent * e)
diff -purN kdebase-3.5.3.orig/konqueror/listview/konq_listviewwidget.h kdebase-3.5.3.pertty/konqueror/listview/konq_listviewwidget.h
--- kdebase-3.5.3.orig/konqueror/listview/konq_listviewwidget.h	2005-10-10 10:04:14.000000000 -0500
+++ kdebase-3.5.3.pertty/konqueror/listview/konq_listviewwidget.h	2006-06-04 10:32:52.000000000 -0500
@@ -206,7 +206,7 @@ protected:
    virtual void viewportPaintEvent( QPaintEvent *e );
    virtual void viewportResizeEvent( QResizeEvent *e );
 
-   virtual void drawRubber();
+   virtual void drawRubber( QPainter * );
    virtual void contentsMousePressEvent( QMouseEvent *e );
    virtual void contentsMouseReleaseEvent( QMouseEvent *e );
    virtual void contentsMouseMoveEvent( QMouseEvent *e );
@@ -246,6 +246,7 @@ protected:
    QColor m_itemColor;
 
    QRect *m_rubber;
+   QPixmap *m_backrubber;
 
    bool m_bTopLevelComplete:1;
    bool m_showIcons:1;
