See http://bugzilla.gnome.org/show_bug.cgi?id=454739

Index: configure.in
===================================================================
--- configure.in	(Revision 2627)
+++ configure.in	(Arbeitskopie)
@@ -56,7 +56,7 @@
 AC_SUBST(DBUS_CFLAGS)
 AC_SUBST(DBUS_LIBS)
 
-PKG_CHECK_MODULES(NETWORK_MANAGER, NetworkManager >= 0.4.1)
+PKG_CHECK_MODULES(NETWORK_MANAGER, NetworkManager >= 0.4.1 libnm-util >= 0.7.0)
 AC_SUBST(NETWORK_MANAGER_CFLAGS)
 AC_SUBST(NETWORK_MANAGER_LIBS)
 
