--- bzr_exporter.py 2009-02-21 22:06:41 +0000
+++ bzr_exporter.py 2009-03-01 16:43:34 +0000
@@ -262,7 +262,7 @@
                     'revision %s' % (tag, revid))
             else:
                 git_ref = 'refs/tags/%s' % tag
- self.print_cmd(commands.ResetCommand(git_ref, ":" + mark))
+ self.print_cmd(commands.ResetCommand(git_ref, ":" + str(mark)))

     def _next_tmp_branch_name(self):
         """Return a unique branch name. The name will start with "tmp"."""