--- gw6c.conf.old	2008-11-28 19:09:13.000000000 +0300
+++ gw6c.conf	2009-03-31 10:28:13.000000000 +0400
@@ -48,8 +48,8 @@ passwd=
 #   Freenet6 account holders should enter broker.freenet6.net, otherwise use
 #   anon.freenet6.net. Your ISP may provide you with a different server name. 
 #
-server=anon.freenet6.net
-#server=broker.freenet6.net
+server=anonymous.freenet6.net
+#server=authenticated.freenet6.net
 
 #
 # Authentication Method:
