--- /dev/null   2007-08-24 09:09:18.263739314 +0200
+++ setup.py    2007-09-18 13:12:52.449378006 +0200
@@ -0,0 +1,25 @@
+"""Setup script for the dynagen module distribution."""
+
+from distutils.core import setup, Extension
+setup( # Distribution meta-data
+        name = "gns3",
+        version = "0.2alpha",
+        description = "A graphical frontend to dynamips",
+        author = "Jeremy Grossmann, David Ruiz, Romain Lamaison, Aurelien Levesque, Xavier Alt",
+        author_email = "contact@gns3.net",
+        url = "http://www.gns3.net/",
+        scripts = [ 'gns3' ],
+       packages = [
+               'GNS3',
+               'GNS3.Config',
+               'GNS3.Globals',
+               'GNS3.Link',
+               'GNS3.Node',
+               'GNS3.NodeConfigs',
+               'GNS3.Ui',
+               'GNS3.Ui.ConfigurationPages',
+               'GNS3.Dynagen',
+               'GNS3.Dynagen.Globals',
+       ],
+       package_data = { },
+)
