--- help/Makefile.am.orig	2008-06-12 22:51:10.000000000 +0400
+++ help/Makefile.am	2008-06-19 14:55:28.000000000 +0400
@@ -4,10 +4,10 @@ EXTRA_DIST = reconf spec
 install-exec-local:
 	@echo
 	@echo "Installing the gambas help files..."
-	@$(INSTALL) -d $(DESTDIR)$(gbdatadir)/help
-	@rm -rf $(DESTDIR)$(gbdatadir)/help
-	@cp -R $(srcdir)/help $(DESTDIR)$(gbdatadir)
-	@(cd $(DESTDIR)$(gbdatadir)/help; tar xfz $(DESTDIR)$(gbdatadir)/help/help.tar.gz; $(LN_S) help/img img; $(LN_S) help/image image; rm -f Makefile*)
+	@$(INSTALL) -d $(DESTDIR)$(htmldir)
+	@cp -R $(srcdir)/help/{tree,*.css,*.png} $(DESTDIR)$(htmldir)
+	@tar xfz $(srcdir)/help/help.tar.gz -C $(DESTDIR)$(htmldir)
+	@(cd $(DESTDIR)$(htmldir); $(LN_S) help/img img; $(LN_S) help/image image)
 
 uninstall-local:
-	@rm -rf $(DESTDIR)$(gbdatadir)/help
+	@rm -rf $(DESTDIR)/help
