diff -urN mc-4.6.0-pre3.orig/doc/mc.1.in mc-4.6.0-pre3/doc/mc.1.in
--- mc-4.6.0-pre3.orig/doc/mc.1.in	2003-01-14 21:01:34.000000000 +0100
+++ mc-4.6.0-pre3/doc/mc.1.in	2003-01-14 21:59:38.000000000 +0100
@@ -1337,7 +1337,7 @@
 but only if it is owned by user or root and is not world-writable.
 If no such file found, ~/.mc/menu is tried in the same way,
 and otherwise mc uses the default system-wide menu
-@prefix@/share/mc/mc.menu.
+/etc/mc/mc.menu.
 .PP
 The format of the menu file is very simple. Lines that start with
 anything but space or tab are considered entries for the menu (in
@@ -1855,7 +1855,7 @@
 At startup the Midnight Commander will try to load initialization
 information from the ~/.mc/ini file. If this file doesn't exist, it will
 load the information from the system-wide configuration file, located in
-@prefix@/share/mc/mc.ini. If the system-wide configuration file doesn't
+/etc/mc/mc.ini. If the system-wide configuration file doesn't
 exist, MC uses the default settings.
 .PP
 The
@@ -3045,7 +3045,7 @@
 .IP
 The help file for the program.
 .PP
-.I @prefix@/share/mc/mc.ext
+.I /etc/mc/mc.ext
 .IP
 The default system-wide extensions file.
 .PP
@@ -3054,12 +3054,12 @@
 User's own extension, view configuration and edit configuration
 file.  They override the contents of the system wide files if present.
 .PP
-.I @prefix@/share/mc/mc.ini
+.I /etc/mc/mc.ini
 .IP
 The default system-wide setup for the Midnight Commander, used only if
 the user doesn't have his own ~/.mc/ini file.
 .PP
-.I @prefix@/share/mc/mc.lib
+.I /etc/mc/mc.lib
 .IP
 Global settings for the Midnight Commander.  Settings in this file
 affect all users, whether they have ~/.mc/ini or not.  Currently, only
@@ -3077,7 +3077,7 @@
 .IP
 This file contains the hints (cookies) displayed by the program.
 .PP
-.I @prefix@/share/mc/mc.menu
+.I /etc/mc/mc.menu
 .IP
 This file contains the default system-wide applications menu.
 .PP
diff -urN mc-4.6.0-pre3.orig/doc/mcedit.1.in mc-4.6.0-pre3/doc/mcedit.1.in
--- mc-4.6.0-pre3.orig/doc/mcedit.1.in	2003-01-20 15:18:36.000000000 +0100
+++ mc-4.6.0-pre3/doc/mcedit.1.in	2003-01-23 18:57:57.000000000 +0100
@@ -436,12 +436,12 @@
 .IP
 The help file for the program.
 .PP
-.I @prefix@/share/mc/mc.ini
+.I /etc/mc/mc.ini
 .IP
 The default system-wide setup for GNU Midnight Commander, used only if
 the user's own ~/.mc/ini file is missing.
 .PP
-.I @prefix@/share/mc/mc.lib
+.I /etc/mc/mc.lib
 .IP
 Global settings for the Midnight Commander.  Settings in this file
 affect all users, whether they have ~/.mc/ini or not.
diff -urN mc-4.6.0-pre3.orig/doc/mcview.1.in mc-4.6.0-pre3/doc/mcview.1.in
--- mc-4.6.0-pre3.orig/doc/mcview.1.in	2003-01-20 15:18:36.000000000 +0100
+++ mc-4.6.0-pre3/doc/mcview.1.in	2003-01-23 18:58:24.000000000 +0100
@@ -65,12 +65,12 @@
 .IP
 The help file for the program.
 .PP
-.I @prefix@/share/mc/mc.ini
+.I /etc/mc/mc.ini
 .IP
 The default system-wide setup for GNU Midnight Commander, used only if
 the user's own ~/.mc/ini file is missing.
 .PP
-.I @prefix@/share/mc/mc.lib
+.I /etc/mc/mc.lib
 .IP
 Global settings for the Midnight Commander.  Settings in this file
 affect all users, whether they have ~/.mc/ini or not.
