Index: qscintilla2-2.8.4+dfsg/Python/configure.py
===================================================================
--- qscintilla2-2.8.4+dfsg.orig/Python/configure.py	2014-09-21 22:26:07.110346928 -0400
+++ qscintilla2-2.8.4+dfsg/Python/configure.py	2014-09-21 22:26:07.106346928 -0400
@@ -1440,7 +1440,10 @@
 
     libs = qmake_config.get('LIBS')
     if libs:
-        pro.write('LIBS += %s\n' % libs)
+        if target_config.pyqt_package == 'PyQt5':
+            pro.write('LIBS += %s -lqt5scintilla2\n' % libs)
+        else:
+            pro.write('LIBS += %s -lqscintilla2\n' % libs)
 
     if not opts.static:
         pro.write('''
