diff --git a/QScintilla-gpl-2.9/Python/configure.py b/QScintilla-gpl-2.9/Python/configure.py
index 0e006ad..bfd8a4b 100644
--- a/QScintilla-gpl-2.9/Python/configure.py
+++ b/QScintilla-gpl-2.9/Python/configure.py
@@ -210,9 +210,17 @@ class ModuleConfiguration(object):
         if lib_dir is None:
             lib_dir = target_configuration.qt_lib_dir
 
-        if not glob.glob(os.path.join(lib_dir, '*qscintilla2*')):
-            error(
-                    "The QScintilla library could not be found in %s. If "
+        if target_configuration.pyqt_package == 'PyQt5':
+            if not glob.glob(os.path.join(lib_dir, '*qt5scintilla2*')):
+                error(
+                    "The qt5scintilla2 library could not be found in %s. If "
+                    "QScintilla is installed then use the --qsci-libdir "
+                    "argument to explicitly specify the correct "
+                    "directory." % lib_dir)
+        else:
+            if not glob.glob(os.path.join(lib_dir, '*qscintilla2*')):
+                error(
+                    "The qscintilla2 library could not be found in %s. If "
                     "QScintilla is installed then use the --qsci-libdir "
                     "argument to explicitly specify the correct "
                     "directory." % lib_dir)
