--- coreutils-5.2.1-r5.ebuild.old	2005-03-24 18:10:16.389413968 +0300
+++ coreutils-5.2.1-r5.ebuild	2005-03-24 18:11:08.759452512 +0300
@@ -61,6 +61,7 @@
 	EPATCH_SUFFIX="patch" epatch ${PATCHDIR}/generic
 	EPATCH_SUFFIX="patch" epatch ${PATCHDIR}/extra
 
+	use acl && sed -i -e "0,23 d" ${PATCHDIR}/selinux/001_all_coreutils-acl-selinux.patch
 	use selinux && EPATCH_SUFFIX="patch" epatch ${PATCHDIR}/selinux
 
 	# Sparc32 SMP bug fix -- see bug #46593
