--- a/src/xine-engine/buffer.h  Mon Aug 18 18:27:55 2008 +0100
+++ b/src/xine-engine/buffer.h  Tue Aug 19 00:26:52 2008 +0200
@@ -676,7 +676,7 @@ void _x_bmiheader_le2me( xine_bmiheader
 /* convert xine_waveformatex struct from little endian */
 void _x_waveformatex_le2me( xine_waveformatex *wavex ) XINE_PROTECTED;

-static inline _x_is_fourcc(void *ptr, void *tag) {
+static __inline int _x_is_fourcc(void *ptr, void *tag) {
   return memcmp(ptr, tag, 4) == 0;
 }
