diff -Naurp xerces-c-3.0.1/src/xercesc/util/MsgLoaders/ICU/resources/Makefile.in xerces-c-3.0.1.new/src/xercesc/util/MsgLoaders/ICU/resources/Makefile.in
--- xerces-c-3.0.1/src/xercesc/util/MsgLoaders/ICU/resources/Makefile.in	2008-06-23 14:57:28.000000000 +0400
+++ xerces-c-3.0.1.new/src/xercesc/util/MsgLoaders/ICU/resources/Makefile.in	2009-06-12 12:21:13.906608546 +0400
@@ -50,8 +50,7 @@ PKGDATA=$(ICU_PREFIX)/bin/pkgdata
 
 # Add -v if you want to see more verbose output.
 #
-PKGDATAOPTS=-d . -M '"CC=$(CC)" "CXX=$(CXX)" "AR=$(AR)" "RANLIB=$(RANLIB)" \
-"CPPFLAGS=$(CPPFLAGS)" "CFLAGS=$(CFLAGS)" "CXXFLAGS=$(CXXFLAGS)" "LDFLAGS=$(LDFLAGS)"'
+PKGDATAOPTS=-d .
 
 # Resource shortname
 PKGNAME=xercesc_messages_$(INTERFACE_VER_U)
