--- gnuserv.el~	2004-10-19 10:50:01.000000000 +0200
+++ gnuserv.el	2007-05-25 22:49:01.000000000 +0200
@@ -733,7 +733,10 @@
   (unless leave-dead
     (setq gnuserv-string ""
 	  gnuserv-current-client nil)
-    (let ((process-connection-type t))
+    (let ((process-connection-type t)
+	  (exec-path
+	   (append exec-path
+		   (file-expand-wildcards "/usr/lib/xemacs-*/*-*"))))
       (setq gnuserv-process
 	    (start-process "gnuserv" nil gnuserv-program)))
     (set-process-sentinel gnuserv-process 'gnuserv-sentinel)
