--- Config.kmk.anc	2008-09-05 13:03:51.000000000 +0200
+++ Config.kmk	2008-09-05 13:07:36.000000000 +0200
@@ -165,22 +165,22 @@
 # This indicates that additions (of some kind or another) is being _built_.
 # VBOX_WITHOUT_ADDITIONS overrides it.
 #if1of ($(KBUILD_TARGET), freebsd linux os2 solaris win)
-if1of ($(KBUILD_TARGET), linux solaris win)
- VBOX_WITH_ADDITIONS = 1
-endif
+#if1of ($(KBUILD_TARGET), linux solaris win)
+# VBOX_WITH_ADDITIONS = 1
+#endif
 # Build the optional ring-0 part of the additions for syntax checking.
 # (Except when cross-building on linux since we're using /usr/src/...)
-ifn1of ($(KBUILD_TARGET).$(KBUILD_HOST_ARCH).$(KBUILD_TARGET_ARCH), linux.x86.amd64 linux.amd64.x86)
- VBOX_WITH_ADDITION_DRIVERS = 1
-endif
+#ifn1of ($(KBUILD_TARGET).$(KBUILD_HOST_ARCH).$(KBUILD_TARGET_ARCH), linux.x86.amd64 linux.amd64.x86)
+# VBOX_WITH_ADDITION_DRIVERS = 1
+#endif
 # Build X11 additions. Can be disabled separately.
 if1of ($(KBUILD_TARGET), dragonfly freebsd linux netbsd openbsd solaris)
  VBOX_WITH_X11_ADDITIONS = 1
 endif
 # Enable the building of shared folders in the windows additions.
-VBOX_WITH_WIN32_ADDITIONS_SHAREDFOLDERS = 1
+#VBOX_WITH_WIN32_ADDITIONS_SHAREDFOLDERS = 1
 # Take the binaries of the OS/2 Additions.
-VBOX_WITH_OS2_ADDITIONS_BIN = 1
+#VBOX_WITH_OS2_ADDITIONS_BIN = 1
 # Use additions from the build server.
 #VBOX_WITH_ADDITIONS_FROM_BUILD_SERVER = 1
 # Don't bother with additions, overrides VBOX_WITH_ADDITIONS.
@@ -242,7 +242,7 @@
 # The SDL based GUI.
 VBOX_WITH_VBOXSDL = 1
 # The basic frontend (w/o Main).
-VBOX_WITH_VBOXBFE = 1
+#VBOX_WITH_VBOXBFE = 1
 # The Qt GUI.
 VBOX_WITH_QTGUI = 1
 # The Qt 4 GUI.
@@ -278,9 +278,9 @@
 # Enable the kchmviewer
 VBOX_WITH_KCHMVIEWER = 1
 # Build the testsuite.
-VBOX_WITH_TESTSUITE = 1
+#VBOX_WITH_TESTSUITE = 1
 # Build the testcases.
-VBOX_WITH_TESTCASES = 1
+#VBOX_WITH_TESTCASES = 1
 # Set this to not use COM or XPCOM in places where it can be avoided.
 #VBOX_WITHOUT_COM = 1
 # Set this to prefix all C symbols in XPCOM, to avoid dynamic linking problems
--- configure.anc	2008-09-05 13:07:49.000000000 +0200
+++ configure	2008-09-05 13:08:58.000000000 +0200
@@ -1959,14 +1959,14 @@
 # some things are not available in for OSE
 if [ $OSE -ge 1 ]; then
   cnf_append "VBOX_OSE" "1"
-  cnf_append "VBOX_WITH_TESTSUITE" ""
+#  cnf_append "VBOX_WITH_TESTSUITE" ""
   cnf_append "VBOX_WITH_WIN32_ADDITIONS" ""
 
-  if [ "$OS" = "linux" ]; then
-    cnf_append "VBOX_WITH_LINUX_ADDITIONS" "1"
-  else
-    cnf_append "VBOX_WITH_LINUX_ADDITIONS" ""
-  fi
+#  if [ "$OS" = "linux" ]; then
+#    cnf_append "VBOX_WITH_LINUX_ADDITIONS" "1"
+#  else
+#    cnf_append "VBOX_WITH_LINUX_ADDITIONS" ""
+#  fi
   echo >> $CNF
 fi
 
