diff -Naurp samba-3.2.1/source/configure samba-3.2.1.new/source/configure
--- samba-3.2.1/source/configure	2008-08-07 19:00:53.000000000 +0400
+++ samba-3.2.1.new/source/configure	2008-08-07 19:35:50.000000000 +0400
@@ -62107,6 +62107,7 @@ cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 
 		#include <krb5.h>
+		#include <stdlib.h>
 		int main(void) {
 			krb5_context context = NULL;
 			char *str = NULL;
