--- pari-2.3.3.p5/spkg-install.orig	2009-10-27 00:26:03.000000000 -0500
+++ pari-2.3.3.p5/spkg-install	2009-10-27 00:27:02.000000000 -0500
@@ -91,7 +91,7 @@
         exit 1
     fi
 
-    cp -f gp "$SAGE_LOCAL/bin/"    # avoid "make install"
+    $MAKE install-bin
 
     # Install documentation source (compressed), for use by gphelp.
     $MKDIR -p "$SAGE_LOCAL/share/pari/doc/"
@@ -146,7 +146,7 @@
         fi
 
         cd `config/objdir`
-        cp -f libpari* $SAGE_LOCAL/lib
+        $MAKE install-lib-dyn
         cd "$CUR"
 
         # Also another patch since paripriv.h won't compile right on OS X
