diff --git a/CHANGELOG b/CHANGELOG
index c3b83df..d3a726e 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,6 +1,11 @@
 ?/6/2006 autofs-5.0.0_beta5
 ---------------------------
 - re-instate v4 directory cleanup.
+- backout master map lookup changes made to beta3.
+- change default master map from /etc/auto.master to auto.master
+  so that we always use nsswitch to locate master map.
+- change default installed master map to include "+auto.master"
+  to pickup NIS master map.
 
 2/6/2006 autofs-5.0.0_beta4
 ---------------------------
diff --git a/README.v5.release b/README.v5.release
index 4d109c7..43f3576 100644
--- a/README.v5.release
+++ b/README.v5.release
@@ -8,12 +8,13 @@ Differences between version 4 and versio
   - the parameters for the `automount' daemon are now very different.
     See automount(8).
 
-- Master map default is /etc/auto.master. If it exists it will be
-  used. This can be overridden by setting the DEFAULT_MASTER_MAP_NAME
-  configuration variable. If the default master map can't be used
-  nsswitch sources are tried. Since the default master map is a file
-  the nsswitch source "files" is not checked even if it is present
-  in the nsswitch configuration.
+- The master map default is "auto.master" and nsswitch is used to
+  locate it. The line "+auto.master" has been added to the default
+  installed "/etc/auto.master" to ensure that those using NIS will
+  still find their master map. This is in line with other industry
+  automount implementations. The name of the default master map can
+  be overridden by setting the DEFAULT_MASTER_MAP_NAME configuration
+  variable.
 
 - The `automount' daemon is now a multi-threaded application and so
   appears as a single process (unless a thread process display option
diff --git a/daemon/lookup.c b/daemon/lookup.c
index 7d55072..1a96e30 100644
--- a/daemon/lookup.c
+++ b/daemon/lookup.c
@@ -99,8 +99,9 @@ static int read_master_map(struct master
 	char *path, *save_name;
 	int result;
 
-	if (strcasecmp(type, "files"))
+	if (strcasecmp(type, "files")) {
 		return do_read_master(master, type, age);
+	}
 
 	/* 
 	 * This is a special case as we need to append the
@@ -137,7 +138,6 @@ int lookup_nss_read_master(struct master
 	struct list_head nsslist;
 	struct list_head *head, *p;
 	int result = NSS_STATUS_UNKNOWN;
-	char *save_master_name = NULL;
 
 	/* If it starts with a '/' it has to be a file or LDAP map */
 	if (*master->name == '/') {
@@ -145,24 +145,13 @@ int lookup_nss_read_master(struct master
 			debug(LOGOPT_NONE,
 			      "reading master ldap %s", master->name);
 			result = do_read_master(master, "ldap", age);
-			return !result;
 		} else {
-			struct stat st;
-			int ret;
-
-			/*
-			 * If the default file map doesn't exist then
-			 * we need to look to nsswitch and exclude the
-			 * files source.
-			 */
-			ret = stat(master->name, &st);
-			if (ret == 0) {
-				debug(LOGOPT_NONE,
-				      "reading master file %s", master->name);
-				result = do_read_master(master, "file", age);
-				return !result;
-			}
+			debug(LOGOPT_NONE,
+			      "reading master file %s", master->name);
+			result = do_read_master(master, "file", age);
 		}
+
+		return !result;
 	} else {
 		char *name = master->name;
 		char *tmp;
@@ -204,27 +193,6 @@ int lookup_nss_read_master(struct master
 		return 0;
 	}
 
-	/*
-	 * A master map file path was specified but didn't exist
-	 * so for good measure we save the path and use the name
-	 * to look in nsswitch sources other that the files source.
-	 * This is possibly non-standard but will likely be
-	 * appreciated because it gives similar semantics to the
-	 * old version.
-	 */
-	if (*master->name == '/') {
-		char *tmp, *start;
-		int len;
-
-		start = strrchr(master->name, '/') + 1;
-		len = strlen(start);
-		tmp = alloca(len + 1);
-		if (len && tmp) {
-			save_master_name = master->name;
-			master->name = tmp;
-		}
-	}
-
 	/* First one gets it */
 	head = &nsslist;
 	list_for_each(p, head) {
@@ -233,16 +201,6 @@ int lookup_nss_read_master(struct master
 
 		this = list_entry(p, struct nss_source, list);
 
-		/*
-		 * If the default file master map has been changed
-		 * to something without a leading "/" or it didn't
-		 * exist (see above) so we probably should ignore
-		 * the files source in nsswitch. Again, this should
-		 * be closer to the semantics of the older version.
-		 */
-		if (!strcasecmp(this->source, "files"))
-			continue;
-
 		debug(LOGOPT_NONE,
 		      "reading master %s %s", this->source, master->name);
 
@@ -256,8 +214,6 @@ int lookup_nss_read_master(struct master
 		status = check_nss_result(this, result);
 		if (status >= 0) {
 			free_sources(&nsslist);
-			if (save_master_name)
-				master->name = save_master_name;
 			return status;
 		}
 	}
@@ -265,9 +221,6 @@ int lookup_nss_read_master(struct master
 	if (!list_empty(&nsslist))
 		free_sources(&nsslist);
 
-	if (save_master_name)
-		master->name = save_master_name;
-
 	return !result;
 }
 
diff --git a/include/defaults.h b/include/defaults.h
index 360aaba..34a0f6c 100644
--- a/include/defaults.h
+++ b/include/defaults.h
@@ -21,7 +21,7 @@ #ident "$Id: defaults.h,v 1.4 2006/04/06
 #ifndef DEFAULTS_H
 #define DEFAULTS_H
 
-#define DEFAULT_MASTER_MAP_NAME	"/etc/auto.master"
+#define DEFAULT_MASTER_MAP_NAME	"auto.master"
 
 #define DEFAULT_TIMEOUT		600
 #define DEFAULT_BROWSE_MODE	1
diff --git a/samples/auto.master b/samples/auto.master
index 78f2f40..942d18f 100644
--- a/samples/auto.master
+++ b/samples/auto.master
@@ -5,7 +5,16 @@ # Sample auto.master file
 # This is an automounter map and it has the following format
 # key [ -mount-options-separated-by-comma ] location
 # For details of the format look at autofs(5).
-#/misc	/etc/auto.misc --timeout=60
-#/smb	/etc/auto.smb
+#
 /misc	/etc/auto.misc
 /net	/etc/auto.net
+#
+# Include central master map if it can be found using
+# nsswitch sources.
+#
+# Note that if there are entries for /net or /misc (as
+# above) in the included master map any keys that are the
+# same will not be seen as the first read key seen takes
+# precedence.
+#
++auto.master
diff --git a/samples/autofs.conf.default.in b/samples/autofs.conf.default.in
index 50ccb6e..9c4b1ca 100644
--- a/samples/autofs.conf.default.in
+++ b/samples/autofs.conf.default.in
@@ -3,7 +3,7 @@ # Define default options for autofs.
 #
 # DEFAULT_MASTER_MAP_NAME - default map name for the master map.
 #
-#DEFAULT_MASTER_MAP_NAME="/etc/auto.master"
+#DEFAULT_MASTER_MAP_NAME="auto.master"
 #
 # DEFAULT_TIMEOUT - set the default mount timeout (default 600).
 #
