diff -Naurp crengine/src/lvdocview.cpp crengine.new/src/lvdocview.cpp
--- crengine/src/lvdocview.cpp	2008-12-13 11:52:04.000000000 +0300
+++ crengine.new/src/lvdocview.cpp	2008-12-13 11:49:22.000000000 +0300
@@ -1580,7 +1580,7 @@ void LVDocView::Render( int dx, int dy, 
         if ( m_showCover )
             pages->add( new LVRendPageInfo( dy ) );
         LVRendPageContext context( pages, dy );
-        CRLog::trace("calling render() for document %08X font=%08X", (unsigned int)m_doc, (unsigned int)m_font.get() );
+        CRLog::trace("calling render() for document %08X font=%08X", (unsigned long)m_doc, (unsigned long)m_font.get() );
         m_doc->render( context, dx, m_showCover ? dy + m_pageMargins.bottom*4 : 0, m_font, m_def_interline_space );
 
     #if 0
