--- unoconv	2011-05-19 12:20:49.000000000 +0200
+++ unoconv	2011-05-19 12:23:44.003394684 +0200
@@ -24,6 +24,8 @@
              glob.glob('/usr/lib*/openoffice*/basis*/program') + \
              glob.glob('/usr/lib*/ooo*/program') + \
              glob.glob('/usr/lib*/ooo*/basis*/program') + \
+             glob.glob('/usr/lib*/libreoffice/program') + \
+             glob.glob('/usr/lib*/libreoffice/basis*/program') + \
              glob.glob('/usr/local/openoffice*/program') + \
              glob.glob('/usr/local/openoffice*/basis*/program') + \
              glob.glob('/usr/local/ooo*/program') + \
