--- unoconv	2010-10-20 11:56:26.000000000 +0200
+++ unoconv	2011-05-17 21:14:38.395241116 +0200
@@ -64,7 +64,18 @@
             sys.path.append(oolibpath)
             import uno, unohelper
             ### Export an environment that OpenOffice is pleased to work with
-            os.environ['LD_LIBRARY_PATH'] = oolibpath + os.pathsep + os.environ['LD_LIBRARY_PATH']
+
+            # Added Wrapper for unset LD_LIBRARY_PATH
+            if 'LD_LIBRARY_PATH' in os.environ:
+                lib_path = os.environ['LD_LIBRARY_PATH']
+                if oolibpath in lib_path:
+                    pass
+                else:
+                    os.environ['LD_LIBRARY_PATH'] += os.pathsep + oolibpath
+            else:
+                os.environ['LD_LIBRARY_PATH'] = oolibpath
+            # This old line fails with python 2.7, the above should work with 2.x
+            # os.environ['LD_LIBRARY_PATH'] = oolibpath + os.pathsep + os.environ['LD_LIBRARY_PATH']
             break
         except ImportError, e:
             sys.path.remove(oolibpath)
