diff --git a/modules/lookup_hosts.c b/modules/lookup_hosts.c
index 1a16b96..8855ed7 100644
--- a/modules/lookup_hosts.c
+++ b/modules/lookup_hosts.c
@@ -189,7 +189,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 		if (!ret)
 			return NSS_STATUS_SUCCESS;
 
-		debug(ap->logopt, MODPREFIX "mount failed - update exports list");
+		return NSS_STATUS_TRYAGAIN;
 	}
 done:
 	/*
