--- kphotoalbum-3.2_pre20080710/ImportExport/Import.cpp~	2008-07-11 13:55:27.000000000 +0400
+++ kphotoalbum-3.2_pre20080710/ImportExport/Import.cpp	2008-07-11 13:58:51.000000000 +0400
@@ -9,6 +9,7 @@
 #include <kio/jobuidelegate.h>
 #include <kio/jobclasses.h>
 #include "MainWindow/Window.h"
+#include "klocale.h"
 
 using namespace ImportExport;
 
--- kphotoalbum-3.2_pre20080710/Exif/Info.cpp~	2008-07-11 14:05:41.000000000 +0400
+++ kphotoalbum-3.2_pre20080710/Exif/Info.cpp	2008-07-11 14:05:44.000000000 +0400
@@ -125,7 +125,7 @@
     std::ostringstream s;
     for ( Exiv2::IfdId kind = Exiv2::canonIfdId; kind < Exiv2::lastIfdId;
             kind = static_cast<Exiv2::IfdId>( kind + 1 ) ) {
-        Exiv2::ExifTags::makerTaglist( s, kind );
+        Exiv2::ExifTags::taglist( s, kind );
     }
 
     // IPTC tags use yet another format...
