Index: knetworkmanager/src/knetworkmanager-devicestore.h
===================================================================
--- knetworkmanager/src/knetworkmanager-devicestore.h.orig
+++ knetworkmanager/src/knetworkmanager-devicestore.h
@@ -28,6 +28,8 @@
 
 #include <NetworkManager/NetworkManager.h>
 
+#include <sys/types.h>
+
 #include "knetworkmanager.h"
 #include "knetworkmanager-device.h"
 #include "knetworkmanager-dialup.h"
Index: knetworkmanager/src/Makefile.am
===================================================================
--- knetworkmanager/src/Makefile.am	(Revision 637674)
+++ knetworkmanager/src/Makefile.am	(Arbeitskopie)
@@ -30,7 +30,7 @@
 lib_LTLIBRARIES =
 kdeinit_LTLIBRARIES = knetworkmanager.la
 
-KDE_CXXFLAGS = -UQT_NO_ASCII_CAST
+KDE_CXXFLAGS = -std=gnu++98 -UQT_NO_ASCII_CAST
 
 # the application source, library search path, and link libraries
 knetworkmanager_la_SOURCES = main.cpp knetworkmanager.cpp knetworkmanager-dbus.cpp \
