diff -Naur xneur-0.12.0/lib/notify/popup.c xneur-0.12.0.new/lib/notify/popup.c
--- xneur-0.12.0/lib/notify/popup.c	2011-01-27 13:25:42.000000000 +0300
+++ xneur-0.12.0.new/lib/notify/popup.c	2011-03-11 12:44:43.691651994 +0300
@@ -74,7 +74,7 @@
 		popup_body->content = NULL;
 	}
 	
-	NotifyNotification *notify = notify_notification_new(popup_body->header, popup_body->content, icon, NULL);
+	NotifyNotification *notify = notify_notification_new(popup_body->header, popup_body->content, icon);
 
 	notify_notification_set_category(notify, type);
 	notify_notification_set_urgency(notify, urgency);
