diff -urN linux-2.4.19/Makefile linux-2.4.19.SuSE/Makefile
--- linux-2.4.19/Makefile	Fri Sep  6 17:07:04 2002
+++ linux-2.4.19.SuSE/Makefile	Fri Sep  6 17:07:29 2002
@@ -3,8 +3,6 @@
 SUBLEVEL = 20
 EXTRAVERSION = -wolk4.0s
 
-KERNELRELEASE=$(VERSION).$(PATCHLEVEL).$(SUBLEVEL)$(EXTRAVERSION)
-
 # SUBARCH tells the usermode build what the underlying arch is.  That is set
 # first, and if a usermode build is happening, the "ARCH=um" on the command
 # line overrides the setting of ARCH below.  If a native build is happening,
@@ -88,6 +86,16 @@
 do-it-all:	config
 endif
 
+#### Appendices for extraversion
+# Test for Win4Lin
+ifdef CONFIG_MKI
+ WIN4LINVERSION=-win4lin
+else
+ WIN4LINVERSION=
+endif
+
+KERNELRELEASE=$(VERSION).$(PATCHLEVEL).$(SUBLEVEL)$(EXTRAVERSION)$(WIN4LINVERSION)
+
 #
 # INSTALL_PATH specifies where to place the updated kernel and system map
 # images.  Uncomment if you want to place them anywhere other than root.
