--- old/include/sound/wrappers.h	2003-09-02 21:46:45.000000000 +0200
+++ new/include/sound/wrappers.h	2003-09-02 22:35:28.000000000 +0200
@@ -33,13 +33,6 @@ void snd_compat_request_module(const cha
 #endif
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 5, 4)
-#include <linux/fs.h>
-static inline struct proc_dir_entry *PDE(const struct inode *inode)
-{
-	return (struct proc_dir_entry *) inode->u.generic_ip;
-}
-#endif
 #include <asm/io.h>
 #if !defined(isa_virt_to_bus)
 #if defined(virt_to_bus) || defined(__alpha__)
@@ -67,12 +60,6 @@ static inline void synchronize_irq_wrapp
 #undef synchronize_irq
 #define synchronize_irq(irq)	synchronize_irq_wrapper(irq)
 #endif /* LINUX_VERSION_CODE < 2.5.28 */
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 5, 68)
-#define IRQ_NONE	/*void*/
-#define IRQ_HANDLED	/*void*/
-#define IRQ_RETVAL(x)	/*void*/
-typedef void irqreturn_t;
-#endif /* LINUX_VERSION_CODE < 2.5.68 */
 
 #include <linux/devfs_fs_kernel.h>
 #ifdef CONFIG_DEVFS_FS
