diff -urNp --exclude-from=/home/mitica/quintela/config/misc/dontdiff --exclude=ipsec --exclude='defconfig*' w5/sound/core/seq/seq_clientmgr.c w4/sound/core/seq/seq_clientmgr.c
--- w5/sound/core/seq/seq_clientmgr.c	2003-07-25 19:22:26.000000000 +0200
+++ w4/sound/core/seq/seq_clientmgr.c	2003-08-07 21:15:34.000000000 +0200
@@ -143,7 +143,9 @@ client_t *snd_seq_client_use_ptr(int cli
 					if (seq_client_load[idx] < 0)
 						break;
 					if (seq_client_load[idx] == clientid) {
-						request_module("snd-seq-client-%i", clientid);
+						char str[32];
+						sprintf(str, "snd-seq-client-%i", clientid);
+						request_module(str);
 						break;
 					}
 				}
diff -urNp --exclude-from=/home/mitica/quintela/config/misc/dontdiff --exclude=ipsec --exclude='defconfig*' w5/sound/core/sound.c w4/sound/core/sound.c
--- w5/sound/core/sound.c	2003-08-07 21:17:11.000000000 +0200
+++ w4/sound/core/sound.c	2003-08-07 20:08:29.000000000 +0200
@@ -528,6 +528,14 @@ EXPORT_SYMBOL(snd_wrapper_kfree);
 EXPORT_SYMBOL(snd_wrapper_vmalloc);
 EXPORT_SYMBOL(snd_wrapper_vfree);
 #endif
+#ifndef CONFIG_HAVE_STRLCPY
 EXPORT_SYMBOL(snd_compat_strlcpy);
 EXPORT_SYMBOL(snd_compat_strlcat);
+#endif
+
 EXPORT_SYMBOL(snd_compat_request_module);
+
+#if defined(CONFIG_DEVFS_FS) && LINUX_VERSION_CODE < KERNEL_VERSION(2, 5, 67)
+EXPORT_SYMBOL(snd_compat_devfs_mk_dir);
+EXPORT_SYMBOL(snd_compat_devfs_mk_cdev);
+#endif
diff -urNp --exclude-from=/home/mitica/quintela/config/misc/dontdiff --exclude=ipsec --exclude='defconfig*' w5/sound/core/wrappers.c w4/sound/core/wrappers.c
--- w5/sound/core/wrappers.c	2003-04-08 17:33:59.000000000 +0200
+++ w4/sound/core/wrappers.c	2003-08-07 19:43:53.000000000 +0200
@@ -25,6 +25,7 @@
 #include <linux/ioport.h>
 #include <linux/vmalloc.h>
 #include <linux/fs.h>
+#include <linux/devfs_fs_kernel.h>
 
 #ifdef CONFIG_SND_DEBUG_MEMORY
 void *snd_wrapper_kmalloc(size_t size, int flags)
