--- linux-2.4.18.SuSE/sound/core/pcm_lib.c	27 Jun 2002 11:39:03 -0000	1.15
+++ linux-2.4.18.SuSE/sound/core/pcm_lib.c	3 Jul 2002 15:16:00 -0000
@@ -120,7 +120,7 @@
 	delta = hw_ptr_interrupt - new_hw_ptr;
 	if (delta > 0) {
 		if ((snd_pcm_uframes_t)delta < runtime->buffer_size / 2) {
-			snd_printd("Unexpected hw_pointer value (stream = %i, delta: -%ld, max jitter = %ld): wrong interrupt acknowledge?\n", substream->stream, (long) delta, runtime->buffer_size / 2);
+			snd_printdd("Unexpected hw_pointer value (stream = %i, delta: -%ld, max jitter = %ld): wrong interrupt acknowledge?\n", substream->stream, (long) delta, runtime->buffer_size / 2);
 			return 0;
 		}
 		runtime->hw_ptr_base += runtime->buffer_size;
@@ -184,7 +184,7 @@
 	delta = old_hw_ptr - new_hw_ptr;
 	if (delta > 0) {
 		if ((snd_pcm_uframes_t)delta < runtime->buffer_size / 2) {
-			snd_printd("Unexpected hw_pointer value (stream = %i, delta: -%ld, max jitter = %ld): wrong interrupt acknowledge?\n", substream->stream, (long) delta, runtime->buffer_size / 2);
+			snd_printdd("Unexpected hw_pointer value (stream = %i, delta: -%ld, max jitter = %ld): wrong interrupt acknowledge?\n", substream->stream, (long) delta, runtime->buffer_size / 2);
 			return 0;
 		}
 		runtime->hw_ptr_base += runtime->buffer_size;
