--- opencv-1.0.0/configure.in	2006-11-03 16:54:33.000000000 +0100
+++ opencv-1.0.0/configure.in.new	2009-06-03 01:50:31.236954981 +0200
@@ -736,16 +736,16 @@
 ######################################################################
 # Parallelization
 #
-#CXXOPENMP="no"
-#
-#AC_ARG_WITH(openmp, dnl
-#  [--with-openmp=COMPILER  use OpenMP supporting compiler [default=no] ],
-#    [CXXOPENMP="$withval"],[CXXOPENMP="no"])
-#if test "$CXXOPENMP" = "no"; then
-#    CXXOPENMP="$CXX"
-#fi
-#
-#AC_SUBST(CXXOPENMP)
+CXXOPENMP="no"
+
+AC_ARG_WITH(openmp, dnl
+  [--with-openmp=COMPILER  use OpenMP supporting compiler [default=no] ],
+    [CXXOPENMP="$withval"],[CXXOPENMP="no"])
+if test "$CXXOPENMP" = "no"; then
+    CXXOPENMP="$CXX"
+fi
+
+AC_SUBST(CXXOPENMP)
 
 AC_SUBST(DEBUG)
 
