--- patches/src680/apply.orig	2006-06-11 22:04:29.000000000 +0200
+++ patches/src680/apply	2006-06-11 22:05:38.000000000 +0200
@@ -737,7 +737,7 @@
 buildfix-connectivity-kab-stdio.diff, i#62715, pmladek
 
 # add missing #include <stdio.h> in autodoc, breaks when building with STLport5
-autodoc-add-missing-stdio-h.diff
+#autodoc-add-missing-stdio-h.diff
 
 # do not pack any potential .orig files with ppds
 psprint_config-no-orig.diff, jholesov
@@ -876,6 +876,8 @@
 [ BFBuildBits ]
 SectionOwner => pmladek
 
+# binfilter stlport fixes
+binfilter-stl.diff
 # fixes warnings about problematic return values
 warning-return-values-binfilter.diff
 # fixes warnings about that a variable is used unitialized
