--- k3b-2.0_pre20081208/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp.orig	2008-12-10 06:52:00.000000000 +0100
+++ k3b-2.0_pre20081208/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2008-12-10 07:19:34.271258226 +0100
@@ -198,7 +198,7 @@
         return i18n("Windows Media v1");
     case CODEC_ID_WMAV2:
         return i18n("Windows Media v2");
-    case CODEC_ID_MP3LAME:
+    case CODEC_ID_MP3:
         return i18n("MPEG 1 Layer III");
     case CODEC_ID_AAC:
         return i18n("Advanced Audio Coding (AAC)");
@@ -291,7 +291,7 @@
 
         d->outputBufferPos = d->outputBuffer;
 
-        int len = avcodec_decode_audio(
+        int len = avcodec_decode_audio2(
 #ifdef FFMPEG_BUILD_PRE_4629
             &d->formatContext->streams[0]->codec,
 #else
