diff -dPNur gnome-translate-0.99/src/gt-app.c gnome-translate-0.99-new/src/gt-app.c
--- gnome-translate-0.99/src/gt-app.c	2005-01-28 13:31:53.000000000 +0100
+++ gnome-translate-0.99-new/src/gt-app.c	2007-10-20 21:30:12.000000000 +0200
@@ -1168,8 +1168,7 @@
 						     GTK_MESSAGE_WARNING,
 						     GTK_BUTTONS_YES_NO,
 						     _("Overwrite file?"),
-						     secondary,
-						     NULL);
+						     secondary);
 	  g_free(secondary);
 
 	  gtk_dialog_set_default_response(GTK_DIALOG(confirmation_dialog), GTK_RESPONSE_NO); /* safe default */
diff -dPNur gnome-translate-0.99/src/gt-util.c gnome-translate-0.99-new/src/gt-util.c
--- gnome-translate-0.99/src/gt-util.c	2005-01-28 13:32:27.000000000 +0100
+++ gnome-translate-0.99-new/src/gt-util.c	2007-10-20 21:30:34.000000000 +0200
@@ -53,8 +53,7 @@
 				GTK_MESSAGE_ERROR,
 				GTK_BUTTONS_OK,
 				primary,
-				secondary,
-				NULL);
+				secondary);
 
   if (blocking)
     {
