--- configure.orig	2010-09-19 21:16:00.223008127 +0200
+++ configure	2010-09-19 21:29:10.194008127 +0200
@@ -18844,7 +18844,7 @@
 
      LDFLAGS_save=$LDFLAGS
    # Provide known paths where distribs/OS can store JNI libs
-        LDFLAGS="-L/usr/lib/jni -L/usr/lib64/jni/ -L$SCI_SRCDIR/bin/ -L$SCI_SRCDIR/thirdparty/  -lpthread $HDF5_LIBS"
+        LDFLAGS="-L/usr/lib/jni -L/usr/lib64/jni/ -L$SCI_SRCDIR/bin/ -L$SCI_SRCDIR/thirdparty/  -lpthread $HDF5_LIBS -L`java-config -i hdf-java`"
    # -lpthread because of packaging bug in jhdf5
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for h5JNIFatalError in -ljhdf5" >&5
 $as_echo_n "checking for h5JNIFatalError in -ljhdf5... " >&6; }
