diff -uNr Makefile.in Makefile.in
--- Makefile.in	2009-02-21 21:44:10.000000000 +1100
+++ Makefile.in	2009-04-10 09:11:58.377289821 +1000
@@ -285,7 +285,7 @@
 $(wildcard $(OBJSXBMC)) :| compile
 
 LIBS += xbmc/lib/libBoxee/libBoxee.a xbmc/lib/libBoxee/libtxpath.a -lcurl
-LIBS += xbmc/lib/libsmb/libtalloc-i486-linux.a xbmc/lib/libsmb/libtdb-i486-linux.a  xbmc/lib/libsmb/libsmbsharemodes-i486-linux.a xbmc/lib/libsmb/libwbclient-i486-linux.a
+LIBS += xbmc/lib/libsmb/libtalloc-@ARCH@.a xbmc/lib/libsmb/libtdb-@ARCH@.a  xbmc/lib/libsmb/libsmbsharemodes-@ARCH@.a xbmc/lib/libsmb/libwbclient-@ARCH@.a
 
 Boxee: $(wildcard $(OBJSXBMC)) $(LIBS) | compile
 	g++ $(DEBUG_FLAGS) -o Boxee $(OBJSXBMC) $(LIBS) -rdynamic
diff -uNr xbmc/lib/cximage-6.0/tiff/Makefile xbmc/lib/cximage-6.0/tiff/Makefile
--- xbmc/lib/cximage-6.0/tiff/Makefile	2008-09-14 21:27:02.000000000 +1000
+++ xbmc/lib/cximage-6.0/tiff/Makefile	2009-04-10 09:30:37.633304076 +1000
@@ -2,6 +2,10 @@
 
 OBJS=fax3sm_winnt.o tif_aux.o tif_close.o tif_codec.o tif_compress.o tif_dir.o tif_dirinfo.o tif_dirread.o tif_dirwrite.o tif_dumpmode.o tif_error.o tif_fax3.o tif_flush.o tif_getimage.o tif_jpeg.o tif_luv.o tif_lzw.o tif_next.o tif_ojpeg.o tif_open.o tif_packbits.o tif_pixarlog.o tif_predict.o tif_print.o tif_read.o tif_strip.o tif_swab.o tif_thunder.o tif_tile.o tif_version.o tif_warning.o tif_write.o tif_zip.o
 
+ifeq ($(ARCH), x86_64-linux)
+    CFLAGS+= -fPIC
+endif
+
 %o : %c
 	$(CC) $(CFLAGS) -c -o $@ $< 
 
diff -uNr xbmc/utils/md5.h xbmc/utils/md5.h
--- xbmc/utils/md5.h	2008-07-27 18:09:47.000000000 +1000
+++ xbmc/utils/md5.h	2009-04-10 09:34:38.405396766 +1000
@@ -44,7 +44,11 @@
 #include "StdString.h"
 
 /* typedef a 32 bit type */
-typedef unsigned long int UINT4;
+#ifdef _WIN32PC
+typedef __int32 UINT4;
+#else
+typedef uint32_t UINT4;
+#endif
 
 /* Data structure for MD5 (Message Digest) computation */
 typedef struct {
diff -uNr xbmc/lib/libsmb/libsmbclient.h xbmc/lib/libsmb/libsmbclient.h
--- xbmc/lib/libsmb/libsmbclient.h	2008-08-18 16:17:01.000000000 +1000
+++ xbmc/lib/libsmb/libsmbclient.h	2009-04-24 21:18:50.208944095 +1000
@@ -2513,7 +2513,11 @@
 extern "C" {
 #endif
 int
+#ifdef __x86_64__
+smbc_urldecode(char *dest, char * src, size_t max_dest_len);
+#else
 SMBC_urldecode(char *dest, char * src, size_t max_dest_len);
+#endif
 #ifdef __cplusplus
 }
 #endif
@@ -2540,7 +2544,11 @@
 extern "C" {
 #endif
 int
+#ifdef __x86_64__
+smbc_urlencode(char * dest, char * src, int max_dest_len);
+#else
 SMBC_urlencode(char * dest, char * src, int max_dest_len);
+#endif
 #ifdef __cplusplus
 }
 #endif
diff -uNr xbmc/FileSystem/FileSmb.cpp xbmc/FileSystem/FileSmb.cpp
--- xbmc/FileSystem/FileSmb.cpp	2009-01-02 09:29:08.000000000 +1100
+++ xbmc/FileSystem/FileSmb.cpp	2009-04-24 09:49:00.349655523 +1000
@@ -283,6 +283,8 @@
   char *strSrc = strdup(value.c_str()); // urlencode requires char* and not const char *. just to be on the safe side
 #ifdef _WIN32
   smbc_urlencode(buffer, strSrc, buffer_len);
+#elif defined(__x86_64__)
+  smbc_urlencode(buffer, strSrc, buffer_len);
 #else
   SMBC_urlencode(buffer, strSrc, buffer_len);
 #endif
@@ -300,6 +302,8 @@
   char *strSrc = strdup(value.c_str()); // urlencode requires char* and not const char *. just to be on the safe side
 #ifdef _WIN32
   smbc_urldecode(buffer, strSrc, buffer_len-1);
+#elif defined(__x86_64__)
+  smbc_urldecode(buffer, strSrc, buffer_len-1);
 #else
   SMBC_urldecode(buffer, strSrc, buffer_len-1);
 #endif
