diff -uNr tools/Linux/boxee.desktop tools/Linux/boxee.desktop
--- tools/Linux/boxee.desktop	1970-01-01 10:00:00.000000000 +1000
+++ tools/Linux/boxee.desktop	2009-04-10 09:13:40.003694670 +1000
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Boxee
+Comment=Boxee social media center
+Exec=/opt/bin/boxee
+Icon=boxee.png
+Terminal=false
+Type=Application
+Categories=Application;AudioVideo;Player;
+StartupNotify=true

