diff -urN drm/linux-core/Makefile drm.work/linux-core/Makefile
--- drm/linux-core/Makefile	2007-10-19 19:33:37.000000000 -0400
+++ drm.work/linux-core/Makefile	2007-10-19 20:07:46.000000000 -0400
@@ -221,14 +221,17 @@
 $(MODULE_LIST)::
 	make DRM_MODULES=$@ modules
 
+install:
+	mkdir -p $(DESTDIR)/lib/modules/$(RUNNING_REL)/kernel/drivers/char/drm
+	cp -p $(MODULE_LIST:.o=.ko) $(DESTDIR)/lib/modules/$(RUNNING_REL)/kernel/drivers/char/drm
+
 # Build test utilities
 
 PRGCFLAGS = $(CFLAGS) -g -ansi -pedantic -DPOSIX_C_SOURCE=199309L \
 	    -D_POSIX_SOURCE -D_XOPEN_SOURCE -D_BSD_SOURCE -D_SVID_SOURCE \
-	    -I. -I../../..
+	    -I. -I../../.. -I../libdrm
 
-DRMSTATLIBS = -L../../.. -L.. -ldrm -lxf86_os \
-	      -L../../../../dummylib -ldummy -lm
+DRMSTATLIBS = -L../../.. -L.. -L../libdrm
 
 programs: $(PROGS)
 
@@ -236,10 +239,7 @@
 	$(CC) $(PRGCFLAGS) $< -o $@
 
 drmstat: drmstat.c
-	$(CC) $(PRGCFLAGS) $< -o $@ $(DRMSTATLIBS)
-
-install:
-	make -C $(LINUXDIR) $(GETCONFIG) SUBDIRS=`pwd` DRMSRCDIR=`pwd` modules_install
+	$(CC) $(PRGCFLAGS) $< -o $@ ../libdrm/.libs/*.o $(DRMSTATLIBS)
 
 else
 
