diff --git a/linux-core/nouveau_drv.c b/linux-core/nouveau_drv.c
index e9623eb..dc5e3d0 100644
--- a/linux-core/nouveau_drv.c
+++ b/linux-core/nouveau_drv.c
@@ -108,6 +108,34 @@ static void __exit nouveau_exit(void)
 	drm_exit(&driver);
 }
 
+#if defined(__powerpc__)
+struct lohistruct {
+	unsigned high, low;
+};
+
+typedef union {
+	struct lohistruct s;
+	unsigned long long ll;
+} llunion;
+
+int __ucmpdi2(unsigned long long a,unsigned long long b)
+{
+	llunion au = {.ll = a};
+	llunion bu = {.ll = b};
+
+	if ((unsigned int) au.s.high < (unsigned int) bu.s.high)
+		return 0;
+	else if ((unsigned int) au.s.high > (unsigned int) bu.s.high)
+		return 2;
+	if ((unsigned int) au.s.low < (unsigned int) bu.s.low)
+		return 0;
+	else if ((unsigned int) au.s.low > (unsigned int) bu.s.low)
+		return 2;
+	return 1;
+
+}
+#endif
+
 module_init(nouveau_init);
 module_exit(nouveau_exit);
 
