=== modified file 'python-updater'
--- python-updater	2010-02-18 12:25:28 +0000
+++ python-updater	2010-02-18 13:23:45 +0000
@@ -456,11 +456,16 @@
 
 # Iterate through the contents of all the installed packages.
 # ${PKG_DBDIR} must be followed by '/' to avoid problems when ${PKG_DBDIR} is a symlink.
+PVR_missing=""
 for content in $(find ${PKG_DBDIR}/ -name CONTENTS | sort); do
     environment_file="${content/CONTENTS/environment.bz2}"
 
     # Extract some variables. PYTHON_ABIS and PYTHON_REQUESTED_ACTIVE_VERSION are optional.
-    get_vdb_variable PVR "${environment_file}" || die "PVR missing"
+    # If any PVR variable is missing, keep collecting missing PVRs and report them together below
+    get_vdb_variable PVR "${environment_file}" || PVR_missing="${PVR_missing} ${environment_file}"
+    if [ "${PVR_missing}" != "" ]; then
+     continue
+    fi
     get_vdb_variable PYTHON_ABIS "${environment_file}"
     get_vdb_variable PYTHON_REQUESTED_ACTIVE_VERSION "${environment_file}"
 
@@ -634,6 +639,10 @@
     fi
 done
 
+if [ "${PVR_missing}" != "" ]; then
+ die "Error: PVR missing in ${PVR_missing}. Please re-emerge the corresponding packages."
+fi
+
 # Pipe to command if we have one
 if [[ -n "${PIPE_COMMAND}" ]]; then
     echo "${PKGS_TO_REMERGE}" | ${PIPE_COMMAND}

