--- Makefile.in.orig	2008-09-21 12:51:30.000000000 +0400
+++ Makefile.in	2008-09-21 12:55:48.000000000 +0400
@@ -40,10 +40,6 @@ EC_DIRS= \
 XBMCTEX_DIRS= \
 	tools/XBMCTex
 
-DVDPCODECS_DIRS= \
-	xbmc/cores/dvdplayer/Codecs \
-	xbmc/lib/libass/xbmc \
-
 PAPCODECS_DIRS= \
 	xbmc/cores/paplayer/AC3Codec \
 	xbmc/cores/paplayer/ADPCMCodec \
@@ -59,7 +55,6 @@ PAPCODECS_DIRS= \
 	xbmc/cores/paplayer/SPCCodec/SNES/SNESAPU \
 	xbmc/cores/paplayer/vgmstream \
 	xbmc/cores/paplayer/MP3Codec \
-	xbmc/cores/paplayer/flac-1.2.1/ \
 	xbmc/cores/paplayer/MACDll \
 	xbmc/cores/paplayer/vorbisfile
 
@@ -82,7 +77,7 @@ VIS_DIRS=\
 PM3_MEDIA=skin/Project\ Mayhem\ III/media
 PM3HD_MEDIA=skin/PM3.HD/media
 
-DIRS= $(BIN_DIRS) $(EC_DIRS) $(XBMCTEX_DIRS) $(DVDPCODECS_DIRS) $(PAPCODECS_DIRS) \
+DIRS= $(BIN_DIRS) $(EC_DIRS) $(XBMCTEX_DIRS) $(PAPCODECS_DIRS) \
 			$(LIB_DIRS) $(SS_DIRS) $(VIS_DIRS)
 
 LIBS=@LIBS@
@@ -164,8 +159,6 @@ dvdplayer:
 	$(MAKE) -C xbmc/cores/VideoRenderers 
 	$(MAKE) -C xbmc/cores/VideoRenderers/VideoShaders 
 dvdpcodecs: dllloader
-	$(MAKE) -C xbmc/cores/dvdplayer/Codecs
-	$(MAKE) -C xbmc/lib/libass/xbmc
 cdrip:
 	$(MAKE) -C xbmc/cdrip
 libcmyth: dllloader 
@@ -192,7 +185,6 @@ papcodecs: dllloader linux dvdpcodecs
 	$(MAKE) -C xbmc/cores/paplayer/AACCodec
 	$(MAKE) -C xbmc/cores/paplayer/AC3Codec
 	$(MAKE) -C xbmc/cores/paplayer/ADPCMCodec
-	$(MAKE) -C xbmc/cores/paplayer/flac-1.2.1
 	$(MAKE) -C xbmc/cores/paplayer/GYMCodec
 	$(MAKE) -C xbmc/cores/paplayer/MACDll
 	$(MAKE) -C xbmc/cores/paplayer/vgmstream
